description: Disables omission of URL elements in Omnibox

--- a/net/base/net_util_icu.cc
+++ b/net/base/net_util_icu.cc
@@ -570,10 +570,10 @@ void AppendFormattedComponent(const std::string& spec,
 
 }  // namespace
 
-const FormatUrlType kFormatUrlOmitNothing                     = 0;
-const FormatUrlType kFormatUrlOmitUsernamePassword            = 1 << 0;
-const FormatUrlType kFormatUrlOmitHTTP                        = 1 << 1;
-const FormatUrlType kFormatUrlOmitTrailingSlashOnBareHostname = 1 << 2;
+const FormatUrlType kFormatUrlOmitNothing                     = 1;
+const FormatUrlType kFormatUrlOmitUsernamePassword            = 0 << 0;
+const FormatUrlType kFormatUrlOmitHTTP                        = 0 << 1;
+const FormatUrlType kFormatUrlOmitTrailingSlashOnBareHostname = 0 << 2;
 const FormatUrlType kFormatUrlOmitAll = kFormatUrlOmitUsernamePassword |
     kFormatUrlOmitHTTP | kFormatUrlOmitTrailingSlashOnBareHostname;
 

