# Add more URL schemes allowed for saving

--- a/content/common/savable_url_schemes.cc
+++ b/content/common/savable_url_schemes.cc
@@ -13,6 +13,10 @@ namespace content {
 namespace {
 
 const char* const kDefaultSavableSchemes[] = {
+  url::kAboutScheme,
+  url::kBlobScheme,
+  url::kContentScheme,
+  url::kGopherScheme,
   url::kHttpScheme,
   url::kHttpsScheme,
   url::kFileScheme,
@@ -21,6 +25,17 @@ const char* const kDefaultSavableSchemes[] = {
   kChromeDevToolsScheme,
   kChromeUIScheme,
   url::kDataScheme,
+  url::kJavaScriptScheme,
+  url::kMailToScheme,
+  url::kWsScheme,
+  url::kWssScheme,
+  kGuestScheme,
+  kMetadataScheme,
+  kSwappedOutScheme,
+  kViewSourceScheme,
+#if defined(OS_CHROMEOS)
+  kExternalFileScheme,
+#endif
   NULL
 };
 
