# Disable domain reliability component

--- a/components/domain_reliability/baked_in_configs.gypi
+++ b/components/domain_reliability/baked_in_configs.gypi
@@ -5,23 +5,6 @@
 {
   'variables': {
     'baked_in_configs': [
-      'domain_reliability/baked_in_configs/c_android_clients_google_com.json',
-      'domain_reliability/baked_in_configs/c_bigcache_googleapis_com.json',
-      'domain_reliability/baked_in_configs/c_doc-0-0-sj_sj_googleusercontent_com.json',
-      'domain_reliability/baked_in_configs/c_docs_google_com.json',
-      'domain_reliability/baked_in_configs/c_drive_google_com.json',
-      'domain_reliability/baked_in_configs/c_googlesyndication_com.json',
-      'domain_reliability/baked_in_configs/c_pack_google_com.json',
-      'domain_reliability/baked_in_configs/c_play_google_com.json',
-      'domain_reliability/baked_in_configs/c_youtube_com.json',
-      'domain_reliability/baked_in_configs/clients2_google_com.json',
-      'domain_reliability/baked_in_configs/docs_google_com.json',
-      'domain_reliability/baked_in_configs/google-analytics_com.json',
-      'domain_reliability/baked_in_configs/googlevideo_com.json',
-      'domain_reliability/baked_in_configs/gvt1_com.json',
-      'domain_reliability/baked_in_configs/gvt2_com.json',
-      'domain_reliability/baked_in_configs/ssl_gstatic_com.json',
-      'domain_reliability/baked_in_configs/www_google_com.json',
     ],
   },
 }
--- a/components/domain_reliability/google_configs.cc
+++ b/components/domain_reliability/google_configs.cc
@@ -28,7 +28,7 @@ struct GoogleConfigParams {
   // www subdomain as the origin for the collector so it matches the config.
   bool duplicate_for_www;
 };
-
+#if 0
 const GoogleConfigParams kGoogleConfigs[] = {
   // Origins with subdomains and same-origin collectors.
   { "google.ac", true, true, true },
@@ -499,7 +499,7 @@ const char* kGoogleStandardCollectors[] = {
 
 const char* kGoogleOriginSpecificCollectorPathString =
   "/domainreliability/upload";
-
+#endif
 static scoped_ptr<DomainReliabilityConfig>
 CreateGoogleConfig(const GoogleConfigParams& params, bool is_www) {
   if (is_www)
@@ -514,12 +514,9 @@ CreateGoogleConfig(const GoogleConfigParams& params, bool is_www) {
   config->collectors.clear();
   if (params.include_origin_specific_collector) {
     GURL::Replacements replacements;
-    replacements.SetPathStr(kGoogleOriginSpecificCollectorPathString);
     config->collectors.push_back(
         new GURL(config->origin.ReplaceComponents(replacements)));
   }
-  for (size_t i = 0; i < arraysize(kGoogleStandardCollectors); i++)
-    config->collectors.push_back(new GURL(kGoogleStandardCollectors[i]));
   config->success_sample_rate = 0.05;
   config->failure_sample_rate = 1.00;
   config->path_prefixes.clear();
@@ -532,12 +529,6 @@ CreateGoogleConfig(const GoogleConfigParams& params, bool is_www) {
 void GetAllGoogleConfigs(
     std::vector<DomainReliabilityConfig*>* configs_out) {
   configs_out->clear();
-
-  for (auto& params : kGoogleConfigs) {
-    configs_out->push_back(CreateGoogleConfig(params, false).release());
-    if (params.duplicate_for_www)
-      configs_out->push_back(CreateGoogleConfig(params, true).release());
-  }
 }
 
 }  // namespace domain_reliability
--- a/components/domain_reliability/uploader.cc
+++ b/components/domain_reliability/uploader.cc
@@ -75,7 +75,7 @@ class DomainReliabilityUploaderImpl
     VLOG(1) << "Uploading report to " << upload_url;
     VLOG(2) << "Report JSON: " << report_json;
 
-    if (discard_uploads_) {
+    if (true) {
       VLOG(1) << "Discarding report instead of uploading.";
       UploadResult result;
       result.status = UploadResult::SUCCESS;
