# Disabe initialization of the Google copresence manager

--- a/components/copresence/copresence_manager_impl.cc
+++ b/components/copresence/copresence_manager_impl.cc
@@ -69,29 +69,7 @@ CopresenceManagerImpl::CopresenceManagerImpl(CopresenceDelegate* delegate)
           kMaxQueuedMessages) {
   DCHECK(delegate_);
   DCHECK(delegate_->GetWhispernetClient());
-  // TODO(ckehoe): Handle whispernet initialization in the whispernet component.
-  delegate_->GetWhispernetClient()->Initialize(
-      whispernet_init_callback_.callback());
-
-  MessagesCallback messages_callback = base::Bind(
-      &CopresenceManagerImpl::DispatchMessages,
-      // This will only be passed to objects that we own.
-      base::Unretained(this));
-
-  if (delegate->GetGCMDriver())
-    gcm_handler_.reset(new GCMHandlerImpl(delegate->GetGCMDriver(),
-                                          directive_handler_.get(),
-                                          messages_callback));
-
-  rpc_handler_.reset(new RpcHandler(delegate,
-                                    directive_handler_.get(),
-                                    state_.get(),
-                                    gcm_handler_.get(),
-                                    messages_callback));
-
-  directive_handler_->Start(delegate_->GetWhispernetClient(),
-                            base::Bind(&CopresenceManagerImpl::ReceivedTokens,
-                                       base::Unretained(this)));
+  whispernet_init_callback_.Cancel();
 }
 
 CopresenceManagerImpl::~CopresenceManagerImpl() {
