# Fix building without one click signin

--- a/chrome/chrome_browser_ui.gypi
+++ b/chrome/chrome_browser_ui.gypi
@@ -3347,6 +3347,12 @@
           ],
           'sources': [ '<@(chrome_browser_ui_media_router_sources)' ],
         }],
+        ['enable_one_click_signin==0', {
+          'sources!': [
+            'browser/ui/views/sync/one_click_signin_bubble_view.cc',
+            'browser/ui/views/sync/one_click_signin_bubble_view.h',
+          ],
+        }],
         ['enable_supervised_users==0', {
           'sources!': [
             'browser/ui/webui/options/supervised_user_create_confirm_handler.cc',
--- a/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
+++ b/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
@@ -348,6 +348,7 @@ void OneClickSigninSyncStarter::CancelSigninAndDelete() {
 }
 
 void OneClickSigninSyncStarter::ConfirmAndSignin() {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   SigninManager* signin = SigninManagerFactory::GetForProfile(profile_);
   if (confirmation_required_ == CONFIRM_UNTRUSTED_SIGNIN) {
     browser_ = EnsureBrowser(browser_, profile_);
@@ -365,6 +366,7 @@ void OneClickSigninSyncStarter::ConfirmAndSignin() {
     // No confirmation required - just sign in the user.
     signin->CompletePendingSignin();
   }
+#endif
 }
 
 void OneClickSigninSyncStarter::UntrustedSigninConfirmed(
--- a/chrome/browser/ui/sync/one_click_signin_bubble_links_delegate.cc
+++ b/chrome/browser/ui/sync/one_click_signin_bubble_links_delegate.cc
@@ -17,13 +17,17 @@ OneClickSigninBubbleLinksDelegate::~OneClickSigninBubbleLinksDelegate() {}
 
 void OneClickSigninBubbleLinksDelegate::OnLearnMoreLinkClicked(
     bool is_dialog) {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   chrome::NavigateParams params(browser_,
                                 GURL(chrome::kChromeSyncLearnMoreURL),
                                 ui::PAGE_TRANSITION_LINK);
   params.disposition = is_dialog ? NEW_WINDOW : NEW_FOREGROUND_TAB;
   chrome::Navigate(&params);
+#endif
 }
 
 void OneClickSigninBubbleLinksDelegate::OnAdvancedLinkClicked() {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   chrome::ShowSettingsSubPage(browser_, chrome::kSyncSetupSubPage);
+#endif
 }
