--- a/net/http/transport_security_state.cc
+++ b/net/http/transport_security_state.cc
@@ -732,7 +732,7 @@
     const X509Certificate* validated_certificate_chain,
     const HashValueVector& public_key_hashes) {
   using CTRequirementLevel = RequireCTDelegate::CTRequirementLevel;
-
+  return false;
   CTRequirementLevel ct_required = CTRequirementLevel::DEFAULT;
   if (require_ct_delegate_)
     ct_required = require_ct_delegate_->IsCTRequiredForHost(hostname);
