# Fix building with iridium::trknotify

--- /dev/null
+++ b/iridium/BUILD.gn
@@ -0,0 +1,13 @@
+source_set("trknotify") {
+    sources = [
+        "trknotify.cpp",
+        "trknotify.h",
+        "trkbar.cpp",
+        "trkbar.h",
+    ]
+    include_dirs = [
+        "..",
+        "../third_party/skia/include/config",
+        "../third_party/skia/include/core",
+    ]
+}
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -283,6 +283,7 @@
 
     deps = [
       ":browser_dependencies",
+      "//iridium:trknotify",
       "//build/config/sanitizers:deps",
       "//chrome/common:features",
       "//third_party/cld_2",
