# Modify other miscellaneous hardcoded settings

--- a/build/config/features.gni
+++ b/build/config/features.gni
@@ -120,7 +120,6 @@ declare_args() {
   enable_webvr = is_android && is_component_build && is_debug &&
                  (current_cpu == "x86" || current_cpu == "x64" ||
                   current_cpu == "arm" || current_cpu == "arm64")
-}
 
 # Additional dependent variables -----------------------------------------------
 
@@ -181,3 +180,4 @@ enable_mac_keystone = is_mac && is_chrome_branded && is_official_build
 # =============================================
 #
 # See comment at the top.
+}
