From 154fa82c50f966cb6473fc0f188d5f52406221b9 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Sun, 20 Dec 2015 11:12:00 +0100
Subject: [PATCH 53/55] build: use -Wl,--no-keep-memory

---
 chrome/chrome_exe.gypi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/chrome/chrome_exe.gypi b/chrome/chrome_exe.gypi
index 3d4701e..5c402bb 100644
--- a/chrome/chrome_exe.gypi
+++ b/chrome/chrome_exe.gypi
@@ -76,6 +76,14 @@
         'app/main_dll_loader_win.cc',
         'app/main_dll_loader_win.h',
       ],
+      # Hey, I know what would be an awesome idea.
+      # Let us statically compile EVERYTHING into one giant
+      # freaking executable. Too bad this means we can't build
+      # natively on an IA32 machine. If only someone would
+      # invent shared library objects. Idiots.
+      'ldflags': [
+        '-Wl,--no-keep-memory'
+      ],
       'mac_bundle_resources': [
         'app/app-Info.plist',
       ],
-- 
2.6.6

