--- a/chrome/browser/ungoogled_flag_choices.h
+++ b/chrome/browser/ungoogled_flag_choices.h
@@ -91,4 +91,18 @@ const FeatureEntry::Choice kReferrerDire
      "referrer-directive",
      "noreferrers"},
 };
+const FeatureEntry::Choice kWebRTCIPPolicy[] = {
+    {"Disable non proxied udp",
+     "webrtc-ip-handling-policy",
+     "disable_non_proxied_udp"},
+    {"Default",
+     "webrtc-ip-handling-policy",
+     "default"},
+    {"Default public and private interfaces",
+     "webrtc-ip-handling-policy",
+     "default_public_and_private_interfaces"},
+    {"Default public interface only",
+     "webrtc-ip-handling-policy",
+     "default_public_interface_only"},
+};
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_CHOICES_H_
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -124,4 +124,8 @@
      "Enable themes in Incognito mode",
      "Allows themes to override Google's built-in Incognito theming. ungoogled-chromium flag.",
      kOsDesktop, SINGLE_VALUE_TYPE("enable-incognito-themes")},
+    {"webrtc-ip-handling-policy",
+     "WebRTC IP policy",
+     "Restrict which IP addresses and interfaces WebRTC uses. Chromium feature, ungoogled-chromium flag.",
+     kOsAll, MULTI_VALUE_TYPE(kWebRTCIPPolicy)},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
