--- a/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
+++ b/chrome/browser/ui/views/tabs/vertical/vertical_tab_view.cc
@@ -7,6 +7,7 @@
 #include <optional>
 #include <string>
 
+#include "base/command_line.h"
 #include "base/functional/callback_helpers.h"
 #include "base/metrics/histogram_functions.h"
 #include "base/metrics/user_metrics.h"
@@ -732,6 +733,10 @@ bool VerticalTabView::IsChildVisible(const views::View* child_view,
       return false;
     }
 
+    if (base::CommandLine::ForCurrentProcess()->HasSwitch("hide-tab-close-buttons")) {
+      return false;
+    }
+
     // When uncollapsing the tabstrip, intentionally start showing the close
     // button on active non-hovered tabs a little bit sooner than reaching the
     // uncollapsed min width, because otherwise the close buttons in a grouped
