--- a/chrome/browser/ui/views/tabs/tab_strip.cc
+++ b/chrome/browser/ui/views/tabs/tab_strip.cc
@@ -1909,6 +1909,8 @@ bool TabStrip::SupportsMultipleSelection
 }
 
 bool TabStrip::ShouldHideCloseButtonForTab(Tab* tab) const {
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("hide-tab-close-buttons"))
+    return true;
   if (tab->IsActive())
     return false;
   return !!touch_layout_;
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -92,4 +92,8 @@
      "Tab Hover Cards",
      "Allows removing the tab hover cards or using a tooltip as a replacement.  ungoogled-chromium flag.",
      kOsDesktop, MULTI_VALUE_TYPE(kTabHoverCards)},
+    {"hide-tab-close-buttons",
+     "Hide tab close buttons",
+     "Hides the close buttons on tabs.  ungoogled-chromium flag.",
+     kOsDesktop, SINGLE_VALUE_TYPE("hide-tab-close-buttons")},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
