--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -314,7 +314,8 @@ void ToolbarView::Init() {
 
   std::unique_ptr<SidePanelToolbarButton> side_panel_button;
   std::unique_ptr<SidePanelToolbarContainer> side_panel_toolbar_container;
-  if (browser_view_->unified_side_panel()) {
+  if (browser_view_->unified_side_panel() &&
+      !base::CommandLine::ForCurrentProcess()->HasSwitch("hide-sidepanel-button")) {
     if (base::FeatureList::IsEnabled(
             companion::features::kSidePanelCompanion)) {
       side_panel_toolbar_container =
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -100,4 +100,8 @@
      "Disable Sharing Hub",
      "Disables the sharing hub button. ungoogled-chromium flag.",
      kOsDesktop, SINGLE_VALUE_TYPE("disable-sharing-hub")},
+    {"hide-sidepanel-button",
+     "Hide SidePanel Button",
+     "Hides the SidePanel Button. ungoogled-chromium flag.",
+     kOsDesktop, SINGLE_VALUE_TYPE("hide-sidepanel-button")},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
