--- a/chrome/browser/ui/browser_tab_strip_model_delegate.cc
+++ b/chrome/browser/ui/browser_tab_strip_model_delegate.cc
@@ -465,6 +465,12 @@ void BrowserTabStripModelDelegate::Close
     }
   }
 
+  if (model->count() == 1 && IsNormalWindow() &&
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "close-window-with-last-tab") == "never") {
+    AddTabAt(GURL(), -1, true, std::nullopt, false);
+  }
+
   auto [cb1, cb2] = base::SplitOnceCallback(std::move(on_approved));
 
   // Only consider pausing the close operation if this is the last remaining
--- a/chrome/browser/ui/tabs/tab_strip_model.cc
+++ b/chrome/browser/ui/tabs/tab_strip_model.cc
@@ -19,6 +19,7 @@
 
 #include "base/check.h"
 #include "base/check_op.h"
+#include "base/command_line.h"
 #include "base/containers/flat_map.h"
 #include "base/containers/span.h"
 #include "base/dcheck_is_on.h"
@@ -3280,9 +3281,17 @@ void TabStripModel::ExecuteCloseTabs(
     base::RepeatingCallback<std::vector<tabs::TabInterface*>()>
         get_tabs_to_close,
     uint32_t close_types) {
-  ReentrancyCheck reentrancy_check(&reentrancy_guard_);
+  ReentrancyCheck::ValidateNotReentrant(&reentrancy_guard_);
   const std::vector<tabs::TabInterface*> tabs_to_close =
       std::move(get_tabs_to_close).Run();
+  if (!tabs_to_close.empty() && !closing_all_ &&
+      static_cast<int>(tabs_to_close.size()) == count() &&
+      delegate()->IsNormalWindow() &&
+      base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+          "close-window-with-last-tab") == "never") {
+    delegate()->AddTabAt(GURL(), -1, true);
+  }
+  ReentrancyCheck reentrancy_check(&reentrancy_guard_);
 
   CreateHistoricalSplitIfClosing(tabs_to_close, close_types);
 
--- a/chrome/browser/ungoogled_flag_choices.h
+++ b/chrome/browser/ungoogled_flag_choices.h
@@ -55,4 +55,10 @@ const FeatureEntry::Choice kOmniboxAutoc
      "omnibox-autocomplete-filtering",
      "search-bookmarks-chrome"},
 };
+const FeatureEntry::Choice kCloseWindowWithLastTab[] = {
+    {flags_ui::kGenericExperimentChoiceDefault, "", ""},
+    {"Never",
+     "close-window-with-last-tab",
+     "never"},
+};
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_CHOICES_H_
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -44,4 +44,8 @@
      "Omnibox Autocomplete Filtering",
      "Restrict omnibox autocomplete results to a combination of search suggestions (if enabled), bookmarks, and internal chrome pages. ungoogled-chromium flag.",
      kOsAll, MULTI_VALUE_TYPE(kOmniboxAutocompleteFiltering)},
+    {"close-window-with-last-tab",
+     "Close window with last tab",
+     "Determines whether a window should close once the last tab is closed. ungoogled-chromium flag.",
+     kOsDesktop, MULTI_VALUE_TYPE(kCloseWindowWithLastTab)},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
