--- a/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
+++ b/chrome/browser/ui/views/fullscreen_control/fullscreen_control_host.cc
@@ -69,6 +69,10 @@ bool IsExitUiEnabled() {
   // menu and controls reveal when the cursor is moved to the top.
   return false;
 #else
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          "hide-fullscreen-exit-ui"))
+    return false;
+
   // Kiosk mode is a fullscreen experience, which makes the exit UI
   // inappropriate.
   return !chrome::IsRunningInAppMode();
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -116,4 +116,8 @@
      "Hide Extensions Menu",
      "Hides the extensions menu (the puzzle piece icon). ungoogled-chromium flag.",
      kOsDesktop, SINGLE_VALUE_TYPE("hide-extensions-menu")},
+    {"hide-fullscreen-exit-ui",
+     "Hide Fullscreen Exit UI",
+     "Hides the \"X\" that appears when the mouse cursor is moved towards the top of the window in fullscreen mode. ungoogled-chromium flag.",
+     kOsLinux | kOsWindows, SINGLE_VALUE_TYPE("hide-fullscreen-exit-ui")},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
