--- a/chrome/browser/sharing/features.cc
+++ b/chrome/browser/sharing/features.cc
@@ -26,3 +26,5 @@ const base::Feature kSharingSendViaSync{
 
 const base::Feature kSharingPreferVapid {
   "SharingPreferVapid", base::FEATURE_DISABLED_BY_DEFAULT};
+
+const base::Feature kDisableQRGenerator{"DisableQRGenerator", base::FEATURE_DISABLED_BY_DEFAULT};
--- a/chrome/browser/sharing/features.h
+++ b/chrome/browser/sharing/features.h
@@ -34,4 +34,5 @@ extern const base::Feature kSharingSendV
 // Feature flag for prefer sending sharing message using VAPID.
 extern const base::Feature kSharingPreferVapid;
 
+extern const base::Feature kDisableQRGenerator;
 #endif  // CHROME_BROWSER_SHARING_FEATURES_H_
--- a/chrome/browser/ui/qrcode_generator/qrcode_generator_bubble_controller.cc
+++ b/chrome/browser/ui/qrcode_generator/qrcode_generator_bubble_controller.cc
@@ -23,7 +23,7 @@ QRCodeGeneratorBubbleController::~QRCode
 
 // static
 bool QRCodeGeneratorBubbleController::IsGeneratorAvailable(const GURL& url) {
-  if (!url.SchemeIsHTTPOrHTTPS())
+  if (!url.SchemeIsHTTPOrHTTPS() || base::FeatureList::IsEnabled(kDisableQRGenerator))
     return false;
 
   return true;
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -305,7 +305,8 @@ void LocationBarView::Init() {
     // the left most icon.
     params.types_enabled.push_back(PageActionIconType::kSendTabToSelf);
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
-    params.types_enabled.push_back(PageActionIconType::kQRCodeGenerator);
+    if (!base::FeatureList::IsEnabled(kDisableQRGenerator))
+      params.types_enabled.push_back(PageActionIconType::kQRCodeGenerator);
     if (base::FeatureList::IsEnabled(kSharedClipboardUI))
       params.types_enabled.push_back(PageActionIconType::kSharedClipboard);
     if (base::FeatureList::IsEnabled(kWebOTPCrossDevice))
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -64,4 +64,8 @@
      "Remove Tabsearch Button",
      "Removes the tabsearch button from the tabstrip.  ungoogled-chromium flag",
      kOsDesktop, SINGLE_VALUE_TYPE("remove-tabsearch-button")},
+    {"disable-qr-generator",
+     "Disable QR Generator",
+     "Disables the QR generator for sharing page links.  ungoogled-chromium flag",
+     kOsDesktop, FEATURE_VALUE_TYPE(kDisableQRGenerator)},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
