--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -255,6 +255,7 @@ void ToolbarView::Init() {
   // Do not create the extensions or browser actions container if it is a guest
   // profile (only regular and incognito profiles host extensions).
   if (!browser_->profile()->IsGuestSession()) {
+    if (!base::CommandLine::ForCurrentProcess()->HasSwitch("hide-extensions-menu"))
     extensions_container =
         std::make_unique<ExtensionsToolbarContainer>(browser_);
   }
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -108,4 +108,8 @@
      "Hide SidePanel Button",
      "Hides the SidePanel Button.  ungoogled-chromium flag.",
      kOsDesktop, SINGLE_VALUE_TYPE("hide-sidepanel-button")},
+    {"hide-extensions-menu",
+     "Hide Extensions Menu",
+     "Hides the extensions menu (the puzzle piece icon) that allows the user to control extensions site access.  ungoogled-chromium flag.",
+     kOsDesktop, SINGLE_VALUE_TYPE("hide-extensions-menu")},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
