--- a/chrome/browser/sharing/features.cc
+++ b/chrome/browser/sharing/features.cc
@@ -17,3 +17,5 @@ const base::FeatureParam<int> kSharingPu
 
 const base::FeatureParam<int> kSharingPulseDeltaAndroidHours = {
     &kSharingMatchPulseInterval, "SharingPulseDeltaAndroidHours", 24};
+
+BASE_FEATURE(kDisableQRGenerator, "DisableQRGenerator", base::FEATURE_DISABLED_BY_DEFAULT);
--- a/chrome/browser/sharing/features.h
+++ b/chrome/browser/sharing/features.h
@@ -26,4 +26,5 @@ extern const base::FeatureParam<int> kSh
 // backgrounded. Such devices cannot be marked stale aggressively.
 extern const base::FeatureParam<int> kSharingPulseDeltaAndroidHours;
 
+BASE_DECLARE_FEATURE(kDisableQRGenerator);
 #endif  // CHROME_BROWSER_SHARING_FEATURES_H_
--- a/chrome/browser/ui/qrcode_generator/qrcode_generator_bubble_controller.cc
+++ b/chrome/browser/ui/qrcode_generator/qrcode_generator_bubble_controller.cc
@@ -23,7 +23,7 @@ QRCodeGeneratorBubbleController::~QRCode
 
 // static
 bool QRCodeGeneratorBubbleController::IsGeneratorAvailable(const GURL& url) {
-  if (!url.SchemeIsHTTPOrHTTPS())
+  if (!url.SchemeIsHTTPOrHTTPS() || base::FeatureList::IsEnabled(kDisableQRGenerator))
     return false;
 
   return true;
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -64,4 +64,8 @@
      "Remove Tabsearch Button",
      "Removes the tabsearch button from the tabstrip. ungoogled-chromium flag",
      kOsDesktop, SINGLE_VALUE_TYPE("remove-tabsearch-button")},
+    {"disable-qr-generator",
+     "Disable QR Generator",
+     "Disables the QR generator for sharing page links. ungoogled-chromium flag",
+     kOsDesktop, FEATURE_VALUE_TYPE(kDisableQRGenerator)},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
