--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -915,6 +915,7 @@ BrowserView::BrowserView(std::unique_ptr
         AddChildView(std::make_unique<ContentsSeparator>());
     side_panel_coordinator_ = std::make_unique<SidePanelCoordinator>(this);
   } else {
+   if (!base::CommandLine::ForCurrentProcess()->HasSwitch("hide-sidepanel-button"))
     right_aligned_side_panel_ = AddChildView(std::make_unique<SidePanel>(this));
   }
 
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -104,4 +104,8 @@
      "Disable Sharing Hub",
      "Disables the sharing hub button. ungoogled-chromium flag.",
      kOsDesktop, SINGLE_VALUE_TYPE("disable-sharing-hub")},
+    {"hide-sidepanel-button",
+     "Hide SidePanel Button",
+     "Hides the SidePanel Button. ungoogled-chromium flag.",
+     kOsDesktop, SINGLE_VALUE_TYPE("hide-sidepanel-button")},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
