--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -4971,6 +4971,10 @@ static_library("browser") {
       # ... get modularized.
       "//chrome/browser/ui/extensions:impl",
     ]
+  } else if (!is_apple) {
+    # This is already handled above in the CfT config and isn't needed on
+    # Apple's platforms (which use a platform-specific hyphenation engine).
+    deps += [ "//third_party/hyphenation-patterns:bundle_hyphen_data" ]
   }
 
   # TODO(https://crbug.com/356905053): Add more dependencies here and merge this
