--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -8376,6 +8376,10 @@ static_library("browser") {
 
   if (is_chrome_for_testing) {
     deps += [ "//chrome/browser/chrome_for_testing" ]
+  } else if (!is_apple) {
+    # This is already handled above in the CfT config and isn't needed on
+    # Apple's platforms (which use a platform-specific hyphenation engine).
+    deps += [ "//third_party/hyphenation-patterns:bundle_hyphen_data" ]
   }
 
   if (enable_builtin_search_provider_assets) {
