--- a/components/optimization_guide/core/optimization_guide_features.cc
+++ b/components/optimization_guide/core/optimization_guide_features.cc
@@ -73,7 +73,7 @@ const base::Feature kOptimizationHintsFi
 
 // Enables fetching from a remote Optimization Guide Service.
 const base::Feature kRemoteOptimizationGuideFetching{
-    "OptimizationHintsFetching", base::FEATURE_ENABLED_BY_DEFAULT};
+    "OptimizationHintsFetching", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kRemoteOptimizationGuideFetchingAnonymousDataConsent {
   "OptimizationHintsFetchingAnonymousDataConsent",
