# Disable Federated Credential Management (FedCM) by default

--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -431,7 +431,7 @@ BASE_FEATURE(kEnsureExistingRendererAliv
 // by the flag in RuntimeEnabledFeatures on the blink side. See also
 // the use of kSetOnlyIfOverridden in content/child/runtime_features.cc.
 // We enable it here by default to support use in origin trials.
-BASE_FEATURE(kFedCm, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kFedCm, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Kill switch for checking if there is an ongoing embedder task in the auto
 // re-authn flow.
