# Enable chrome://flags/#prefetch-privacy-changes by default

--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -242,7 +242,7 @@ const base::Feature kWebComponentsV0Enab
 // Prefetch request properties are updated to be privacy-preserving. See
 // crbug.com/988956.
 const base::Feature kPrefetchPrivacyChanges{"PrefetchPrivacyChanges",
-                                            base::FEATURE_DISABLED_BY_DEFAULT};
+                                            base::FEATURE_ENABLED_BY_DEFAULT};
 
 const char kMixedContentAutoupgradeModeParamName[] = "mode";
 const char kMixedContentAutoupgradeModeAllPassive[] = "all-passive";
