--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -4528,8 +4528,7 @@ const FeatureEntry kFeatureEntries[] = {
      FEATURE_VALUE_TYPE(blink::features::kDecodeLossyWebPImagesToYUV)},
 
     {"dns-over-https", flag_descriptions::kDnsOverHttpsName,
-     flag_descriptions::kDnsOverHttpsDescription,
-     kOsMac | kOsWin | kOsCrOS | kOsAndroid,
+     flag_descriptions::kDnsOverHttpsDescription, kOsAll,
      FEATURE_VALUE_TYPE(features::kDnsOverHttps)},
 
 #if defined(OS_ANDROID)
