# Enable chrome://flags/#prefetch-privacy-changes by default

--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -491,7 +491,7 @@ const base::Feature kComputePressure{"Co
 // Prefetch request properties are updated to be privacy-preserving. See
 // crbug.com/988956.
 const base::Feature kPrefetchPrivacyChanges{"PrefetchPrivacyChanges",
-                                            base::FEATURE_DISABLED_BY_DEFAULT};
+                                            base::FEATURE_ENABLED_BY_DEFAULT};
 
 // Decodes jpeg 4:2:0 formatted images to YUV instead of RGBX and stores in this
 // format in the image decode cache. See crbug.com/919627 for details on the
