# Enable chrome://flags/#prefetch-privacy-changes by default

--- a/third_party/blink/common/features.cc
+++ b/third_party/blink/common/features.cc
@@ -2060,7 +2060,7 @@ BASE_FEATURE(kPrefetchFontLookupTables,
 // crbug.com/988956.
 BASE_FEATURE(kPrefetchPrivacyChanges,
              "PrefetchPrivacyChanges",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kPreloadingHeuristicsMLModel,
              "PreloadingHeuristicsMLModel",
