--- a/chrome/browser/browser_features.cc
+++ b/chrome/browser/browser_features.cc
@@ -50,7 +50,7 @@ BASE_FEATURE(kBookmarkTriggerForPrerender2,
 // switch.
 BASE_FEATURE(kCertificateTransparencyAskBeforeEnabling,
              "CertificateTransparencyAskBeforeEnabling",
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
+#if true
              base::FEATURE_ENABLED_BY_DEFAULT);
 #else
              base::FEATURE_DISABLED_BY_DEFAULT);
--- a/chrome/browser/ungoogled_flag_entries.h
+++ b/chrome/browser/ungoogled_flag_entries.h
@@ -140,4 +140,8 @@
      "Spoof WebGL Info",
      "Return generic values for WebGLDebugRendererInfo to remove a potential data leak while preventing potential website breakage. ungoogled-chromium flag.",
      kOsAll, FEATURE_VALUE_TYPE(blink::features::kSpoofWebGLInfo)},
+    {"enforce-certificate-transparency",
+     "Enforce Certificate Transparency",
+     "Enforce Certificate Transparency for certificates that sites present. This is enabled by default. ungoogled-chromium flag.",
+     kOsAll, FEATURE_VALUE_TYPE(features::kCertificateTransparencyAskBeforeEnabling)},
 #endif  // CHROME_BROWSER_UNGOOGLED_FLAG_ENTRIES_H_
