# Enable chrome://flags/#reduced-referrer-granularity by default

--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -464,7 +464,7 @@ const base::Feature kProactivelySwapBrowsingInstance{
 // Reduce the amount of information in the default 'referer' header for
 // cross-origin requests.
 const base::Feature kReducedReferrerGranularity{
-    "ReducedReferrerGranularity", base::FEATURE_DISABLED_BY_DEFAULT};
+    "ReducedReferrerGranularity", base::FEATURE_ENABLED_BY_DEFAULT};
 
 // Controls whether FileURLLoaderFactory can fetch additional files based on the
 // isolated world's origin. This feature is disabled by default because we want
