# Make popups go to tabs instead

--- a/content/renderer/render_view_impl.cc
+++ b/content/renderer/render_view_impl.cc
@@ -82,7 +82,7 @@ WindowOpenDisposition RenderViewImpl::Na
     case blink::kWebNavigationPolicyNewWindow:
       return WindowOpenDisposition::NEW_WINDOW;
     case blink::kWebNavigationPolicyNewPopup:
-      return WindowOpenDisposition::NEW_POPUP;
+      return WindowOpenDisposition::NEW_FOREGROUND_TAB;
     default:
       NOTREACHED() << "Unexpected WebNavigationPolicy";
       return WindowOpenDisposition::IGNORE_ACTION;
--- a/ui/base/mojom/window_open_disposition_mojom_traits.h
+++ b/ui/base/mojom/window_open_disposition_mojom_traits.h
@@ -28,7 +28,7 @@ struct EnumTraits<ui::mojom::WindowOpenD
       case WindowOpenDisposition::NEW_BACKGROUND_TAB:
         return ui::mojom::WindowOpenDisposition::NEW_BACKGROUND_TAB;
       case WindowOpenDisposition::NEW_POPUP:
-        return ui::mojom::WindowOpenDisposition::NEW_POPUP;
+        return ui::mojom::WindowOpenDisposition::NEW_FOREGROUND_TAB;
       case WindowOpenDisposition::NEW_WINDOW:
         return ui::mojom::WindowOpenDisposition::NEW_WINDOW;
       case WindowOpenDisposition::SAVE_TO_DISK:
@@ -62,7 +62,7 @@ struct EnumTraits<ui::mojom::WindowOpenD
         *out = WindowOpenDisposition::NEW_BACKGROUND_TAB;
         return true;
       case ui::mojom::WindowOpenDisposition::NEW_POPUP:
-        *out = WindowOpenDisposition::NEW_POPUP;
+        *out = WindowOpenDisposition::NEW_FOREGROUND_TAB;
         return true;
       case ui::mojom::WindowOpenDisposition::NEW_WINDOW:
         *out = WindowOpenDisposition::NEW_WINDOW;
