--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1502,9 +1502,6 @@ config("default_warnings") {
           # Ignore warnings about MSVC optimization pragmas.
           # TODO(thakis): Only for no_chromium_code? http://crbug.com/505314
           "-Wno-ignored-pragma-optimize",
-
-          # TODO(hans): https://crbug.com/890307
-          "-Wno-defaulted-function-deleted",
         ]
       }
     }
