description: fix path to speechd headers
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/linux/system.gyp
+++ b/build/linux/system.gyp
@@ -1141,7 +1141,7 @@
             # speech-dispatcher < 0.8 we can get rid of all this (including
             # third_party/speech-dispatcher) and just include
             # speech-dispatcher/libspeechd.h unconditionally.
-            'libspeechd_h_prefix%': '',
+            'libspeechd_h_prefix%': 'speech-dispatcher/',
           },
           'action_name': 'generate_libspeechd_loader',
           'inputs': [
