description: changes required to support ffmpeg 3.2 in stretch
author: Michael Gilbert <mgilbert@debian.org>

--- a/media/ffmpeg/ffmpeg_common.cc
+++ b/media/ffmpeg/ffmpeg_common.cc
@@ -189,8 +189,6 @@ static VideoCodec CodecIDToVideoCodec(AV
       return kCodecVP8;
     case AV_CODEC_ID_VP9:
       return kCodecVP9;
-    case AV_CODEC_ID_AV1:
-      return kCodecAV1;
     default:
       DVLOG(1) << "Unknown video CodecID: " << codec_id;
   }
@@ -213,8 +211,6 @@ AVCodecID VideoCodecToCodecID(VideoCodec
       return AV_CODEC_ID_VP8;
     case kCodecVP9:
       return AV_CODEC_ID_VP9;
-    case kCodecAV1:
-      return AV_CODEC_ID_AV1;
     default:
       DVLOG(1) << "Unknown VideoCodec: " << video_codec;
   }
