# Removes the -Wno-ignored-pragma-optimize flag

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1486,13 +1486,6 @@ config("default_warnings") {
       }
 
       if (current_toolchain == host_toolchain || !use_xcode_clang) {
-        # Flags NaCl (Clang 3.7) and Xcode 9.2 (Clang clang-900.0.39.2) do not
-        # recognize.
-        cflags += [
-          # Ignore warnings about MSVC optimization pragmas.
-          # TODO(thakis): Only for no_chromium_code? http://crbug.com/505314
-          "-Wno-ignored-pragma-optimize",
-        ]
         if (is_fuchsia) {
           cflags += [
             # TODO(hans): https://crbug.com/890307
