# Fix GN safe_browsing on macOS

--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1734,7 +1734,6 @@ jumbo_split_static_library("browser") {
     "//chrome/browser/profiling_host",
     "//chrome/browser/push_messaging:budget_proto",
     "//chrome/browser/resource_coordinator:mojo_bindings",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/ssl:proto",
     "//chrome/browser/ui",
     "//chrome/browser/ui/webui/bluetooth_internals",
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -775,7 +775,6 @@ jumbo_static_library("extensions") {
     "//chrome/browser/media/router",
     "//chrome/browser/media/router/discovery",
     "//chrome/browser/resource_coordinator:mojo_bindings",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/web_applications/components",
     "//chrome/browser/web_applications/extensions",
     "//chrome/common",
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -384,7 +384,6 @@ jumbo_split_static_library("ui") {
     "//chrome/browser/engagement:mojo_bindings",
     "//chrome/browser/media:mojo_bindings",
     "//chrome/browser/profiling_host",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/ssl:proto",
     "//chrome/browser/ui/webui/app_management:mojo_bindings",
     "//chrome/browser/ui/webui/bluetooth_internals",
--- a/chrome/browser/ui/webui/settings/change_password_handler.cc
+++ b/chrome/browser/ui/webui/settings/change_password_handler.cc
@@ -52,19 +52,19 @@ void ChangePasswordHandler::HandleInitia
 }
 
 void ChangePasswordHandler::HandleChangePassword(const base::ListValue* args) {
-  service_->OnUserAction(web_ui()->GetWebContents(),
-                         safe_browsing::LoginReputationClientRequest::
-                             PasswordReuseEvent::SIGN_IN_PASSWORD,
-                         safe_browsing::WarningUIType::CHROME_SETTINGS,
-                         safe_browsing::WarningAction::CHANGE_PASSWORD);
+//   service_->OnUserAction(web_ui()->GetWebContents(),
+//                          safe_browsing::LoginReputationClientRequest::
+//                              PasswordReuseEvent::SIGN_IN_PASSWORD,
+//                          safe_browsing::WarningUIType::CHROME_SETTINGS,
+//                          safe_browsing::WarningAction::CHANGE_PASSWORD);
 }
 
 void ChangePasswordHandler::UpdateChangePasswordCardVisibility() {
-  FireWebUIListener(
-      "change-password-visibility",
-      base::Value(service_->IsWarningEnabled() &&
-                  safe_browsing::ChromePasswordProtectionService::
-                      ShouldShowChangePasswordSettingUI(profile_)));
+//   FireWebUIListener(
+//       "change-password-visibility",
+//       base::Value(service_->IsWarningEnabled() &&
+//                   safe_browsing::ChromePasswordProtectionService::
+//                       ShouldShowChangePasswordSettingUI(profile_)));
 }
 
 }  // namespace settings
