# Fix GN bootstrap.py script on macOS

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -809,6 +809,10 @@ def write_gn_ninja(path, root_gen_dir, o
         'base/time/time_exploded_posix.cc',
         'base/time/time_mac.cc',
         'base/threading/platform_thread_mac.mm',
+        'base/mac/dispatch_source_mach.cc',
+        'base/mac/mac_util.mm',
+        'base/mac/scoped_nsobject.mm',
+        'base/mac/mac_logging.mm'
     ])
     static_libraries['libevent']['include_dirs'].extend([
         os.path.join(SRC_ROOT, 'base', 'third_party', 'libevent', 'mac')
@@ -822,6 +826,7 @@ def write_gn_ninja(path, root_gen_dir, o
         '-framework', 'CoreFoundation',
         '-framework', 'Foundation',
         '-framework', 'Security',
+        '-framework', 'IOKit',
     ])
 
   if is_win:
