# Fix GN safe_browsing on macOS

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1413,7 +1413,7 @@ config("default_warnings") {
         "-Wno-user-defined-warnings",
 
         # TODO(thakis): https://crbug.com/753973
-        "-Wno-enum-compare-switch",
+        # "-Wno-enum-compare-switch",
 
         # TODO(hans): https://crbug.com/766891
         "-Wno-null-pointer-arithmetic",
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -1626,7 +1626,6 @@ jumbo_split_static_library("browser") {
     "//chrome/browser/metrics/variations:chrome_ui_string_overrider_factory",
     "//chrome/browser/net:probe_message_proto",
     "//chrome/browser/profiling_host",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/ssl:proto",
     "//chrome/browser/ui",
     "//chrome/browser/ui/webui/bluetooth_internals",
--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -803,7 +803,6 @@ static_library("extensions") {
     "//chrome/browser/devtools",
     "//chrome/browser/media/router",
     "//chrome/browser/media/router/discovery",
-    "//chrome/browser/safe_browsing",
     "//chrome/common",
     "//chrome/common/extensions:mojo_bindings",
     "//chrome/common/extensions/api:api_registration",
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -903,7 +903,6 @@ split_static_library("ui") {
     "//chrome/browser/engagement:mojo_bindings",
     "//chrome/browser/media:mojo_bindings",
     "//chrome/browser/profiling_host",
-    "//chrome/browser/safe_browsing",
     "//chrome/browser/ssl:proto",
     "//chrome/browser/ui/webui/bluetooth_internals",
     "//chrome/browser/ui/webui/interventions_internals:mojo_bindings",
