From 73ddab8ed6059afbc91e3e835632b0401426ec3a Mon Sep 17 00:00:00 2001
From: Joachim Bauch <jojo@struktur.de>
Date: Thu, 25 Jun 2015 15:51:59 +0200
Subject: [PATCH 44/52] browser: disable profile auto-import on first run

---
 chrome/browser/chrome_browser_main.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/chrome/browser/chrome_browser_main.cc b/chrome/browser/chrome_browser_main.cc
index 04230f8..5fe3cf1 100644
--- a/chrome/browser/chrome_browser_main.cc
+++ b/chrome/browser/chrome_browser_main.cc
@@ -1822,11 +1822,13 @@ int ChromeBrowserMainParts::PreMainMessageLoopRunImpl() {
   // preferences are registered, since some of the code that the importer
   // touches reads preferences.
   if (first_run::IsChromeFirstRun()) {
+#if 0
     first_run::AutoImport(profile_,
                           master_prefs_->homepage_defined,
                           master_prefs_->do_import_items,
                           master_prefs_->dont_import_items,
                           master_prefs_->import_bookmarks_path);
+#endif
 
     // Note: this can pop the first run consent dialog on linux.
     first_run::DoPostImportTasks(profile_,
-- 
2.6.6

