--- a/components/signin/internal/identity_manager/primary_account_policy_manager_impl.cc
+++ b/components/signin/internal/identity_manager/primary_account_policy_manager_impl.cc
@@ -86,7 +86,7 @@ void PrimaryAccountPolicyManagerImpl::On
 }
 
 bool PrimaryAccountPolicyManagerImpl::IsSigninAllowed() const {
-  return signin_allowed_.GetValue();
+  return false;
 }
 
 void PrimaryAccountPolicyManagerImpl::OnSigninAllowedPrefChanged(
