--- a/tools/gn/gn_main.cc.orig	2015-11-12 11:49:56.260549036 +0000
+++ b/tools/gn/gn_main.cc	2015-11-12 11:50:14.780931485 +0000
@@ -11,13 +11,7 @@
 #include "tools/gn/standard_out.h"
 #include "tools/gn/switches.h"
 
-// Only the GN-generated build makes this header for now.
-// TODO(brettw) consider adding this if we need it in GYP.
-#if defined(GN_BUILD)
-#include "tools/gn/last_commit_position.h"
-#else
 #define LAST_COMMIT_POSITION "UNKNOWN"
-#endif
 
 namespace {
 
--- a/tools/gn/BUILD.gn.orig	2015-11-12 11:50:43.513524769 +0000
+++ b/tools/gn/BUILD.gn	2015-11-12 11:50:57.697817632 +0000
@@ -227,7 +227,6 @@
 
   deps = [
     ":gn_lib",
-    ":last_commit_position",
     "//base",
     "//build/config/sanitizers:deps",
   ]
