diff -upr chromium-48.0.2564.82.orig/third_party/widevine/cdm/stub/widevine_cdm_version.h chromium-48.0.2564.82/third_party/widevine/cdm/stub/widevine_cdm_version.h
--- chromium-48.0.2564.82.orig/third_party/widevine/cdm/stub/widevine_cdm_version.h	2016-01-14 01:05:17.000000000 +0200
+++ chromium-48.0.2564.82/third_party/widevine/cdm/stub/widevine_cdm_version.h	2016-01-21 19:18:51.287978456 +0200
@@ -12,4 +12,6 @@
 
 #define WIDEVINE_CDM_AVAILABLE
 
+#define WIDEVINE_CDM_VERSION_STRING "@WIDEVINE_VERSION@"
+
 #endif  // WIDEVINE_CDM_VERSION_H_
