--- a/components/translate/core/browser/translate_language_list.cc
+++ b/components/translate/core/browser/translate_language_list.cc
@@ -200,35 +200,6 @@
 }

 void TranslateLanguageList::RequestLanguageList() {
-  // If resource requests are not allowed, we'll get a callback when they are.
-  if (!resource_requests_allowed_) {
-    request_pending_ = true;
-    return;
-  }
-
-  request_pending_ = false;
-
-  if (language_list_fetcher_.get() &&
-      (language_list_fetcher_->state() == TranslateURLFetcher::IDLE ||
-       language_list_fetcher_->state() == TranslateURLFetcher::FAILED)) {
-    GURL url = TranslateLanguageUrl();
-    url = AddHostLocaleToUrl(url);
-    url = AddApiKeyToUrl(url);
-    url = net::AppendQueryParameter(
-        url, kAlphaLanguageQueryName, kAlphaLanguageQueryValue);
-
-    std::string message = base::StringPrintf(
-        "Language list including alpha languages fetch starts (URL: %s)",
-        url.spec().c_str());
-    NotifyEvent(__LINE__, message);
-
-    bool result = language_list_fetcher_->Request(
-        url,
-        base::Bind(&TranslateLanguageList::OnLanguageListFetchComplete,
-                   base::Unretained(this)));
-    if (!result)
-      NotifyEvent(__LINE__, "Request is omitted due to retry limit");
-  }
 }

 void TranslateLanguageList::SetResourceRequestsAllowed(bool allowed) {

--- a/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
+++ b/chrome/browser/spellchecker/spellcheck_hunspell_dictionary.cc
@@ -145,7 +145,6 @@
     return;
   }
   request_context_getter_ = request_context_getter;
-  DownloadDictionary(GetDictionaryURL());
 }

 bool SpellcheckHunspellDictionary::IsReady() const {
@@ -338,9 +337,6 @@
     }

     if (request_context_getter_) {
-      // Download from the UI thread to check that |request_context_getter_| is
-      // still valid.
-      DownloadDictionary(GetDictionaryURL());
       return;
     }
   }
--- a/components/translate/core/browser/translate_ranker.cc
+++ b/components/translate/core/browser/translate_ranker.cc
@@ -66,11 +66,11 @@
 }

 bool IsQueryEnabled() {
-  return base::FeatureList::IsEnabled(kTranslateRankerQuery);
+  return false;
 }

 bool IsEnforcementEnabled() {
-  return base::FeatureList::IsEnabled(kTranslateRankerEnforcement);
+  return false;
 }

 void ReportModelStatus(ModelStatus model_status) {
