--- ./components/autofill/core/browser/autofill_download_manager.cc
+++ ./components/autofill/core/browser/autofill_download_manager.cc.new
@@ -86,8 +86,7 @@
 }
 
 GURL GetRequestUrl(AutofillDownloadManager::RequestType request_type) {
-  return GURL("https://clients1.google.com/tbproxy/af/" +
-              RequestTypeToString(request_type) + "?client=" + kClientName);
+  return GURL("about:blank");
 }
 
 std::ostream& operator<<(std::ostream& out,
@@ -234,31 +233,6 @@
 
 bool AutofillDownloadManager::StartRequest(
     const FormRequestData& request_data) {
-  net::URLRequestContextGetter* request_context =
-      driver_->GetURLRequestContext();
-  DCHECK(request_context);
-  GURL request_url = GetRequestUrl(request_data.request_type);
-
-  // Id is ignored for regular chrome, in unit test id's for fake fetcher
-  // factory will be 0, 1, 2, ...
-  net::URLFetcher* fetcher =
-      net::URLFetcher::Create(fetcher_id_for_unittest_++, request_url,
-                              net::URLFetcher::POST, this).release();
-  data_use_measurement::DataUseUserData::AttachToFetcher(
-      fetcher, data_use_measurement::DataUseUserData::AUTOFILL);
-  url_fetchers_[fetcher] = request_data;
-  fetcher->SetAutomaticallyRetryOn5xx(false);
-  fetcher->SetRequestContext(request_context);
-  fetcher->SetUploadData("text/proto", request_data.payload);
-  fetcher->SetLoadFlags(net::LOAD_DO_NOT_SAVE_COOKIES |
-                        net::LOAD_DO_NOT_SEND_COOKIES);
-  // Add Chrome experiment state to the request headers.
-  net::HttpRequestHeaders headers;
-  variations::AppendVariationHeaders(
-      fetcher->GetOriginalURL(), driver_->IsOffTheRecord(), false, &headers);
-  fetcher->SetExtraRequestHeaders(headers.ToString());
-  fetcher->Start();
-
   return true;
 }
 
