--- ./chrome/browser/search/search.cc
+++ ./chrome/browser/search/search.cc.new
@@ -260,17 +260,7 @@
         template_url->new_tab_url_ref(), UIThreadSearchTermsData(profile),
         false, false);
     NewTabURLState state = IsValidNewTabURL(profile, search_provider_url);
-    switch (state) {
-      case NEW_TAB_URL_VALID:
-        // We can use the search provider's page.
-        return NewTabURLDetails(search_provider_url, state);
-      case NEW_TAB_URL_INCOGNITO:
-        // Incognito has its own New Tab.
-        return NewTabURLDetails(GURL(), state);
-      default:
-        // Use the local New Tab otherwise.
-        return NewTabURLDetails(local_url, state);
-    }
+    return NewTabURLDetails(local_url, state);
   }
 
   GURL url;
