--- a/chrome/browser/extensions/external_component_loader.cc
+++ b/chrome/browser/extensions/external_component_loader.cc
@@ -35,6 +35,9 @@ ExternalComponentLoader::~ExternalCompon
 
 void ExternalComponentLoader::StartLoading() {
   prefs_.reset(new base::DictionaryValue());
+  LoadFinished();
+  return;
+
   AddExternalExtension(extension_misc::kInAppPaymentsSupportAppId);
 
   if (HotwordServiceFactory::IsHotwordAllowed(profile_))
