# Modify libc++.a build script to make it build

--- a/third_party/libc++-static/build.sh
+++ b/third_party/libc++-static/build.sh
@@ -8,7 +8,7 @@ set -eux
 REV=245965
 DIR=$(mktemp -d -t libcpp)
 
-THIS_DIR="${PWD}/$(dirname "${0}")"
+THIS_DIR="$(dirname "${BASH_SOURCE[0]}")"
 
 # TODO(thakis): Figure out why our clang complains about visibility and
 # redeclarations.
@@ -16,7 +16,7 @@ THIS_DIR="${PWD}/$(dirname "${0}")"
 CXX=c++
 
 
-FLAGS="-nostdinc++ -O3 -std=c++11 -fstrict-aliasing -fvisibility=hidden -fvisibility-inlines-hidden -mmacosx-version-min=10.6 -arch i386 -arch x86_64 -isysroot $(xcrun -show-sdk-path)"
+FLAGS="-nostdinc++ -O3 -std=c++11 -fstrict-aliasing -fvisibility=hidden -fvisibility-inlines-hidden -mmacosx-version-min=10.6 -arch x86_64 -isysroot $(xcrun -show-sdk-path)"
 
 pushd "${DIR}"
 
@@ -47,7 +47,6 @@ popd
 libtool -static -o libc++.a libcxx*/*.o
 
 cp libc++.a "${THIS_DIR}/libc++.a"
-upload_to_google_storage.py -b chromium-libcpp "${THIS_DIR}/libc++.a"
 
 popd
 rm -rf "${DIR}"
