# Modify libc++.a build script to make it build

--- a/third_party/libc++-static/build.sh
+++ b/third_party/libc++-static/build.sh
@@ -13,15 +13,15 @@ THIS_DIR="${PWD}/$(dirname "${0}")"
 # TODO(thakis): Figure out why our clang complains about visibility and
 # redeclarations.
 #CXX="$THIS_DIR/../llvm-build/Release+Asserts/bin/clang++"
-CXX=c++
+CXX=c++-4.9
 
 
-FLAGS="-nostdinc++ -O3 -std=c++11 -fstrict-aliasing -fvisibility=hidden -fvisibility-inlines-hidden -mmacosx-version-min=10.6 -arch i386 -arch x86_64 -isysroot $(xcrun -show-sdk-path)"
+FLAGS="-nostdinc++ -O3 -std=c++11 -fstrict-aliasing -fvisibility=hidden -fvisibility-inlines-hidden -mmacosx-version-min=10.7 -arch x86_64 -isysroot $(xcrun -show-sdk-path)"
 
 pushd "${DIR}"
 
-svn co --force https://llvm.org/svn/llvm-project/libcxx/trunk@$REV libcxx
-svn co --force https://llvm.org/svn/llvm-project/libcxxabi/trunk@$REV libcxxabi
+svn --non-interactive --trust-server-cert co --force https://llvm.org/svn/llvm-project/libcxx/trunk@$REV libcxx
+svn --non-interactive --trust-server-cert co --force https://llvm.org/svn/llvm-project/libcxxabi/trunk@$REV libcxxabi
 
 mkdir libcxxbuild
 cd libcxxbuild
@@ -47,7 +47,6 @@ popd
 libtool -static -o libc++.a libcxx*/*.o
 
 cp libc++.a "${THIS_DIR}/libc++.a"
-upload_to_google_storage.py -b chromium-libcpp "${THIS_DIR}/libc++.a"
 
 popd
 rm -rf "${DIR}"
