#Fix WidevineCDM compiling on MacOS

--- a/third_party/widevine/cdm/BUILD.gn
+++ b/third_party/widevine/cdm/BUILD.gn
@@ -92,17 +92,7 @@
       "//build/config/sanitizers:deps",
     ]
 
-    if (is_mac) {
-      ldflags = [
-        # Not to strip important symbols by -Wl,-dead_strip.
-        "-Wl,-exported_symbol,_PPP_GetInterface",
-        "-Wl,-exported_symbol,_PPP_InitializeModule",
-        "-Wl,-exported_symbol,_PPP_ShutdownModule",
-      ]
-      #TODO(jrummell) Mac: 'DYLIB_INSTALL_NAME_BASE': '@loader_path',
-    } else if (is_posix && !is_mac) {
-      cflags = [ "-fvisibility=hidden" ]
-    }
+
   }
 } else {
   group("widevinecdm") {
