description: fix errors in gn's bootstrapping script
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -451,7 +451,6 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/memory/ref_counted_memory.cc',
       'base/memory/singleton.cc',
       'base/memory/shared_memory_handle.cc',
-      'base/memory/shared_memory_tracker.cc',
       'base/memory/weak_ptr.cc',
       'base/message_loop/incoming_task_queue.cc',
       'base/message_loop/message_loop.cc',
@@ -460,6 +459,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/message_loop/message_pump_default.cc',
       'base/metrics/bucket_ranges.cc',
       'base/metrics/field_trial.cc',
+      'base/metrics/field_trial_params.cc',
       'base/metrics/field_trial_param_associator.cc',
       'base/metrics/histogram.cc',
       'base/metrics/histogram_base.cc',
@@ -510,6 +510,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/task_scheduler/scheduler_lock_impl.cc',
       'base/task_scheduler/scheduler_single_thread_task_runner_manager.cc',
       'base/task_scheduler/scheduler_worker.cc',
+      'base/task_scheduler/scheduler_worker_pool.cc',
       'base/task_scheduler/scheduler_worker_pool_impl.cc',
       'base/task_scheduler/scheduler_worker_pool_params.cc',
       'base/task_scheduler/scheduler_worker_stack.cc',
@@ -529,6 +530,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/threading/sequence_local_storage_map.cc',
       'base/threading/sequenced_task_runner_handle.cc',
       'base/threading/sequenced_worker_pool.cc',
+      'base/threading/scoped_blocking_call.cc',
       'base/threading/simple_thread.cc',
       'base/threading/thread.cc',
       'base/threading/thread_checker_impl.cc',
@@ -580,7 +582,6 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/tracked_objects.cc',
       'base/tracking_info.cc',
       'base/unguessable_token.cc',
-      'base/value_iterators.cc',
       'base/values.cc',
       'base/value_iterators.cc',
       'base/vlog.cc',
