description: add a file now required by gn
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -384,6 +384,7 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/base_paths.cc',
       'base/base_switches.cc',
       'base/build_time.cc',
+      'base/callback_helpers.cc',
       'base/callback_internal.cc',
       'base/command_line.cc',
       'base/debug/activity_tracker.cc',
