description: set chromedriver version as undefined
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/test/chromedriver/embed_version_in_cpp.py
+++ b/chrome/test/chromedriver/embed_version_in_cpp.py
@@ -89,7 +89,7 @@ def main():
   # On success, version_info.revision_id is a 40-digit Git hash,
   # and version_info.revision is a longer string with more information.
   # On failure, version_info is None.
-  version_info = lastchange.FetchGitRevision(None, '^Change-Id:')
+  version_info = None
 
   if version_info:
     version = get_release_version(options.chrome_version_file, version_info)
