description: map doesn't support insertion from a pair in gcc 6
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/browser/background_fetch/background_fetch_context.cc
+++ b/content/browser/background_fetch/background_fetch_context.cc
@@ -218,7 +218,7 @@ void BackgroundFetchContext::InitializeC
 
   scheduler_->AddJobController(controller.get());
 
-  job_controllers_.insert({unique_id, std::move(controller)});
+  job_controllers_[unique_id] = std::move(controller);
   std::move(done_closure).Run();
 }
 
--- a/content/renderer/input/input_event_prediction.cc
+++ b/content/renderer/input/input_event_prediction.cc
@@ -119,7 +119,7 @@ void InputEventPrediction::UpdateSingleP
     if (predictor != pointer_id_predictor_map_.end()) {
       predictor->second->Update(data);
     } else {
-      pointer_id_predictor_map_.insert({event.id, SetUpPredictor()});
+      pointer_id_predictor_map_[event.id] = SetUpPredictor();
       pointer_id_predictor_map_[event.id]->Update(data);
     }
   }
