description: gcc 6 isn't able to correctly deduce the namespace
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/platform/graphics/path.h
+++ b/third_party/blink/renderer/platform/graphics/path.h
@@ -199,13 +199,13 @@ class PLATFORM_EXPORT Path {
   SkPath path_;
 };
 
-class PLATFORM_EXPORT RefCountedPath : public Path,
+class PLATFORM_EXPORT RefCountedPath : public blink::Path,
                                        public RefCounted<RefCountedPath> {
   USING_FAST_MALLOC(RefCountedPath);
 
  public:
   template <typename... Args>
-  RefCountedPath(Args&&... args) : Path(std::forward<Args>(args)...) {}
+  RefCountedPath(Args&&... args) : blink::Path(std::forward<Args>(args)...) {}
 };
 
 // Only used for DCHECKs
