description: maintain compatibility with system vpx
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_impl.cc
+++ b/third_party/webrtc/modules/video_coding/codecs/vp8/vp8_impl.cc
@@ -1161,9 +1161,6 @@ int VP8DecoderImpl::Decode(const Encoded
 
   img = vpx_codec_get_frame(decoder_, &iter);
   int qp;
-  vpx_codec_err_t vpx_ret =
-      vpx_codec_control(decoder_, VPXD_GET_LAST_QUANTIZER, &qp);
-  RTC_DCHECK_EQ(vpx_ret, VPX_CODEC_OK);
   ret = ReturnFrame(img, input_image._timeStamp, input_image.ntp_time_ms_, qp);
   if (ret != 0) {
     // Reset to avoid requesting key frames too often.
@@ -1217,7 +1214,9 @@ int VP8DecoderImpl::ReturnFrame(const vp
 
   VideoFrame decoded_image(buffer, timestamp, 0, kVideoRotation_0);
   decoded_image.set_ntp_time_ms(ntp_time_ms);
-  decode_complete_callback_->Decoded(decoded_image, rtc::nullopt, qp);
+  int ret = decode_complete_callback_->Decoded(decoded_image);
+  if (ret != 0)
+    return ret;
 
   return WEBRTC_VIDEO_CODEC_OK;
 }
--- a/media/base/decode_capabilities.cc
+++ b/media/base/decode_capabilities.cc
@@ -16,8 +16,8 @@
 // backwards compatibility for legacy applications using the library.
 #define VPX_CODEC_DISABLE_COMPAT 1
 extern "C" {
-#include "third_party/libvpx/source/libvpx/vpx/vp8dx.h"      // nogncheck
-#include "third_party/libvpx/source/libvpx/vpx/vpx_codec.h"  // nogncheck
+#include <vpx/vp8dx.h>
+#include <vpx/vpx_codec.h>
 }
 #endif
 
@@ -106,7 +106,7 @@ bool IsColorSpaceSupported(const media::
 }
 
 bool IsVp9ProfileSupported(VideoCodecProfile profile) {
-#if BUILDFLAG(ENABLE_LIBVPX)
+#if 0
   // High bit depth capabilities may be toggled via LibVPX config flags.
   static bool vpx_supports_high_bit_depth =
       (vpx_codec_get_caps(vpx_codec_vp9_dx()) & VPX_CODEC_CAP_HIGHBITDEPTH) !=
