description: use system libevent
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/webrtc/rtc_base/task_queue_libevent.cc
+++ b/third_party/webrtc/rtc_base/task_queue_libevent.cc
@@ -22,7 +22,7 @@
 #include <type_traits>
 #include <utility>
 
-#include "base/third_party/libevent/event.h"
+#include <event.h>
 #include "rtc_base/checks.h"
 #include "rtc_base/criticalsection.h"
 #include "rtc_base/logging.h"
--- a/base/message_loop/message_pump_libevent.cc
+++ b/base/message_loop/message_pump_libevent.cc
@@ -14,7 +14,7 @@
 #include "base/files/file_util.h"
 #include "base/logging.h"
 #include "base/posix/eintr_wrapper.h"
-#include "base/third_party/libevent/event.h"
+#include <event.h>
 #include "base/time/time.h"
 #include "base/trace_event/trace_event.h"
 #include "build/build_config.h"
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -323,6 +323,7 @@ def WriteGNNinja(path, platform, host, o
           '-lpthread',
           # Additional system libraries that are used.
           '-lnspr4',
+          '-levent',
       ])
     elif platform.is_darwin():
       min_mac_version_flag = '-mmacosx-version-min=10.9'
