description: disable build commands for embedded fontconfig
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/config/posix/BUILD.gn
+++ b/build/config/posix/BUILD.gn
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//buildtools/deps_revisions.gni")
 import("//build/config/c++/c++.gni")
 import("//build/config/clang/clang.gni")
 import("//build/config/compiler/compiler.gni")
--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -206,7 +206,7 @@ static_library("test_support") {
     deps += [ "//third_party/fuchsia-sdk:zx" ]
   }
 
-  if (is_linux) {
+  if (false) {
     public_deps += [ ":fontconfig_util_linux" ]
     data_deps = [
       "//third_party/test_fonts",
@@ -362,7 +362,7 @@ if (is_linux) {
     ]
   }
 
-  if (current_toolchain == host_toolchain) {
+  if (false) {
     executable("generate_fontconfig_caches") {
       testonly = true
       sources = [
--- a/content/shell/test_runner/BUILD.gn
+++ b/content/shell/test_runner/BUILD.gn
@@ -178,7 +178,7 @@ group("resources") {
   if (is_mac) {
     deps += [ ":test_runner_bundle_data" ]
     data_deps += [ ":test_runner_bundle_data" ]
-  } else {
+  } else if (false) {
     deps += [ "//third_party/test_fonts" ]
     data_deps += [ "//third_party/test_fonts" ]
   }
