description: update api to support harfbuzz 2.1
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.cc
+++ b/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.cc
@@ -139,7 +139,7 @@ static hb_position_t HarfBuzzGetGlyphHor
 static void HarfBuzzGetGlyphHorizontalAdvances(hb_font_t* font,
                                                void* font_data,
                                                unsigned count,
-                                               hb_codepoint_t* first_glyph,
+                                               const hb_codepoint_t* first_glyph,
                                                unsigned int glyph_stride,
                                                hb_position_t* first_advance,
                                                unsigned int advance_stride,
--- a/third_party/blink/renderer/platform/fonts/skia/skia_text_metrics.cc
+++ b/third_party/blink/renderer/platform/fonts/skia/skia_text_metrics.cc
@@ -39,13 +39,14 @@ void SkiaTextMetrics::GetGlyphWidthForHa
 }
 
 void SkiaTextMetrics::GetGlyphWidthForHarfBuzz(unsigned count,
-                                               hb_codepoint_t* glyphs,
+                                               const hb_codepoint_t* first_glyph,
                                                unsigned glyph_stride,
                                                hb_position_t* advances,
                                                unsigned advance_stride) {
   // Batch the call to getTextWidths because its function entry cost is not
   // cheap. getTextWidths accepts multiple glyphd ID, but not from a sparse
   // array that copy them to a regular array.
+  hb_codepoint_t* glyphs = const_cast<hb_codepoint_t*>(first_glyph);
   Vector<Glyph, 256> glyph_array(count);
   for (unsigned i = 0; i < count;
        i++, glyphs = advance_by_byte_size(glyphs, glyph_stride)) {
--- a/third_party/blink/renderer/platform/fonts/skia/skia_text_metrics.h
+++ b/third_party/blink/renderer/platform/fonts/skia/skia_text_metrics.h
@@ -19,7 +19,7 @@ class SkiaTextMetrics final {
 
   void GetGlyphWidthForHarfBuzz(hb_codepoint_t, hb_position_t* width);
   void GetGlyphWidthForHarfBuzz(unsigned count,
-                                hb_codepoint_t* first_glyph,
+                                const hb_codepoint_t* first_glyph,
                                 unsigned glyph_stride,
                                 hb_position_t* first_advance,
                                 unsigned advance_stride);
