description: use system jsoncpp
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/jsoncpp/BUILD.gn
+++ b/third_party/jsoncpp/BUILD.gn
@@ -2,34 +2,12 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-config("jsoncpp_config") {
-  include_dirs = [
-    "overrides/include",
-    "source/include",
-  ]
-}
+import("//build/config/linux/pkg_config.gni")
 
-source_set("jsoncpp") {
-  sources = [
-    "overrides/include/json/value.h",
-    "overrides/src/lib_json/json_reader.cpp",
-    "overrides/src/lib_json/json_value.cpp",
-    "source/include/json/assertions.h",
-    "source/include/json/autolink.h",
-    "source/include/json/config.h",
-    "source/include/json/features.h",
-    "source/include/json/forwards.h",
-    "source/include/json/json.h",
-    "source/include/json/reader.h",
-    "source/include/json/writer.h",
-    "source/src/lib_json/json_batchallocator.h",
-    "source/src/lib_json/json_tool.h",
-    "source/src/lib_json/json_writer.cpp",
-  ]
+pkg_config("jsoncpp_config") {
+  packages= [ "jsoncpp" ]
+}
 
+group("jsoncpp") {
   public_configs = [ ":jsoncpp_config" ]
-
-  defines = [ "JSON_USE_EXCEPTION=0" ]
-
-  include_dirs = [ "source/src/lib_json" ]
 }
