description: crashpad relies on multicharacter integer assigments
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/crashpad/crashpad/build/BUILD.gn
+++ b/third_party/crashpad/crashpad/build/BUILD.gn
@@ -20,6 +20,7 @@ import("crashpad_buildconfig.gni")
 
 config("crashpad_is_in_chromium") {
   if (crashpad_is_in_chromium) {
+    cflags = [ "-Wno-multichar" ]
     defines = [ "CRASHPAD_IS_IN_CHROMIUM" ]
   }
 }
