description: avoid chromium's embedded C++ library when bootstrapping gn
author: Michael Gilbert <mgilbert@debian.org>

--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -368,10 +368,6 @@ def WriteGNNinja(path, platform, host, o
         # Use the sid sysroot that UpdateLinuxSysroot() downloads.
         cflags.append('--sysroot=' + linux_sysroot)
         ldflags.append('--sysroot=' + linux_sysroot)
-      ldflags.extend([
-          '-static-libstdc++',
-          '-Wl,--as-needed',
-      ])
       libs.extend([
           # These are needed by libc++.
           '-ldl',
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -88,14 +88,6 @@ def main(argv):
         ['ninja', '-C', gn_build_dir, 'gn', '-w', 'dupbuild=err'])
   shutil.copy2(os.path.join(gn_build_dir, 'gn'), gn_path)
 
-  gn_gen_args = options.gn_gen_args or ''
-  if not options.debug:
-    gn_gen_args += ' is_debug=false'
-  subprocess.check_call([
-      gn_path, 'gen', out_dir,
-      '--args=%s' % gn_gen_args, "--root=" + SRC_ROOT
-  ])
-
 
 if __name__ == '__main__':
   sys.exit(main(sys.argv[1:]))
