--- a/chrome/browser/extensions/external_component_loader.cc
+++ b/chrome/browser/extensions/external_component_loader.cc
@@ -49,6 +49,9 @@ bool ExternalComponentLoader::IsModifiab
 
 void ExternalComponentLoader::StartLoading() {
   prefs_.reset(new base::DictionaryValue());
+  LoadFinished();
+  return;
+
   std::string app_id = extension_misc::kInAppPaymentsSupportAppId;
   prefs_->SetString(app_id + ".external_update_url",
                     extension_urls::GetWebstoreUpdateUrl().spec());
