description: disable hotword service, which downloads binary blobs
bug-debian: http://bugs.debian.org/786909

--- a/chrome/browser/search/hotword_service.cc
+++ b/chrome/browser/search/hotword_service.cc
@@ -632,6 +632,7 @@ bool HotwordService::IsServiceAvailable(
 }
 
 bool HotwordService::IsHotwordAllowed() {
+  return false;
   std::string group = base::FieldTrialList::FindFullName(
       hotword_internal::kHotwordFieldTrialName);
   // Allow hotwording by default, and only disable if the field trial has been
