description: disable third-party cookies by default
author: Chad Miller <chad.miller@canonical.com>

--- a/chrome/browser/content_settings/cookie_settings.cc
+++ b/chrome/browser/content_settings/cookie_settings.cc
@@ -68,7 +68,7 @@ void CookieSettings::Factory::RegisterPr
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
       prefs::kBlockThirdPartyCookies,
-      false,
+      true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
 }
 
