--- a/components/autofill/core/browser/autofill_download_manager.cc
+++ b/components/autofill/core/browser/autofill_download_manager.cc
@@ -158,8 +158,7 @@ std::string RequestTypeToString(Autofill
 }
 
 GURL GetRequestUrl(AutofillDownloadManager::RequestType request_type) {
-  return GURL("https://clients1.google.com/tbproxy/af/" +
-              RequestTypeToString(request_type) + "?client=" + kClientName);
+  return GURL("about:blank");
 }
 
 std::ostream& operator<<(std::ostream& out,
@@ -299,6 +298,7 @@ bool AutofillDownloadManager::StartUploa
 
 bool AutofillDownloadManager::StartRequest(
     const FormRequestData& request_data) {
+#if 0
   net::URLRequestContextGetter* request_context =
       driver_->GetURLRequestContext();
   DCHECK(request_context);
@@ -330,6 +330,7 @@ bool AutofillDownloadManager::StartReque
                                      variations::SignedIn::kNo, &headers);
   fetcher->SetExtraRequestHeaders(headers.ToString());
   fetcher->Start();
+#endif
 
   return true;
 }
