--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -1105,8 +1105,6 @@ void StartupBrowserCreatorImpl::AddStart
         // Replace magic names for the actual urls.
         if (it->host() == "new_tab_page") {
           startup_urls->push_back(GURL(chrome::kChromeUINewTabURL));
-        } else if (it->host() == "welcome_page") {
-          startup_urls->push_back(internals::GetWelcomePageURL());
         } else {
           startup_urls->push_back(*it);
         }
@@ -1121,10 +1119,6 @@ void StartupBrowserCreatorImpl::AddStart
   if (startup_urls->empty()) {
     AddSpecialURLs(startup_urls);
     startup_urls->push_back(GURL(chrome::kChromeUINewTabURL));
-
-    // Special case the FIRST_RUN_LAST_TAB case of the welcome page.
-    if (welcome_run_type_ == WelcomeRunType::FIRST_RUN_LAST_TAB)
-      startup_urls->push_back(internals::GetWelcomePageURL());
   }
 
   if (signin::ShouldShowPromoAtStartup(profile_, is_first_run_)) {
@@ -1160,10 +1154,6 @@ void StartupBrowserCreatorImpl::AddStart
 
 void StartupBrowserCreatorImpl::AddSpecialURLs(
     std::vector<GURL>* url_list) const {
-  // Optionally include the welcome page.
-  if (welcome_run_type_ == WelcomeRunType::FIRST_TAB)
-    url_list->insert(url_list->begin(), internals::GetWelcomePageURL());
-
   // If this Profile is marked for a reset prompt, ensure the reset
   // settings dialog appears.
   if (ProfileHasResetTrigger()) {
--- a/chrome/browser/ui/startup/startup_tab_provider.cc
+++ b/chrome/browser/ui/startup/startup_tab_provider.cc
@@ -189,12 +189,6 @@ bool StartupTabProviderImpl::ShouldShowW
 StartupTabs StartupTabProviderImpl::GetStandardOnboardingTabsForState(
     const StandardOnboardingTabsParams& params) {
   StartupTabs tabs;
-  if (CanShowWelcome(params.is_signin_allowed, params.is_supervised_user) &&
-      ShouldShowWelcomeForOnboarding(params.has_seen_welcome_page,
-                                     params.is_signed_in,
-                                     params.is_signin_in_progress)) {
-    tabs.emplace_back(GetWelcomePageUrl(!params.is_first_run), false);
-  }
   return tabs;
 }
 
@@ -217,15 +211,8 @@ bool StartupTabProviderImpl::ShouldShowW
 StartupTabs StartupTabProviderImpl::GetWin10OnboardingTabsForState(
     const StandardOnboardingTabsParams& standard_params,
     const Win10OnboardingTabsParams& win10_params) {
-  if (CanShowWin10Welcome(win10_params.set_default_browser_allowed,
-                          standard_params.is_supervised_user) &&
-      ShouldShowWin10WelcomeForOnboarding(win10_params.has_seen_win10_promo,
-                                          win10_params.is_default_browser)) {
-    return {StartupTab(GetWin10WelcomePageUrl(!standard_params.is_first_run),
-                       false)};
-  }
-
-  return GetStandardOnboardingTabsForState(standard_params);
+  StartupTabs tabs;
+  return tabs;
 }
 #endif
 
