--- a/services/device/battery/battery_status_service.cc
+++ b/services/device/battery/battery_status_service.cc
@@ -41,6 +41,7 @@ BatteryStatusService::AddCallback(const
   if (!battery_fetcher_)
     battery_fetcher_ = BatteryStatusManager::Create(update_callback_);
 
+#if 0
   if (callback_list_.empty()) {
     bool success = battery_fetcher_->StartListeningBatteryChange();
     // On failure pass the default values back.
@@ -52,6 +53,10 @@ BatteryStatusService::AddCallback(const
     // Send recent status to the new callback if already available.
     callback.Run(status_);
   }
+#else
+  // Always pass the default values.
+  callback.Run(mojom::BatteryStatus());
+#endif
 
   return callback_list_.Add(callback);
 }
