--- a/base/tracked_objects.cc
+++ b/base/tracked_objects.cc
@@ -897,10 +897,14 @@ void ThreadData::EnsureTlsInitialization
   // never again change in this process.
   ++incarnation_counter_;
 
+  ThreadData::Status initial_startup_state = DEACTIVATED;
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch("enable-profiler"))
+    initial_startup_state = kInitialStartupState;
+
   // The lock is not critical for setting status_, but it doesn't hurt.  It also
   // ensures that if we have a racy initialization, that we'll bail as soon as
   // we get the lock earlier in this method.
-  base::subtle::Release_Store(&status_, kInitialStartupState);
+  base::subtle::Release_Store(&status_, initial_startup_state);
   DCHECK(base::subtle::NoBarrier_Load(&status_) != UNINITIALIZED);
 }
 
