--- a/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
+++ b/third_party/ffmpeg/chromium/scripts/build_ffmpeg.py
@@ -640,18 +640,8 @@ def ConfigureAndBuild(target_arch, targe
   if 'win' not in target_os:
     configure_flags['Common'].extend([
         '--enable-pic',
-        '--cc=clang',
-        '--cxx=clang++',
-        '--ld=clang',
     ])
 
-    # Clang Linux will use the first 'ld' it finds on the path, which will
-    # typically be the system one, so explicitly configure use of Clang's
-    # ld.lld, to ensure that things like cross-compilation and LTO work.
-    # This does not work for arm64, ia32 and is always used on mac.
-    if target_arch not in ['arm64', 'ia32', 'mipsel'] and target_os != 'mac':
-      configure_flags['Common'].append('--extra-ldflags=-fuse-ld=lld')
-
   # Should be run on Mac.
   if target_os == 'mac':
     if host_os != 'mac':
