From 0235c2b657d936f3cdb09053776e5929fc84704b Mon Sep 17 00:00:00 2001
From: Tomas Popela <tomas.popela@gmail.com>
Date: Wed, 31 Jan 2018 18:57:07 +0000
Subject: [PATCH] Add missing stdint include

--- a/chrome/browser/vr/sample_queue.cc
+++ b/chrome/browser/vr/sample_queue.cc
@@ -4,6 +4,8 @@
 
 #include <stdint.h>
 
+#include <stdint.h>
+
 #include "chrome/browser/vr/sample_queue.h"
 
 namespace vr {
