--- a/content/common/sandbox_linux/bpf_gpu_policy_linux.cc
+++ b/content/common/sandbox_linux/bpf_gpu_policy_linux.cc
@@ -313,11 +313,11 @@ bool GpuProcessPolicy::PreSandboxHook()
       dlopen(I965DrvVideoPath, RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
       if (I965HybridDrvVideoPath)
         dlopen(I965HybridDrvVideoPath, RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
-      dlopen("libva.so.1", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
+      dlopen("libva.so.2", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
 #if defined(USE_OZONE)
-      dlopen("libva-drm.so.1", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
+      dlopen("libva-drm.so.2", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
 #elif defined(USE_X11)
-      dlopen("libva-x11.so.1", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
+      dlopen("libva-x11.so.2", RTLD_NOW | RTLD_GLOBAL | RTLD_NODELETE);
 #endif
     }
   }
--- a/media/gpu/vaapi_wrapper.cc
+++ b/media/gpu/vaapi_wrapper.cc
@@ -1124,12 +1124,12 @@ void VaapiWrapper::PreSandboxInitializat
 bool VaapiWrapper::PostSandboxInitialization() {
   StubPathMap paths;
 
-  paths[kModuleVa].push_back("libva.so.1");
+  paths[kModuleVa].push_back("libva.so.2");
 
 #if defined(USE_X11)
-  paths[kModuleVa_x11].push_back("libva-x11.so.1");
+  paths[kModuleVa_x11].push_back("libva-x11.so.2");
 #elif defined(USE_OZONE)
-  paths[kModuleVa_drm].push_back("libva-drm.so.1");
+  paths[kModuleVa_drm].push_back("libva-drm.so.2");
 #endif
 
   return InitializeStubs(paths);
