--- a/build/config/compiler/compiler.gni
+++ b/build/config/compiler/compiler.gni
@@ -63,19 +63,19 @@ declare_args() {
   # It's currently not possible to collect AFDO profiles on anything but
   # x86{,_64}.
   using_mismatched_sample_profile = current_cpu != "x64" && current_cpu != "x86"
+
+  # Exclude unwind tables for official builds as unwinding can be done from stack
+  # dumps produced by Crashpad at a later time "offline" in the crash server.
+  # For unofficial (e.g. development) builds and non-Chrome branded (e.g. Cronet
+  # which doesn't use Crashpad, crbug.com/479283) builds it's useful to be able
+  # to unwind at runtime.
+  exclude_unwind_tables =
+      (is_chrome_branded && is_official_build) ||
+      (is_chromecast && !is_cast_desktop_build && !is_debug && !is_fuchsia)
 }
 
 assert(!is_cfi || use_thin_lto, "CFI requires ThinLTO")
 
-# Exclude unwind tables for official builds as unwinding can be done from stack
-# dumps produced by Crashpad at a later time "offline" in the crash server.
-# For unofficial (e.g. development) builds and non-Chrome branded (e.g. Cronet
-# which doesn't use Crashpad, crbug.com/479283) builds it's useful to be able
-# to unwind at runtime.
-exclude_unwind_tables =
-    (is_chrome_branded && is_official_build) ||
-    (is_chromecast && !is_cast_desktop_build && !is_debug && !is_fuchsia)
-
 # If true, optimize for size. Does not affect windows builds.
 # Linux & Mac favor speed over size.
 # TODO(brettw) it's weird that Mac and desktop Linux are different. We should
