--- a/media/gpu/vaapi_wrapper.cc
+++ b/media/gpu/vaapi_wrapper.cc
@@ -1123,12 +1123,12 @@ void VaapiWrapper::PreSandboxInitialization() {
 bool VaapiWrapper::PostSandboxInitialization() {
   StubPathMap paths;
 
-  paths[kModuleVa].push_back("libva.so.1");
+  paths[kModuleVa].push_back("libva.so.2");
 
 #if defined(USE_X11)
-  paths[kModuleVa_x11].push_back("libva-x11.so.1");
+  paths[kModuleVa_x11].push_back("libva-x11.so.2");
 #elif defined(USE_OZONE)
-  paths[kModuleVa_drm].push_back("libva-drm.so.1");
+  paths[kModuleVa_drm].push_back("libva-drm.so.2");
 #endif
 
   return InitializeStubs(paths);
