--- ./chrome/browser/ui/webui/options/browser_options_handler.cc
+++ ./chrome/browser/ui/webui/options/browser_options_handler.cc.new
@@ -1506,7 +1506,7 @@
   SigninManagerBase* signin = SigninManagerFactory::GetForProfile(profile);
   DCHECK(signin);
   sync_status->SetBoolean("signoutAllowed", !signout_prohibited);
-  sync_status->SetBoolean("signinAllowed", signin->IsSigninAllowed());
+  sync_status->SetBoolean("signinAllowed", false);
   sync_status->SetBoolean("syncSystemEnabled", (service != NULL));
   sync_status->SetBoolean("setupCompleted",
                           service && service->HasSyncSetupCompleted());

