description: GCC ICE with optimized version
author: Riku Voipio
bug-debian: http://bugs.debian.org/901290

--- a/third_party/skia/third_party/skcms/skcms.gni
+++ b/third_party/skia/third_party/skcms/skcms.gni
@@ -3,6 +3,10 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+if ((current_cpu == "arm64" || current_cpu == "arm") && !is_clang) {
+    defines = [ "SKCMS_PORTABLE=1" ]
+}
+
 skcms_sources = [
   "src/Curve.c",
   "src/Curve.h",
