description: disable dependency on chrome/android
author: Michael Gilbert <mgilbert@debian.org>

--- a/device/vr/buildflags/buildflags.gni
+++ b/device/vr/buildflags/buildflags.gni
@@ -5,7 +5,6 @@
 import("//build/config/chrome_build.gni")
 import("//build/config/chromecast_build.gni")
 import("//build/config/gclient_args.gni")
-import("//chrome/android/channel.gni")
 
 declare_args() {
   # TODO(733935): Enable for other Android architectures too. Currently we only
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -350,7 +350,7 @@ group("gn_all") {
   #       seems currently broken for this platform at the moment, and the
   #       corresponding code build and works on Linux unmodified.
   #       See instructions in the corresponding BUILD.gn.
-  if (is_linux) {
+  if (is_android) {
     deps +=
         [ "//third_party/android_crazy_linker:android_crazy_linker_zip_fuzzer" ]
   }
