description: never show the default browser question

--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -814,7 +814,7 @@ void StartupBrowserCreatorImpl::AddInfoB
         browser->tab_strip_model()->GetActiveWebContents()));
 
 #if !defined(OS_CHROMEOS)
-    if (!command_line_.HasSwitch(switches::kNoDefaultBrowserCheck)) {
+    if (false) {
       // Generally, the default browser prompt should not be shown on first
       // run. However, when the set-as-default dialog has been suppressed, we
       // need to allow it.
