description: disable device discovery notifications by default
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/856571

--- a/chrome/browser/printing/cloud_print/privet_notifications.cc
+++ b/chrome/browser/printing/cloud_print/privet_notifications.cc
@@ -233,8 +233,7 @@ void PrivetNotificationService::DeviceCa
 // static
 bool PrivetNotificationService::IsEnabled() {
   base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  return !command_line->HasSwitch(
-      switches::kDisableDeviceDiscoveryNotifications);
+  return command_line->HasSwitch(switches::kEnableDeviceDiscoveryNotifications);
 }
 
 // static
