description: fuzzers aren't built, so don't depend on them
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -713,8 +713,7 @@ group("gn_all") {
     }
   }
 
-  if ((is_linux && !is_chromeos && !is_chromecast) || (is_win && use_drfuzz) ||
-      (use_libfuzzer && is_mac)) {
+  if (false) {
     deps += [
       "//testing/libfuzzer/fuzzers",
       "//testing/libfuzzer/tests:libfuzzer_tests",
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -2122,9 +2122,3 @@ test("content_perftests") {
     deps += [ "//testing/android/native_test:native_test_native_code" ]
   }
 }
-
-group("fuzzers") {
-  deps = [
-    "//content/test/fuzzer",
-  ]
-}
--- a/v8/tools/BUILD.gn
+++ b/v8/tools/BUILD.gn
@@ -11,7 +11,6 @@ group("gn_all") {
   data_deps = [
     ":v8_check_static_initializers",
     "gcmole:v8_run_gcmole",
-    "jsfunfuzz:v8_jsfunfuzz",
   ]
 }
 
