description: disable support for openh264, will be added later
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/webrtc/webrtc.gni
+++ b/third_party/webrtc/webrtc.gni
@@ -125,7 +125,7 @@ declare_args() {
   # H.264, compilation succeeds but |H264DecoderImpl| fails to initialize.
   # CHECK THE OPENH264, FFMPEG AND H.264 LICENSES/PATENTS BEFORE BUILDING.
   # http://www.openh264.org, https://www.ffmpeg.org/
-  rtc_use_h264 = proprietary_codecs && !is_android && !is_ios
+  rtc_use_h264 = false
 
   # By default, use normal platform audio support or dummy audio, but don't
   # use file-based audio playout and record.
