author: Michael Gilbert <mgilbert@debian.org>
description: disable the ad promo system by default
bug-debian: http://bugs.debian.org/634101

--- a/chrome/browser/ui/app_list/app_list_service.cc
+++ b/chrome/browser/ui/app_list/app_list_service.cc
@@ -156,7 +156,7 @@ void AppListService::RegisterPrefs(PrefR
 
   // Identifies whether we should show the app launcher promo or not. This
   // becomes false when the user dismisses the promo.
-  registry->RegisterBooleanPref(prefs::kShowAppLauncherPromo, true);
+  registry->RegisterBooleanPref(prefs::kShowAppLauncherPromo, false);
 }
 
 // static
