description: source_ could be uninitialized
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/cacheinvalidation/src/google/cacheinvalidation/include/types.h
+++ b/third_party/cacheinvalidation/src/google/cacheinvalidation/include/types.h
@@ -174,7 +174,7 @@ class ErrorInfo {
  */
 class ObjectId {
  public:
-  ObjectId() : is_initialized_(false) {}
+  ObjectId() : is_initialized_(false), source_(0) {}
 
   /* Creates an object id for the given source and name (the name is copied). */
   ObjectId(int source, const string& name)
