description: fix misleading indentation warning
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/modules/accessibility/ax_menu_list_popup.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_menu_list_popup.cc
@@ -151,8 +151,8 @@ void AXMenuListPopup::DidHide() {
   cache.PostNotification(this, AXObjectCacheImpl::kAXHide);
   if (ActiveDescendant())
     cache.PostNotification(this, AXObjectCacheImpl::kAXChildrenChanged);
-    cache.PostNotification(ActiveDescendant(),
-                           AXObjectCacheImpl::kAXMenuListItemUnselected);
+  cache.PostNotification(ActiveDescendant(),
+                         AXObjectCacheImpl::kAXMenuListItemUnselected);
 }
 
 void AXMenuListPopup::DidShow() {
