description: disable dependencies on third_party/perfetto
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -640,7 +640,7 @@ group("gn_all") {
     }
   }
 
-  if (is_mac || is_linux || is_android) {
+  if (false) {
     deps += [ "//third_party/perfetto:all" ]
   }
 
--- a/services/tracing/BUILD.gn
+++ b/services/tracing/BUILD.gn
@@ -28,7 +28,7 @@ source_set("lib") {
     "//services/tracing/public/cpp",
   ]
 
-  if (is_mac || is_linux || is_android || is_win) {
+  if (false) {
     sources += [
       "perfetto/json_trace_exporter.cc",
       "perfetto/json_trace_exporter.h",
@@ -76,7 +76,7 @@ source_set("tests") {
     "test_util.h",
   ]
 
-  if (is_mac || is_linux || is_android) {
+  if (false) {
     sources += [ "public/cpp/perfetto/trace_event_data_source_unittest.cc" ]
   }
 
@@ -97,7 +97,7 @@ source_set("tests") {
     "//testing/gtest",
   ]
 
-  if (is_mac || is_linux || is_android) {
+  if (false) {
     sources += [
       "perfetto/json_trace_exporter_unittest.cc",
       "perfetto/perfetto_integration_unittest.cc",
--- a/services/tracing/tracing_service.h
+++ b/services/tracing/tracing_service.h
@@ -17,8 +17,7 @@
 #include "services/tracing/agent_registry.h"
 #include "services/tracing/coordinator.h"
 
-#if defined(OS_ANDROID) || defined(OS_LINUX) || defined(OS_MACOSX) || \
-    defined(OS_WIN)
+#if 0
 #define PERFETTO_SERVICE_AVAILABLE
 #endif
 
--- a/services/tracing/public/cpp/BUILD.gn
+++ b/services/tracing/public/cpp/BUILD.gn
@@ -22,7 +22,7 @@ component("cpp") {
     "//services/tracing/public/mojom",
   ]
 
-  if (is_mac || is_linux || is_android || is_win) {
+  if (false) {
     sources += [
       "perfetto/producer_client.cc",
       "perfetto/producer_client.h",
--- a/services/tracing/public/cpp/trace_event_agent.cc
+++ b/services/tracing/public/cpp/trace_event_agent.cc
@@ -22,8 +22,7 @@
 #include "services/tracing/public/cpp/tracing_features.h"
 #include "services/tracing/public/mojom/constants.mojom.h"
 
-#if defined(OS_ANDROID) || defined(OS_LINUX) || defined(OS_MACOSX) || \
-    defined(OS_WIN)
+#if 0
 #define PERFETTO_AVAILABLE
 #include "services/tracing/public/cpp/perfetto/producer_client.h"
 #include "services/tracing/public/cpp/perfetto/trace_event_data_source.h"
