Include system copy of prtime.h

--- a/base/base.gypi
+++ b/base/base.gypi
@@ -622,8 +622,6 @@
           'third_party/dmg_fp/g_fmt.cc',
           'third_party/icu/icu_utf.cc',
           'third_party/icu/icu_utf.h',
-          'third_party/nspr/prtime.cc',
-          'third_party/nspr/prtime.h',
           'third_party/superfasthash/superfasthash.c',
           'third_party/xdg_mime/xdgmime.h',
           'thread_task_runner_handle.cc',
--- a/base/time/pr_time_unittest.cc
+++ b/base/time/pr_time_unittest.cc
@@ -6,7 +6,7 @@
 #include <time.h>
 
 #include "base/compiler_specific.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time/time.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
--- a/base/time/time.cc
+++ b/base/time/time.cc
@@ -13,7 +13,8 @@
 #include "base/lazy_instance.h"
 #include "base/logging.h"
 #include "base/strings/stringprintf.h"
-#include "base/third_party/nspr/prtime.h"
+
+#include <nspr/prtime.h>
 
 namespace base {
 
