--- a/clean
+++ b/clean
@@ -1,23 +1 @@
 debian/files
-
-Makefile
-third_party/flot/*.js
-chrome/test/data/webui/i18n_process_css_test.html
-
-third_party/ffmpeg/BUILD.gn
-third_party/flac/BUILD.gn
-base/third_party/libevent/BUILD.gn
-build/secondary/third_party/libjpeg_turbo/BUILD.gn
-third_party/libdrm/BUILD.gn
-third_party/libpng/BUILD.gn
-third_party/libvpx/BUILD.gn
-third_party/libwebp/BUILD.gn
-third_party/libxml/BUILD.gn
-third_party/libxslt/BUILD.gn
-third_party/re2/BUILD.gn
-third_party/snappy/BUILD.gn
-third_party/yasm/yasm_assemble.gni
-third_party/zlib/BUILD.gn
-third_party/icu/BUILD.gn
-third_party/opus/BUILD.gn
-third_party/freetype/BUILD.gn
--- a/compat
+++ b/compat
@@ -1 +1 @@
-10
+9
--- a/control
+++ b/control
@@ -12,7 +12,7 @@ Build-Depends:
 # clang (>= 3.5),
  clang-3.9,
  llvm-3.9-dev,
- debhelper (>= 10),
+ debhelper (>= 9),
  python,
  python3,
  pkg-config,
@@ -51,7 +51,6 @@ Build-Depends:
  libffi-dev,
  libkrb5-dev,
  libexif-dev,
- libflac-dev,
  libudev-dev,
  libopus-dev,
  libwebp-dev,
@@ -67,13 +66,12 @@ Build-Depends:
  libsnappy-dev,
  libgconf2-dev,
  libavutil-dev,
- libavcodec-dev (>= 7),
  libavformat-dev,
  libglib2.0-dev,
  libasound2-dev,
  libsqlite3-dev,
  libjsoncpp-dev,
- libspeechd-dev (>= 0.8.3),
+ libspeechd-dev,
  libminizip-dev,
  libhunspell-dev,
  libharfbuzz-dev,
--- a/rules.in
+++ b/rules.in
@@ -42,10 +42,6 @@ ifneq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
 njobs=-j$(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
 endif
 
-# paths to files needed by flot
-flotpaths=/usr/share/javascript/jquery/*min.js \
-          /usr/share/javascript/jquery-flot/*min.js \
-
 %:
 	dh $@
 
@@ -56,13 +52,6 @@ $ungoog{build_output}/gn:
 override_dh_auto_configure:
 	# output compiler information
 	$(CXX) --version
-	# use system flot
-	for file in $(flotpaths); do ln -sf $$file third_party/flot; done
-	# strip out system third_party libraries
-	cp third_party/freetype/src/src/psnames/pstables.h .
-	./debian/scripts/unbundle
-	mkdir -p third_party/freetype/src/src/psnames
-	mv pstables.h third_party/freetype/src/src/psnames
 
 override_dh_auto_build-arch: $ungoog{build_output}/gn
 	./$ungoog{build_output}/gn gen $ungoog{build_output} --args="$(defines)" --fail-on-unused-args
--- a/ungoogled-chromium-common.install.in
+++ b/ungoogled-chromium-common.install.in
@@ -1 +1,6 @@
 $ungoog{build_output}/*.bin usr/lib/chromium
+$ungoog{build_output}/icudtl.dat usr/lib/chromium
+$ungoog{build_output}/libEGL.so usr/lib/chromium
+$ungoog{build_output}/libGLESv2.so usr/lib/chromium
+$ungoog{build_output}/xdg-mime usr/lib/chromium
+$ungoog{build_output}/xdg-settings usr/lib/chromium
