diff --git standard/clean conservative/clean
index c41c88c..bbe760d 100644
--- standard/clean
+++ conservative/clean
@@ -1,24 +1 @@
 debian/files
-
-Makefile
-third_party/flot/*.js
-chrome/test/data/webui/i18n_process_css_test.html
-
-third_party/ffmpeg/BUILD.gn
-third_party/flac/BUILD.gn
-third_party/harfbuzz-ng/BUILD.gn
-base/third_party/libevent/BUILD.gn
-build/secondary/third_party/libjpeg_turbo/BUILD.gn
-third_party/libdrm/BUILD.gn
-third_party/libpng/BUILD.gn
-third_party/libvpx/BUILD.gn
-third_party/libwebp/BUILD.gn
-third_party/libxml/BUILD.gn
-third_party/libxslt/BUILD.gn
-third_party/re2/BUILD.gn
-third_party/snappy/BUILD.gn
-third_party/yasm/yasm_assemble.gni
-third_party/zlib/BUILD.gn
-third_party/icu/BUILD.gn
-third_party/opus/BUILD.gn
-third_party/freetype/BUILD.gn
diff --git standard/control conservative/control
index fe97098..9397cca 100644
--- standard/control
+++ conservative/control
@@ -12,7 +12,7 @@
 # clang (>= 3.5),
  clang-3.9,
  llvm-3.9-dev,
- debhelper (>= 10),
+ debhelper (>= 9),
  python,
  python3,
  pkg-config,
@@ -50,7 +50,6 @@ Build-Depends:
  libffi-dev,
  libkrb5-dev,
  libexif-dev,
- libflac-dev,
  libudev-dev,
  libopus-dev,
  libwebp-dev,
@@ -65,13 +64,12 @@ Build-Depends:
  libsnappy-dev,
  libgconf2-dev,
  libavutil-dev,
- libavcodec-dev (>= 7),
  libavformat-dev,
  libglib2.0-dev,
  libasound2-dev,
  libsqlite3-dev,
  libjsoncpp-dev,
- libspeechd-dev (>= 0.8.3),
+ libspeechd-dev,
  libminizip-dev,
  libhunspell-dev,
  libharfbuzz-dev,
diff --git standard/rules.in conservative/rules.in
index edcea2a..7f60b26 100755
--- standard/rules.in
+++ conservative/rules.in
@@ -42,12 +42,8 @@
 njobs=-j$(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
 endif
 
-# paths to files needed by flot
-flotpaths=/usr/share/javascript/jquery/*min.js \
-          /usr/share/javascript/jquery-flot/*min.js \
-
 %:
-	dh $@
+	dh $@ --parallel

 $ungoog{build_output}/gn:
 	mkdir -p $ungoog{build_output} || true
@@ -56,13 +52,6 @@
 override_dh_auto_configure:
 	# output compiler information
 	$(CXX) --version
-	# use system flot
-	for file in $(flotpaths); do ln -sf $$file third_party/flot; done
-	# strip out system third_party libraries
-	cp third_party/freetype/src/src/psnames/pstables.h .
-	./debian/scripts/unbundle
-	mkdir -p third_party/freetype/src/src/psnames
-	mv pstables.h third_party/freetype/src/src/psnames

 override_dh_auto_build-arch: $ungoog{build_output}/gn
 	./$ungoog{build_output}/gn gen $ungoog{build_output} --args="$(defines)" --fail-on-unused-args
diff --git standard/ungoogled-chromium.install.in conservative/ungoogled-chromium.install.in
index 39d32b6..35f6e66 100644
--- standard/ungoogled-chromium.install.in
+++ conservative/ungoogled-chromium.install.in
@@ -2,6 +2,8 @@ $ungoog{build_output}/chromium usr/lib/chromium
 
 $ungoog{build_output}/chrome-sandbox usr/lib/chromium
 
+$ungoog{build_output}/icudtl.dat usr/lib/chromium
+
 $ungoog{build_output}/chrome_*.pak usr/lib/chromium
 $ungoog{build_output}/resources.pak usr/lib/chromium
 $ungoog{build_output}/keyboard_resources.pak usr/lib/chromium
 
