description: use system lcms2
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/879153

--- a/third_party/pdfium/BUILD.gn
+++ b/third_party/pdfium/BUILD.gn
@@ -192,6 +192,7 @@ jumbo_static_library("pdfium") {
     ":fxge",
     ":fxjs",
     ":pwl",
+    "third_party:lcms2",
     "third_party:bigint",
     "third_party:pdfium_base",
     "third_party:skia_shared",
@@ -682,6 +683,7 @@ jumbo_static_library("fpdftext") {
   configs += [ ":pdfium_core_config" ]
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 }
 
@@ -1080,6 +1082,7 @@ jumbo_static_library("fxge") {
 
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 
   if (is_component_build || use_system_freetype) {
