description: maintain compatibility with openjpeg 2.1
author: chromium 57 authors

--- a/third_party/pdfium/core/fxcodec/codec/fx_codec_jpx_opj.cpp
+++ b/third_party/pdfium/core/fxcodec/codec/fx_codec_jpx_opj.cpp
@@ -40,20 +40,20 @@ opj_stream_t* fx_opj_stream_create_memor
 }
 
 bool alloc_rgb(int** out_r, int** out_g, int** out_b, size_t size) {
-  int* r = static_cast<int*>(opj_image_data_alloc(size));
+  int* r = FX_Alloc(int, size);
   if (!r)
     return false;
 
-  int* g = static_cast<int*>(opj_image_data_alloc(size));
+  int* g = FX_Alloc(int, size);
   if (!g) {
-    opj_image_data_free(r);
+    FX_Free(r);
     return false;
   }
 
-  int* b = static_cast<int*>(opj_image_data_alloc(size));
+  int* b = FX_Alloc(int, size);
   if (!b) {
-    opj_image_data_free(r);
-    opj_image_data_free(g);
+    FX_Free(r);
+    FX_Free(g);
     return false;
   }
 
@@ -114,9 +114,9 @@ void sycc444_to_rgb(opj_image_t* img) {
   for (size_t i = 0; i < max_size.ValueOrDie(); ++i) {
     sycc_to_rgb(offset, upb, *y++, *cb++, *cr++, r++, g++, b++);
   }
-  opj_image_data_free(img->comps[0].data);
-  opj_image_data_free(img->comps[1].data);
-  opj_image_data_free(img->comps[2].data);
+  FX_Free(img->comps[0].data);
+  FX_Free(img->comps[1].data);
+  FX_Free(img->comps[2].data);
   img->comps[0].data = d0;
   img->comps[1].data = d1;
   img->comps[2].data = d2;
@@ -182,9 +182,9 @@ void sycc422_to_rgb(opj_image_t* img) {
       sycc_to_rgb(offset, upb, *y++, *cb++, *cr++, r++, g++, b++);
     }
   }
-  opj_image_data_free(img->comps[0].data);
-  opj_image_data_free(img->comps[1].data);
-  opj_image_data_free(img->comps[2].data);
+  FX_Free(img->comps[0].data);
+  FX_Free(img->comps[1].data);
+  FX_Free(img->comps[2].data);
   img->comps[0].data = d0;
   img->comps[1].data = d1;
   img->comps[2].data = d2;
@@ -438,9 +438,9 @@ void sycc420_to_rgb(opj_image_t* img) {
     }
   }
 
-  opj_image_data_free(img->comps[0].data);
-  opj_image_data_free(img->comps[1].data);
-  opj_image_data_free(img->comps[2].data);
+  FX_Free(img->comps[0].data);
+  FX_Free(img->comps[1].data);
+  FX_Free(img->comps[2].data);
   img->comps[0].data = d0;
   img->comps[1].data = d1;
   img->comps[2].data = d2;
--- a/third_party/pdfium/core/fxcodec/codec/cjpx_decoder.h
+++ b/third_party/pdfium/core/fxcodec/codec/cjpx_decoder.h
@@ -12,7 +12,7 @@
 
 #include "core/fxcodec/codec/codec_int.h"
 #include "core/fxcrt/unowned_ptr.h"
-#include <openjpeg-2.3/openjpeg.h>
+#include <openjpeg-2.1/openjpeg.h>
 
 class CPDF_ColorSpace;
 
--- a/third_party/pdfium/core/fxcodec/codec/codec_int.h
+++ b/third_party/pdfium/core/fxcodec/codec/codec_int.h
@@ -15,7 +15,7 @@
 #include <vector>
 
 #include "core/fxcodec/jbig2/JBig2_Context.h"
-#include <openjpeg-2.3/openjpeg.h>
+#include <openjpeg-2.1/openjpeg.h>
 
 class CPDF_ColorSpace;
 
