--- a/third_party/leveldatabase/leveldatabase.gyp
+++ b/third_party/leveldatabase/leveldatabase.gyp
@@ -56,7 +56,7 @@
       'conditions': [
         ['use_snappy', {
           'dependencies': [
-            '../../third_party/snappy/snappy.gyp:snappy',
+            '../../build/linux/unbundle/snappy.gyp:snappy',
           ],
         }],
       ],
--- a/third_party/WebKit/Source/core/core.gyp
+++ b/third_party/WebKit/Source/core/core.gyp
@@ -187,7 +187,7 @@
         '<(DEPTH)/third_party/libxslt/libxslt.gyp:libxslt',
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         '<(DEPTH)/third_party/qcms/qcms.gyp:qcms',
-        '<(DEPTH)/third_party/snappy/snappy.gyp:snappy',
+        '<(DEPTH)/build/linux/unbundle/snappy.gyp:snappy',
         '<(DEPTH)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(DEPTH)/url/url.gyp:url_lib',
         '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
@@ -703,7 +703,7 @@
         '<(DEPTH)/third_party/libxslt/libxslt.gyp:libxslt',
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         '<(DEPTH)/third_party/qcms/qcms.gyp:qcms',
-        '<(DEPTH)/third_party/snappy/snappy.gyp:snappy',
+        '<(DEPTH)/build/linux/unbundle/snappy.gyp:snappy',
         '<(DEPTH)/third_party/sqlite/sqlite.gyp:sqlite',
         '<(DEPTH)/url/url.gyp:url_lib',
         '<(DEPTH)/v8/tools/gyp/v8.gyp:v8',
--- a/build/linux/unbundle/snappy.gyp
+++ b/build/linux/unbundle/snappy.gyp
@@ -17,7 +17,7 @@
         ],
       },
       'includes': [
-        '../../build/shim_headers.gypi',
+        '../../shim_headers.gypi',
       ],
       'link_settings': {
         'libraries': [
--- a/third_party/leveldatabase/port/port_chromium.cc
+++ b/third_party/leveldatabase/port/port_chromium.cc
@@ -8,7 +8,7 @@
 #include "util/logging.h"
 
 #if defined(USE_SNAPPY)
-#  include "third_party/snappy/src/snappy.h"
+#  include <snappy.h>
 #endif
 
 namespace leveldb {
--- a/third_party/WebKit/Source/bindings/core/v8/V8ScriptRunner.cpp
+++ b/third_party/WebKit/Source/bindings/core/v8/V8ScriptRunner.cpp
@@ -38,7 +38,7 @@
 #include "platform/ScriptForbiddenScope.h"
 #include "platform/TraceEvent.h"
 #include "public/platform/Platform.h"
-#include "third_party/snappy/src/snappy.h"
+#include <snappy.h>
 #include "wtf/CurrentTime.h"
 
 #if defined(WTF_OS_WIN)
