# Fix compiling on macOS 10.11

--- a/base/files/memory_mapped_file_posix.cc
+++ b/base/files/memory_mapped_file_posix.cc
@@ -116,7 +116,7 @@ bool MemoryMappedFile::MapFileRegionToMe
       // available via fcntl.
       // See also: https://openradar.appspot.com/32720223
       fstore_t params = {F_ALLOCATEALL, F_PEOFPOSMODE, region.offset,
-                         region.size, 0};
+                         (off_t)region.size, 0};
       if (fcntl(file_.GetPlatformFile(), F_PREALLOCATE, &params) != 0) {
         DPLOG(ERROR) << "F_PREALLOCATE";
         // This can fail because the filesystem doesn't support it so don't
