--- a/chrome/browser/ui/webui/options/chromeos/system_settings_provider.h
+++ b/chrome/browser/ui/webui/options/chromeos/system_settings_provider.h
@@ -10,7 +10,7 @@
 #include "base/string16.h"
 #include "chrome/browser/chromeos/cros_settings_provider.h"
 #include "chrome/browser/chromeos/system_access.h"
-#include "third_party/icu/public/i18n/unicode/timezone.h"
+#include <unicode/timezone.h>
 
 class Value;
 class ListValue;
--- a/app/app_base.gypi
+++ b/app/app_base.gypi
@@ -17,8 +17,7 @@
         '../ui/ui.gyp:ui_gfx',
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libpng/libpng.gyp:libpng',
         '../third_party/sqlite/sqlite.gyp:sqlite',
         '../third_party/zlib/zlib.gyp:zlib',
--- a/app/app.gyp
+++ b/app/app.gyp
@@ -22,8 +22,7 @@
         '../skia/skia.gyp:skia',
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libpng/libpng.gyp:libpng',
         '../third_party/zlib/zlib.gyp:zlib',
         '../ui/ui.gyp:ui_base',
--- a/base/base.gyp
+++ b/base/base.gyp
@@ -16,8 +16,7 @@
       'msvs_guid': '968F3222-9798-4D21-BE08-15ECB5EF2994',
       'dependencies': [
         'base',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'conditions': [
         ['toolkit_uses_gtk==1', {
@@ -234,8 +233,7 @@
         'third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'conditions': [
         ['toolkit_uses_gtk==1', {
--- a/build/all.gyp
+++ b/build/all.gyp
@@ -34,7 +34,6 @@
         '../third_party/codesighs/codesighs.gyp:*',
         '../third_party/ffmpeg/ffmpeg.gyp:*',
         '../third_party/iccjpeg/iccjpeg.gyp:*',
-        '../third_party/icu/icu.gyp:*',
         '../third_party/libpng/libpng.gyp:*',
         '../third_party/libwebp/libwebp.gyp:*',
         '../third_party/libxml/libxml.gyp:*',
--- a/build/linux/system.gyp
+++ b/build/linux/system.gyp
@@ -360,6 +360,26 @@
       }]]
     },
     {
+      'target_name': 'icu',
+      'type': 'settings',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@()',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(icu-config --ldflags)',
+            ],
+            'libraries': [
+              '<!@(icu-config --ldflags-libsonly)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome-keyring',
       'type': 'settings',
       'conditions': [
--- a/build/temp_gyp/googleurl.gyp
+++ b/build/temp_gyp/googleurl.gyp
@@ -14,9 +14,7 @@
       'msvs_guid': 'EF5E94AB-B646-4E5B-A058-52EF07B8351C',
       'dependencies': [
         '../../base/base.gyp:base',
-        '../../third_party/icu/icu.gyp:icudata',
-        '../../third_party/icu/icu.gyp:icui18n',
-        '../../third_party/icu/icu.gyp:icuuc',
+        '../../build/linux/system.gyp:icu',
       ],
       'sources': [
         '../../googleurl/src/gurl.cc',
@@ -73,7 +71,7 @@
         '../../base/base.gyp:base_i18n',
         '../../base/base.gyp:test_support_base',
         '../../testing/gtest.gyp:gtest',
-        '../../third_party/icu/icu.gyp:icuuc',
+        '../../build/linux/system.gyp:icu',
       ],
       'sources': [
         '../../googleurl/src/gurl_unittest.cc',
--- a/chrome/chrome_browser.gypi
+++ b/chrome/chrome_browser.gypi
@@ -39,8 +39,7 @@
         '../third_party/bzip2/bzip2.gyp:bzip2',
         '../third_party/expat/expat.gyp:expat',
         '../third_party/hunspell/hunspell.gyp:hunspell',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libjingle/libjingle.gyp:libjingle',
         '../third_party/libphonenumber/libphonenumber.gyp:libphonenumber',
         '../third_party/libphonenumber/libphonenumber.gyp:phonenumber_proto',
--- a/chrome/chrome_common.gypi
+++ b/chrome/chrome_common.gypi
@@ -124,8 +124,7 @@
         '../printing/printing.gyp:printing',
         '../skia/skia.gyp:skia',
         '../third_party/bzip2/bzip2.gyp:bzip2',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libxml/libxml.gyp:libxml',
         '../third_party/sqlite/sqlite.gyp:sqlite',
         '../third_party/zlib/zlib.gyp:zlib',
@@ -375,8 +374,7 @@
         '../gpu/gpu.gyp:gpu_ipc',
         '../net/net.gyp:net_resources',
         '../net/net.gyp:net',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'conditions': [
         ['os_posix == 1 and OS != "mac"', {
--- a/chrome/chrome.gyp
+++ b/chrome/chrome.gyp
@@ -427,8 +427,7 @@
         'theme_resources',
         'theme_resources_standard',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'include_dirs': [
         '..',
@@ -535,7 +534,7 @@
         '../base/base.gyp:base',
         '../build/temp_gyp/googleurl.gyp:googleurl',
         '../jingle/jingle.gyp:notifier',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/sqlite/sqlite.gyp:sqlite',
         'app/policy/cloud_policy_codegen.gyp:policy',
         'browser/sync/protocol/sync_proto.gyp:sync_proto_cpp',
@@ -1335,7 +1334,7 @@
             '../third_party/bzip2/bzip2.gyp:*',
             '../third_party/codesighs/codesighs.gyp:*',
             '../third_party/iccjpeg/iccjpeg.gyp:*',
-            '../third_party/icu/icu.gyp:*',
+            '../build/linux/system.gyp:icu',
             '../third_party/libpng/libpng.gyp:*',
             '../third_party/libwebp/libwebp.gyp:*',
             '../third_party/libxslt/libxslt.gyp:*',
--- a/chrome/chrome_renderer.gypi
+++ b/chrome/chrome_renderer.gypi
@@ -21,8 +21,7 @@
         '../skia/skia.gyp:skia',
         '../third_party/cld/cld.gyp:cld',
         '../third_party/hunspell/hunspell.gyp:hunspell',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/npapi/npapi.gyp:npapi',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
         '../ui/gfx/surface/surface.gyp:surface',
--- a/chrome/chrome_tests.gypi
+++ b/chrome/chrome_tests.gypi
@@ -437,7 +437,7 @@
         '../net/net.gyp:net_resources',
         '../net/net.gyp:net_test_support',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
+        '../build/linux/system.gyp:icu',
         '../third_party/libpng/libpng.gyp:libpng',
         '../third_party/zlib/zlib.gyp:zlib',
         '../testing/gtest.gyp:gtest',
@@ -613,8 +613,7 @@
         '../skia/skia.gyp:skia',
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libxml/libxml.gyp:libxml',
         # run time dependencies
         'default_plugin/default_plugin.gyp:default_plugin',
@@ -790,8 +789,7 @@
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libxml/libxml.gyp:libxml',
       ],
       'include_dirs': [
@@ -1051,8 +1049,7 @@
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libxml/libxml.gyp:libxml',
       ],
       'include_dirs': [
@@ -1119,8 +1116,7 @@
         '../net/net.gyp:net',
         '../skia/skia.gyp:skia',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'include_dirs': [
         '..',
@@ -1192,8 +1188,7 @@
         '../skia/skia.gyp:skia',
         '../third_party/bzip2/bzip2.gyp:bzip2',
         '../third_party/cld/cld.gyp:cld',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libjingle/libjingle.gyp:libjingle',
         '../third_party/libxml/libxml.gyp:libxml',
         '../ui/gfx/gl/gl.gyp:gl',
@@ -2232,8 +2227,7 @@
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
         '../third_party/cld/cld.gyp:cld',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../v8/tools/gyp/v8.gyp:v8',
         # Runtime dependencies
         '../third_party/mesa/mesa.gyp:osmesa',
@@ -3260,8 +3254,7 @@
         '../skia/skia.gyp:skia',
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/npapi/npapi.gyp:npapi',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
       ],
@@ -3420,8 +3413,7 @@
         '../net/net.gyp:net_test_support',
         '../skia/skia.gyp:skia',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         # Runtime dependencies
         '../third_party/mesa/mesa.gyp:osmesa',
       ],
--- a/chrome/default_plugin/default_plugin.gyp
+++ b/chrome/default_plugin/default_plugin.gyp
@@ -14,8 +14,7 @@
         ':default_plugin_resources',
         '<(DEPTH)/base/base.gyp:base',
         '<(DEPTH)/net/net.gyp:net_resources',
-        '<(DEPTH)/third_party/icu/icu.gyp:icui18n',
-        '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
+        '<(DEPTH)/build/linux/system.gyp:icu',
         '<(DEPTH)/third_party/libxml/libxml.gyp:libxml',
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         '<(DEPTH)/webkit/support/webkit_support.gyp:webkit_resources',
--- a/chrome_frame/chrome_frame.gyp
+++ b/chrome_frame/chrome_frame.gyp
@@ -392,8 +392,7 @@
         '../net/net.gyp:net_test_support',
         '../skia/skia.gyp:skia',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         'chrome_frame_npapi',
         'chrome_frame_ie',
         'chrome_tab_idl',
--- a/content/content_common.gypi
+++ b/content/content_common.gypi
@@ -11,7 +11,7 @@
         '../base/base.gyp:base',
         '../ipc/ipc.gyp:ipc',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/npapi/npapi.gyp:npapi',
         '../third_party/WebKit/Source/WebKit/chromium/WebKit.gyp:webkit',
         '../ui/gfx/gl/gl.gyp:gl',
--- a/content/content_renderer.gypi
+++ b/content/content_renderer.gypi
@@ -14,8 +14,7 @@
         '../ppapi/ppapi_internal.gyp:ppapi_proxy',
         '../skia/skia.gyp:skia',
         '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
-        '../third_party/icu/icu.gyp:icuuc',
-        '../third_party/icu/icu.gyp:icui18n',
+        '../build/linux/system.gyp:icu',
         '../third_party/libjingle/libjingle.gyp:libjingle',
         '../third_party/libjingle/libjingle.gyp:libjingle_p2p',
         '../third_party/npapi/npapi.gyp:npapi',
--- a/net/net.gyp
+++ b/net/net.gyp
@@ -17,8 +17,7 @@
         '../build/temp_gyp/googleurl.gyp:googleurl',
         '../crypto/crypto.gyp:crypto',
         '../sdch/sdch.gyp:sdch',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/zlib/zlib.gyp:zlib',
         'net_resources',
         'ssl_false_start_blacklist_process#host',
@@ -1034,8 +1033,7 @@
             # This is needed to trigger the dll copy step on windows.
             # TODO(mark): Specifying this here shouldn't be necessary.
             'dependencies': [
-              '../third_party/icu/icu.gyp:icudata',
-            ],
+              '../build/linux/system.gyp:icu',        ],
           },
         ],
       ],
@@ -1063,7 +1061,7 @@
         # TODO(mark): Specifying this here shouldn't be necessary.
         [ 'OS == "win"', {
             'dependencies': [
-              '../third_party/icu/icu.gyp:icudata',
+              '../build/linux/system.gyp:icu',
             ],
           },
         ],
--- a/printing/printing.gyp
+++ b/printing/printing.gyp
@@ -16,8 +16,7 @@
         '../base/base.gyp:base_i18n',
         '../build/temp_gyp/googleurl.gyp:googleurl',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'msvs_guid': '9E5416B9-B91B-4029-93F4-102C1AD5CAF4',
       'include_dirs': [
--- a/skia/skia.gyp
+++ b/skia/skia.gyp
@@ -766,7 +766,7 @@
             '../build/linux/system.gyp:fontconfig',
             '../build/linux/system.gyp:freetype2',
             '../third_party/harfbuzz/harfbuzz.gyp:harfbuzz',
-            '../third_party/icu/icu.gyp:icuuc',
+            '../build/linux/system.gyp:icu',
           ],
           'cflags': [
             '-Wno-unused',
--- a/third_party/cld/cld.gyp
+++ b/third_party/cld/cld.gyp
@@ -8,7 +8,7 @@
       'target_name': 'cld',
       'type': 'static_library',
       'dependencies': [
-      	'../icu/icu.gyp:icuuc',
+        '../../build/linux/system.gyp:icu',
       ],
       'include_dirs': [
         '.',
--- a/third_party/harfbuzz-ng/harfbuzz.gyp
+++ b/third_party/harfbuzz-ng/harfbuzz.gyp
@@ -60,7 +60,7 @@
       },
       'dependencies': [
         '../../build/linux/system.gyp:freetype2',
-        '../../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
     },
   ],
--- a/third_party/hunspell/hunspell.gyp
+++ b/third_party/hunspell/hunspell.gyp
@@ -10,7 +10,7 @@
       'msvs_guid': 'D5E8DCB2-9C61-446F-8BEE-B18CA0E0936E',
       'dependencies': [
         '../../base/base.gyp:base',
-        '../icu/icu.gyp:icuuc',
+        '../../build/linux/system.gyp:icu',
       ],
       'defines': [
         'HUNSPELL_STATIC',
--- a/third_party/libxml/libxml.gyp
+++ b/third_party/libxml/libxml.gyp
@@ -163,11 +163,11 @@
             'src/include',
           ],
           'dependencies': [
-            '../icu/icu.gyp:icuuc',
+            '../../build/linux/system.gyp:icu',
             '../zlib/zlib.gyp:zlib',
           ],
           'export_dependent_settings': [
-            '../icu/icu.gyp:icuuc',
+            '../../build/linux/system.gyp:icu',
           ],
           'direct_dependent_settings': {
             'defines': [
--- a/third_party/sqlite/sqlite.gyp
+++ b/third_party/sqlite/sqlite.gyp
@@ -97,8 +97,7 @@
             'src/src',
           ],
           'dependencies': [
-            '../icu/icu.gyp:icui18n',
-            '../icu/icu.gyp:icuuc',
+            '../../build/linux/system.gyp:icu',
           ],
           'direct_dependent_settings': {
             'include_dirs': [
@@ -131,7 +130,7 @@
           'target_name': 'sqlite_shell',
           'type': 'executable',
           'dependencies': [
-            '../icu/icu.gyp:icuuc',
+            '../../build/linux/system.gyp:icu',
             'sqlite',
           ],
           'sources': [
--- a/third_party/WebKit/Source/JavaScriptCore/JavaScriptCore.gyp/JavaScriptCore.gyp
+++ b/third_party/WebKit/Source/JavaScriptCore/JavaScriptCore.gyp/JavaScriptCore.gyp
@@ -97,8 +97,7 @@
       'msvs_guid': 'AA8A5A85-592B-4357-BC60-E0E91E026AF6',
       'dependencies': [
         'wtf_config',
-        '<(chromium_src_dir)/third_party/icu/icu.gyp:icui18n',
-        '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
+        '<(chromium_src_dir)/build/linux/system.gyp:icu',
       ],
       'include_dirs': [
         '../',
@@ -157,8 +156,7 @@
       },
       'export_dependent_settings': [
         'wtf_config',
-        '<(chromium_src_dir)/third_party/icu/icu.gyp:icui18n',
-        '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
+        '<(chromium_src_dir)/build/linux/system.gyp:icu',
       ],
       'msvs_disabled_warnings': [4127, 4355, 4510, 4512, 4610, 4706],
       'conditions': [
--- a/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp
+++ b/third_party/WebKit/Source/WebKit/chromium/WebKit.gyp
@@ -607,7 +607,7 @@
                                 '<(chromium_src_dir)/build/temp_gyp/googleurl.gyp:googleurl',
                                 '<(chromium_src_dir)/testing/gtest.gyp:gtest',
                                 '<(chromium_src_dir)/testing/gmock.gyp:gmock',
-                                '<(chromium_src_dir)/third_party/icu/icu.gyp:*',
+                                '<(chromium_src_dir)/build/linux/system.gyp:icu',
                                 '<(chromium_src_dir)/third_party/libjpeg/libjpeg.gyp:libjpeg',
                                 '<(chromium_src_dir)/third_party/libpng/libpng.gyp:libpng',
                                 '<(chromium_src_dir)/third_party/libxml/libxml.gyp:libxml',
@@ -936,7 +936,7 @@
                 'copy_TestNetscapePlugIn',
                 'webkit',
                 '../../JavaScriptCore/JavaScriptCore.gyp/JavaScriptCore.gyp:wtf_config',
-                '<(chromium_src_dir)/third_party/icu/icu.gyp:icuuc',
+                '<(chromium_src_dir)/build/linux/system.gyp:icu',
                 '<(chromium_src_dir)/third_party/mesa/mesa.gyp:osmesa',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:blob',
                 '<(chromium_src_dir)/webkit/support/webkit_support.gyp:webkit_support',
--- a/ui/ui_base.gypi
+++ b/ui/ui_base.gypi
@@ -10,14 +10,12 @@
       'dependencies': [
         '../base/base.gyp:base',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         'ui_gfx',
       ],
       # Export these dependencies since text_elider.h includes ICU headers.
       'export_dependent_settings': [
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
       ],
       'sources': [
         'base/accessibility/accessibility_types.h',
--- a/ui/ui_gfx.gypi
+++ b/ui/ui_gfx.gypi
@@ -44,8 +44,7 @@
         '../base/base.gyp:base',
         '../base/base.gyp:base_i18n',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../third_party/libpng/libpng.gyp:libpng',
         '../third_party/zlib/zlib.gyp:zlib',
         'gfx_resources',
--- a/views/views.gyp
+++ b/views/views.gyp
@@ -66,8 +66,7 @@
         '../app/app.gyp:app_resources',
         '../base/base.gyp:base',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../ui/base/strings/ui_strings.gyp:ui_strings',
         '../ui/gfx/compositor/compositor.gyp:compositor',
         '../ui/ui.gyp:ui_base',
@@ -471,8 +470,7 @@
         '../skia/skia.gyp:skia',
         '../testing/gmock.gyp:gmock',
         '../testing/gtest.gyp:gtest',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../ui/base/strings/ui_strings.gyp:ui_strings',
         'views',
       ],
@@ -553,8 +551,7 @@
         '../app/app.gyp:app_resources',
         '../base/base.gyp:base',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         'views',
       ],
       'include_dirs': [
@@ -639,8 +636,7 @@
         '../app/app.gyp:app_resources',
         '../base/base.gyp:base',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icui18n',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         'views',
       ],
       'include_dirs': [
--- a/webkit/glue/webkit_glue.gypi
+++ b/webkit/glue/webkit_glue.gypi
@@ -118,8 +118,7 @@
         '<(DEPTH)/ppapi/ppapi_internal.gyp:ppapi_shared',
         '<(DEPTH)/printing/printing.gyp:printing',
         '<(DEPTH)/skia/skia.gyp:skia',
-        '<(DEPTH)/third_party/icu/icu.gyp:icui18n',
-        '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
+        '<(DEPTH)/build/linux/system.gyp:icu',
         '<(DEPTH)/third_party/npapi/npapi.gyp:npapi',
         'webkit_resources',
         'webkit_strings',
--- a/webkit/tools/test_shell/test_shell.gypi
+++ b/webkit/tools/test_shell/test_shell.gypi
@@ -559,7 +559,7 @@
           'mac_bundle': 1,
           'msvs_guid': '0D04AEC1-6B68-492C-BCCF-808DFD69ABC6',
           'dependencies': [
-            '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
+            '<(DEPTH)/build/linux/system.gyp:icu',
             'npapi_test_common',
           ],
           'sources': [
--- a/chrome/chrome_installer_util.gypi
+++ b/chrome/chrome_installer_util.gypi
@@ -103,8 +103,7 @@
             '<(DEPTH)/base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
             '<(DEPTH)/courgette/courgette.gyp:courgette_lib',
             '<(DEPTH)/third_party/bspatch/bspatch.gyp:bspatch',
-            '<(DEPTH)/third_party/icu/icu.gyp:icui18n',
-            '<(DEPTH)/third_party/icu/icu.gyp:icuuc',
+            '<(DEPTH)/build/linux/system.gyp:icu',
             '<(DEPTH)/third_party/lzma_sdk/lzma_sdk.gyp:lzma_sdk',
           ],
           'sources': [
--- a/ppapi/ppapi_shared.gypi
+++ b/ppapi/ppapi_shared.gypi
@@ -13,7 +13,7 @@
         '../base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
         '../build/temp_gyp/googleurl.gyp:googleurl',
         '../skia/skia.gyp:skia',
-        '../third_party/icu/icu.gyp:icuuc',
+        '../build/linux/system.gyp:icu',
         '../ui/gfx/surface/surface.gyp:surface',
       ],
       'include_dirs': [
--- a/third_party/libphonenumber/libphonenumber.gyp
+++ b/third_party/libphonenumber/libphonenumber.gyp
@@ -32,8 +32,7 @@
     'target_name': 'libphonenumber',
     'type': 'static_library',
     'dependencies': [
-      '../icu/icu.gyp:icui18n',
-      '../icu/icu.gyp:icuuc',
+      '../../build/linux/system.gyp:icu',
       '../protobuf/protobuf.gyp:protobuf_lite',
       '../../base/base.gyp:base',
       '../../base/third_party/dynamic_annotations/dynamic_annotations.gyp:dynamic_annotations',
@@ -118,8 +117,7 @@
       'cpp/src/test_metadata.cc',
     ],
     'dependencies': [
-      '../icu/icu.gyp:icui18n',
-      '../icu/icu.gyp:icuuc',
+      '../../build/linux/system.gyp:icu',
       '../protobuf/protobuf.gyp:protobuf_lite',
       '../../base/base.gyp:base',
       '../../base/base.gyp:test_support_base',
--- a/v8/src/extensions/experimental/experimental.gyp
+++ b/v8/src/extensions/experimental/experimental.gyp
@@ -56,7 +56,7 @@
         '../..',
       ],
       'dependencies': [
-        '<(icu_src_dir)/icu.gyp:*',
+        '../../../../build/linux/system.gyp:icu',
         'js2c_i18n#host',
         '../../../tools/gyp/v8.gyp:v8',
       ],
