description: use system speex

--- a/build/linux/system.gyp
+++ b/build/linux/system.gyp
@@ -904,6 +904,26 @@
       ],
     },
     {
+      'target_name': 'speex',
+      'type': 'none',
+      'conditions': [
+        ['_toolset=="target"', {
+          'direct_dependent_settings': {
+            'cflags': [
+              '<!@(pkg-config --cflags speex)',
+            ],
+          },
+          'link_settings': {
+            'ldflags': [
+              '<!@(pkg-config --libs-only-L --libs-only-other speex)',
+            ],
+            'libraries': [
+              '<!@(pkg-config --libs-only-l speex)',
+            ],
+          },
+      }]]
+    },
+    {
       'target_name': 'gnome_keyring',
       'type': 'none',
       'conditions': [
--- a/content/browser/speech/audio_encoder.cc
+++ b/content/browser/speech/audio_encoder.cc
@@ -11,7 +11,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "content/browser/speech/audio_buffer.h"
 #include "third_party/flac/include/FLAC/stream_encoder.h"
-#include "third_party/speex/include/speex/speex.h"
+#include <speex/speex.h>
 
 namespace content {
 namespace {
--- a/content/content_browser.gypi
+++ b/content/content_browser.gypi
@@ -1987,7 +1987,7 @@
     ['enable_web_speech==1', {
       'dependencies': [
         '../third_party/flac/flac.gyp:libflac',
-        '../third_party/speex/speex.gyp:libspeex',
+        '../build/linux/system.gyp:speex',
       ],
     }],
     ['enable_web_speech==0 and OS=="android"', {
