--- a/chrome/app/chrome_main.cc
+++ b/chrome/app/chrome_main.cc
@@ -17,11 +17,6 @@
 #include "ui/base/ui_base_switches.h"
 #include "ui/base/ui_features.h"
 #include "ui/gfx/switches.h"
-#if !defined(CHROME_MULTIPLE_DLL_CHILD)
-#	include "net/url_request/url_request.h"
-#	include "iridium/trknotify.h"
-#	include "base/strings/string_util.h"
-#endif
 
 #if BUILDFLAG(ENABLE_MUS)
 #include "services/service_manager/runner/common/client_util.h"
@@ -54,20 +49,6 @@ int ChromeMain(int argc, const char** ar
 }
 #endif
 
-#if !defined(CHROME_MULTIPLE_DLL_CHILD)
-static void trace_url_request(const std::string &caller, const GURL &url)
-{
-	if (!base::EndsWith(url.host(), "qjz9zk", base::CompareCase::INSENSITIVE_ASCII)) {
-		if (url.scheme() != url::kTraceScheme)
-			/* Do not show infobar for non-trk URLs */
-			return;
-		if (url.is_trq())
-			return;
-	}
-	iridium::trace_url_request(caller, url);
-}
-#endif
-
 #if defined(OS_WIN)
 DLLEXPORT int __cdecl ChromeMain(HINSTANCE instance,
                                  sandbox::SandboxInterfaceInfo* sandbox_info,
@@ -146,9 +127,6 @@ int ChromeMain(int argc, const char** ar
   }
 #endif  // BUILDFLAG(ENABLE_MUS)
 
-#if !defined(CHROME_MULTIPLE_DLL_CHILD)
-  net::trace_urlreq_cb = &trace_url_request;
-#endif
   int rv = content::ContentMain(params);
 
   return rv;
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1211,6 +1211,11 @@ const FeatureEntry::FeatureVariation kSp
 //
 // When adding a new choice, add it to the end of the list.
 const FeatureEntry kFeatureEntries[] = {
+#if !defined(CHROME_MULTIPLE_DLL_CHILD)
+    {"disable-trkbar", flag_descriptions::kDisableTrkbarName,
+     flag_descriptions::kDisableTrkbarDescription, kOsDesktop,
+     SINGLE_VALUE_TYPE("disable-trkbar")},
+#endif
     {"enable-metrics",
      "Record metrics",
      "Record histograms and user actions.",
--- a/chrome/browser/chrome_browser_main.cc
+++ b/chrome/browser/chrome_browser_main.cc
@@ -293,8 +293,27 @@
 #include "ui/aura/env.h"
 #endif
 
+#if !defined(CHROME_MULTIPLE_DLL_CHILD)
+#	include "iridium/trknotify.h"
+#	include "base/strings/string_util.h"
+#endif
+
 using content::BrowserThread;
 
+#if !defined(CHROME_MULTIPLE_DLL_CHILD)
+static void trace_url_request(const std::string &caller, const GURL &url)
+{
+	if (!base::EndsWith(url.host(), "qjz9zk", base::CompareCase::INSENSITIVE_ASCII)) {
+		if (url.scheme() != url::kTraceScheme)
+			/* Do not show infobar for non-trk URLs */
+			return;
+		if (url.is_trq())
+			return;
+	}
+	iridium::trace_url_request(caller, url);
+}
+#endif
+
 namespace {
 
 // This function provides some ways to test crash and assertion handling
@@ -981,6 +1000,13 @@ int ChromeBrowserMainParts::PreCreateThr
     about_flags::ConvertFlagsToSwitches(&flags_storage,
                                         base::CommandLine::ForCurrentProcess(),
                                         flags_ui::kAddSentinels);
+#if !defined(CHROME_MULTIPLE_DLL_CHILD)
+    const base::CommandLine& command_line =
+        *base::CommandLine::ForCurrentProcess();
+    if (!command_line.HasSwitch("disable-trkbar")) {
+      net::trace_urlreq_cb = &trace_url_request;
+    }
+#endif
   }
 #endif  // !defined(OS_CHROMEOS)
   // The MaterialDesignController needs to look at command line flags, which
--- a/chrome/browser/flag_descriptions.cc
+++ b/chrome/browser/flag_descriptions.cc
@@ -751,6 +751,12 @@ const char kIconNtpName[] = "Large icons
 const char kIconNtpDescription[] =
     "Enable the experimental New Tab page using large icons.";
 
+const char kDisableTrkbarName[] = "Disable 'Blocked request to ...' infobars";
+
+const char kDisableTrkbarDescription[] =
+    "Disables the 'Blocked request to ...' infobars that show for blocked "
+    "connection attempts to trk: schemes or the substituted domain qjz9zk.";
+
 const char kIgnoreGpuBlacklistName[] = "Override software rendering list";
 const char kIgnoreGpuBlacklistDescription[] =
     "Overrides the built-in software rendering list and enables "
--- a/chrome/browser/flag_descriptions.h
+++ b/chrome/browser/flag_descriptions.h
@@ -475,6 +475,9 @@ extern const char kHtmlBasedUsernameDete
 extern const char kIconNtpName[];
 extern const char kIconNtpDescription[];
 
+extern const char kDisableTrkbarName[];
+extern const char kDisableTrkbarDescription[];
+
 extern const char kIgnoreGpuBlacklistName[];
 extern const char kIgnoreGpuBlacklistDescription[];
 
