--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -2073,11 +2073,11 @@
     {"enable-wifi-credential-sync", flag_descriptions::kWifiCredentialSyncName,
      flag_descriptions::kWifiCredentialSyncDescription, kOsCrOS,
      SINGLE_VALUE_TYPE(switches::kEnableWifiCredentialSync)},
+#endif  // OS_CHROMEOS
     {"enable-potentially-annoying-security-features",
      flag_descriptions::kExperimentalSecurityFeaturesName,
      flag_descriptions::kExperimentalSecurityFeaturesDescription, kOsAll,
      SINGLE_VALUE_TYPE(switches::kEnablePotentiallyAnnoyingSecurityFeatures)},
-#endif  // OS_CHROMEOS
     {"mark-non-secure-as", flag_descriptions::kMarkHttpAsName,
      flag_descriptions::kMarkHttpAsDescription, kOsAll,
      MULTI_VALUE_TYPE(kMarkHttpAsChoices)},
