# Add flag to disable automatic search engine collection

--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -1238,6 +1238,10 @@ const FeatureEntry kFeatureEntries[] = {
      "Enable stacking in tab strip",
      "Forces tabs to be stacked in the tab strip. Otherwise, they follow default behavior.",
      kOsAll, SINGLE_VALUE_TYPE("enable-stacked-tab-strip")},
+    {"disable-search-engine-collection",
+     "Disable search engine collection",
+     "Prevents search engines from being added automatically.",
+     kOsAll, SINGLE_VALUE_TYPE("disable-search-engine-collection")},
     {"enable-metrics",
      "Record metrics",
      "Record histograms and user actions.",
--- a/chrome/browser/ui/search_engines/search_engine_tab_helper.cc
+++ b/chrome/browser/ui/search_engines/search_engine_tab_helper.cc
@@ -4,6 +4,7 @@
 
 #include "chrome/browser/ui/search_engines/search_engine_tab_helper.h"
 
+#include "base/command_line.h"
 #include "base/memory/ptr_util.h"
 #include "base/metrics/histogram_macros.h"
 #include "chrome/browser/profiles/profile.h"
@@ -105,6 +106,13 @@ void SearchEngineTabHelper::PageHasOpenS
       web_contents()->GetMainFrame())
     return;
 
+  // Abort now if search engines shouldn't be added automatically
+  const base::CommandLine& command_line =
+      *base::CommandLine::ForCurrentProcess();
+  if (command_line.HasSwitch("disable-search-engine-collection")) {
+    return;
+  }
+
   // Make sure that the page is the current page and other basic checks.
   // When |page_url| has file: scheme, this method doesn't work because of
   // http://b/issue?id=863583. For that reason, this doesn't check and allow
