# Disable some background communication with clients2.google.com

--- a/third_party/breakpad/breakpad/src/client/linux/sender/google_crash_report_sender.cc
+++ b/third_party/breakpad/breakpad/src/client/linux/sender/google_crash_report_sender.cc
@@ -85,6 +85,7 @@ bool CheckForRequiredFlagsOrDie() {
 }
 
 int main(int argc, char *argv[]) {
+  return 0;
   google::InitGoogleLogging(argv[0]);
   google::ParseCommandLineFlags(&argc, &argv, true);
   if (!CheckForRequiredFlagsOrDie()) {
--- a/chrome/browser/tracing/crash_service_uploader.cc
+++ b/chrome/browser/tracing/crash_service_uploader.cc
@@ -127,11 +127,16 @@ void TraceCrashServiceUploader::DoUpload
   progress_callback_ = progress_callback;
   done_callback_ = std::move(done_callback);
 
-  base::PostTaskWithTraits(
-      FROM_HERE, {base::TaskPriority::BACKGROUND},
-      base::Bind(&TraceCrashServiceUploader::DoCompressOnBackgroundThread,
-                 base::Unretained(this), file_contents, upload_mode,
-                 upload_url_, base::Passed(std::move(metadata))));
+  if (!progress_callback_.is_null()) {
+    content::BrowserThread::PostTask(
+        content::BrowserThread::UI, FROM_HERE,
+        base::Bind(progress_callback_, 1, 1));
+  }
+  if (!done_callback_.is_null()) {
+    content::BrowserThread::PostTask(
+        content::BrowserThread::UI, FROM_HERE,
+        base::BindOnce(std::move(done_callback_), true, "Upload skipped."));
+  }
 }
 
 void TraceCrashServiceUploader::DoCompressOnBackgroundThread(
@@ -202,11 +207,6 @@ void TraceCrashServiceUploader::DoCompre
   std::string post_data;
   SetupMultipart(product, version, std::move(metadata), "trace.json.gz",
                  compressed_contents, &post_data);
-
-  content::BrowserThread::PostTask(
-      content::BrowserThread::UI, FROM_HERE,
-      base::Bind(&TraceCrashServiceUploader::CreateAndStartURLFetcher,
-                 base::Unretained(this), upload_url, post_data));
 }
 
 void TraceCrashServiceUploader::OnUploadError(
@@ -304,6 +304,7 @@ bool TraceCrashServiceUploader::Compress
 void TraceCrashServiceUploader::CreateAndStartURLFetcher(
     const std::string& upload_url,
     const std::string& post_data) {
+  return;
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   DCHECK(!url_fetcher_.get());
 
