# Disable Google Cloud Messaging (GCM) client

--- a/components/gcm_driver/gcm_client_impl.cc
+++ b/components/gcm_driver/gcm_client_impl.cc
@@ -502,6 +502,7 @@ void GCMClientImpl::StartGCM() {
 }
 
 void GCMClientImpl::InitializeMCSClient() {
+  return;
   std::vector<GURL> endpoints;
   endpoints.push_back(gservices_settings_.GetMCSMainEndpoint());
   GURL fallback_endpoint = gservices_settings_.GetMCSFallbackEndpoint();
@@ -699,6 +700,7 @@ void GCMClientImpl::RemoveHeartbeatInter
 }
 
 void GCMClientImpl::StartCheckin() {
+  return;
   // Make sure no checkin is in progress.
   if (checkin_request_)
     return;
@@ -777,6 +779,7 @@ void GCMClientImpl::SetGServicesSettings
 }
 
 void GCMClientImpl::SchedulePeriodicCheckin() {
+  return;
   // Make sure no checkin is in progress.
   if (checkin_request_.get() || !device_checkin_info_.accounts_set)
     return;
