# Disables file download quarantining

--- a/content/browser/renderer_host/pepper/pepper_file_io_host.cc
+++ b/content/browser/renderer_host/pepper/pepper_file_io_host.cc
@@ -21,7 +21,6 @@
 #include "content/public/browser/render_process_host.h"
 #include "content/public/browser/storage_partition.h"
 #include "content/public/common/content_client.h"
-#include "content/public/common/quarantine.h"
 #include "ipc/ipc_platform_file.h"
 #include "ppapi/c/pp_errors.h"
 #include "ppapi/c/ppb_file_io.h"
@@ -447,7 +446,7 @@
     ppapi::host::ReplyMessageContext reply_context,
     const base::FilePath& path,
     base::File::Error error_code) {
-#if defined(OS_WIN) || defined(OS_LINUX)
+#if 0
   // Quarantining a file before its contents are available is only supported on
   // Windows and Linux.
   if (!FileOpenForWrite(open_flags_) || error_code != base::File::FILE_OK) {
@@ -467,7 +466,7 @@
 #endif
 }
 
-#if defined(OS_WIN) || defined(OS_LINUX)
+#if 0
 void PepperFileIOHost::OnLocalFileQuarantined(
     ppapi::host::ReplyMessageContext reply_context,
     const base::FilePath& path,
--- a/content/browser/renderer_host/pepper/pepper_file_io_host.h
+++ b/content/browser/renderer_host/pepper/pepper_file_io_host.h
@@ -16,7 +16,6 @@
 #include "base/memory/ref_counted.h"
 #include "base/memory/weak_ptr.h"
 #include "content/browser/renderer_host/pepper/browser_ppapi_host_impl.h"
-#include "content/public/common/quarantine.h"
 #include "ipc/ipc_listener.h"
 #include "ipc/ipc_platform_file.h"
 #include "ppapi/c/pp_file_info.h"
@@ -92,10 +91,6 @@
                          const base::FilePath& path,
                          base::File::Error error_code);
 
-  void OnLocalFileQuarantined(ppapi::host::ReplyMessageContext reply_context,
-                              const base::FilePath& path,
-                              QuarantineFileResult quarantine_result);
-
   void SendFileOpenReply(ppapi::host::ReplyMessageContext reply_context,
                          base::File::Error error_code);
 
--- a/content/common/BUILD.gn
+++ b/content/common/BUILD.gn
@@ -259,11 +259,6 @@
     "presentation/presentation_struct_traits.cc",
     "presentation/presentation_struct_traits.h",
     "process_type.cc",
-    "quarantine/quarantine.cc",
-    "quarantine/quarantine_constants_linux.h",
-    "quarantine/quarantine_linux.cc",
-    "quarantine/quarantine_mac.mm",
-    "quarantine/quarantine_win.cc",
     "render_widget_surface_properties.cc",
     "render_widget_surface_properties.h",
     "resize_params.cc",
--- a/content/common/quarantine/quarantine.cc
+++ b/content/common/quarantine/quarantine.cc
@@ -6,8 +6,6 @@
 
 #include "build/build_config.h"
 
-#if !defined(OS_WIN) && !defined(OS_MACOSX) && !defined(OS_LINUX)
-
 namespace content {
 
 QuarantineFileResult QuarantineFile(const base::FilePath& file,
@@ -24,5 +22,3 @@
 }
 
 }  // namespace content
-
-#endif  // !WIN && !MAC && !LINUX
--- a/content/public/common/BUILD.gn
+++ b/content/public/common/BUILD.gn
@@ -195,7 +195,6 @@
     "process_type.h",
     "push_event_payload.h",
     "push_subscription_options.h",
-    "quarantine.h",
     "referrer.cc",
     "referrer.h",
     "renderer_preferences.cc",
--- a/content/browser/download/base_file.cc
+++ b/content/browser/download/base_file.cc
@@ -21,7 +21,6 @@
 #include "content/browser/download/download_stats.h"
 #include "content/public/browser/content_browser_client.h"
 #include "content/public/browser/download_item.h"
-#include "content/public/common/quarantine.h"
 #include "crypto/secure_hash.h"
 #include "net/base/net_errors.h"
 
@@ -437,7 +436,7 @@
   return reason;
 }
 
-#if defined(OS_WIN) || defined(OS_MACOSX) || defined(OS_LINUX)
+#if 0
 
 namespace {
 
@@ -521,7 +520,7 @@
   }
   return DOWNLOAD_INTERRUPT_REASON_FILE_FAILED;
 }
-#else  // !OS_WIN && !OS_MACOSX && !OS_LINUX
+#else  // 1
 DownloadInterruptReason BaseFile::AnnotateWithSourceInformation(
     const std::string& client_guid,
     const GURL& source_url,
