# Disable some background communication with clients2.google.com

--- a/breakpad/src/client/linux/sender/google_crash_report_sender.cc
+++ b/breakpad/src/client/linux/sender/google_crash_report_sender.cc
@@ -84,6 +84,7 @@ bool CheckForRequiredFlagsOrDie() {
 }
 
 int main(int argc, char *argv[]) {
+  return 0;
   google::InitGoogleLogging(argv[0]);
   google::ParseCommandLineFlags(&argc, &argv, true);
   if (!CheckForRequiredFlagsOrDie()) {
--- a/chrome/browser/tracing/crash_service_uploader.cc
+++ b/chrome/browser/tracing/crash_service_uploader.cc
@@ -135,6 +135,7 @@ void TraceCrashServiceUploader::DoUpload
     std::unique_ptr<const base::DictionaryValue> metadata,
     const UploadProgressCallback& progress_callback,
     const UploadDoneCallback& done_callback) {
+  return;
   DCHECK_CURRENTLY_ON(content::BrowserThread::FILE);
   DCHECK(!url_fetcher_.get());
 
@@ -302,6 +303,7 @@ bool TraceCrashServiceUploader::Compress
 void TraceCrashServiceUploader::CreateAndStartURLFetcher(
     const std::string& upload_url,
     const std::string& post_data) {
+  return;
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   DCHECK(!url_fetcher_.get());
 
