# Disables file download quarantining

--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -545,12 +545,6 @@ source_set("browser") {
     "download/drag_download_util.h",
     "download/mhtml_generation_manager.cc",
     "download/mhtml_generation_manager.h",
-    "download/quarantine.cc",
-    "download/quarantine.h",
-    "download/quarantine_constants_linux.h",
-    "download/quarantine_linux.cc",
-    "download/quarantine_mac.mm",
-    "download/quarantine_win.cc",
     "download/rate_estimator.cc",
     "download/rate_estimator.h",
     "download/save_file.cc",
--- a/content/browser/download/quarantine.cc
+++ b/content/browser/download/quarantine.cc
@@ -6,8 +6,6 @@
 
 #include "build/build_config.h"
 
-#if !defined(OS_WIN) && !defined(OS_MACOSX) && !defined(OS_LINUX)
-
 namespace content {
 
 QuarantineFileResult QuarantineFile(const base::FilePath& file,
@@ -18,5 +16,3 @@ QuarantineFileResult QuarantineFile(cons
 }
 
 } // namespace content
-
-#endif // !WIN && !MAC && !LINUX
--- a/content/browser/download/base_file.cc
+++ b/content/browser/download/base_file.cc
@@ -18,7 +18,6 @@
 #include "content/browser/download/download_interrupt_reasons_impl.h"
 #include "content/browser/download/download_net_log_parameters.h"
 #include "content/browser/download/download_stats.h"
-#include "content/browser/download/quarantine.h"
 #include "content/public/browser/browser_thread.h"
 #include "content/public/browser/content_browser_client.h"
 #include "crypto/secure_hash.h"
@@ -373,7 +372,7 @@ DownloadInterruptReason BaseFile::LogInt
   return reason;
 }
 
-#if defined(OS_WIN) || defined(OS_MACOSX) || defined(OS_LINUX)
+#if 0
 
 namespace {
 
@@ -456,7 +455,7 @@ DownloadInterruptReason BaseFile::Annota
   }
   return DOWNLOAD_INTERRUPT_REASON_FILE_FAILED;
 }
-#else  // !OS_WIN && !OS_MACOSX && !OS_LINUX
+#else  // 1
 DownloadInterruptReason BaseFile::AnnotateWithSourceInformation(
     const std::string& client_guid,
     const GURL& source_url,
