# Add more URL schemes allowed for saving

--- a/content/public/common/url_utils.cc
+++ b/content/public/common/url_utils.cc
@@ -18,11 +18,7 @@ bool HasWebUIScheme(const GURL& url) {
 }
 
 bool IsSavableURL(const GURL& url) {
-  for (auto& scheme : GetSavableSchemes()) {
-    if (url.SchemeIs(scheme))
-      return true;
-  }
-  return false;
+  return true;
 }
 
 // PlzNavigate
--- a/chrome/browser/ui/browser_commands.cc
+++ b/chrome/browser/ui/browser_commands.cc
@@ -343,12 +343,6 @@ int GetContentRestrictions(const Browser
     CoreTabHelper* core_tab_helper =
         CoreTabHelper::FromWebContents(current_tab);
     content_restrictions = core_tab_helper->content_restrictions();
-    NavigationEntry* last_committed_entry =
-        current_tab->GetController().GetLastCommittedEntry();
-    if (!content::IsSavableURL(
-            last_committed_entry ? last_committed_entry->GetURL() : GURL()) ||
-        current_tab->ShowingInterstitialPage())
-      content_restrictions |= CONTENT_RESTRICTION_SAVE;
     if (current_tab->ShowingInterstitialPage())
       content_restrictions |= CONTENT_RESTRICTION_PRINT;
   }
@@ -899,8 +893,7 @@ bool CanSavePage(const Browser* browser)
       prefs::kAllowFileSelectionDialogs)) {
     return false;
   }
-  return !browser->is_devtools() &&
-      !(GetContentRestrictions(browser) & CONTENT_RESTRICTION_SAVE);
+  return true;
 }
 
 void ShowFindBar(Browser* browser) {
--- a/components/offline_pages/core/offline_page_model.cc
+++ b/components/offline_pages/core/offline_page_model.cc
@@ -29,7 +29,7 @@ OfflinePageModel::DeletedPageInfo::Delet
 
 // static
 bool OfflinePageModel::CanSaveURL(const GURL& url) {
-  return url.is_valid() && url.SchemeIsHTTPOrHTTPS();
+  return url.is_valid();
 }
 
 OfflinePageModel::OfflinePageModel() = default;
--- a/content/common/url_schemes.cc
+++ b/content/common/url_schemes.cc
@@ -25,6 +25,10 @@ std::vector<GURL>* secure_origins = null
 std::vector<std::string>* service_worker_schemes = nullptr;
 
 const char* const kDefaultSavableSchemes[] = {
+  url::kAboutScheme,
+  url::kBlobScheme,
+  url::kContentScheme,
+  url::kGopherScheme,
   url::kHttpScheme,
   url::kHttpsScheme,
   url::kFileScheme,
@@ -32,7 +36,13 @@ const char* const kDefaultSavableSchemes
   url::kFtpScheme,
   kChromeDevToolsScheme,
   kChromeUIScheme,
-  url::kDataScheme
+  url::kDataScheme,
+  url::kJavaScriptScheme,
+  url::kMailToScheme,
+  url::kWsScheme,
+  url::kWssScheme,
+  kGuestScheme,
+  kViewSourceScheme
 };
 
 }  // namespace
