# Fix building with enable_webrtc=false

--- a/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc
+++ b/chrome/browser/extensions/api/webrtc_audio_private/webrtc_audio_private_api.cc
@@ -315,7 +315,9 @@ bool WebrtcAudioPrivateSetAudioExperimen
       return false;
     }
 
+#if BUILDFLAG(ENABLE_WEBRTC)
     host->SetEchoCanceller3(*params->audio_experiments.enable_aec3);
+#endif
   }
 
   SendResponse(true);
