# Fix building without one click signin

--- a/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
+++ b/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
@@ -424,6 +424,7 @@ void OneClickSigninSyncStarter::CancelSi
 }
 
 void OneClickSigninSyncStarter::ConfirmAndSignin() {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   SigninManager* signin = SigninManagerFactory::GetForProfile(profile_);
   if (confirmation_required_ == CONFIRM_UNTRUSTED_SIGNIN) {
     browser_ = EnsureBrowser(browser_, profile_);
@@ -438,6 +439,7 @@ void OneClickSigninSyncStarter::ConfirmA
     // No confirmation required - just sign in the user.
     signin->CompletePendingSignin();
   }
+#endif
 }
 
 void OneClickSigninSyncStarter::UntrustedSigninConfirmed(
--- a/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
+++ b/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
@@ -17,13 +17,7 @@ OneClickSigninLinksDelegateImpl::OneClic
 OneClickSigninLinksDelegateImpl::~OneClickSigninLinksDelegateImpl() {}
 
 void OneClickSigninLinksDelegateImpl::OnLearnMoreLinkClicked(bool is_dialog) {
-  chrome::NavigateParams params(browser_, GURL(chrome::kChromeSyncLearnMoreURL),
-                                ui::PAGE_TRANSITION_LINK);
-  params.disposition = is_dialog ? WindowOpenDisposition::NEW_WINDOW
-                                 : WindowOpenDisposition::NEW_FOREGROUND_TAB;
-  chrome::Navigate(&params);
 }
 
 void OneClickSigninLinksDelegateImpl::OnAdvancedLinkClicked() {
-  chrome::ShowSettingsSubPage(browser_, chrome::kSyncSetupSubPage);
 }
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -1613,8 +1613,6 @@ split_static_library("ui") {
         "views/profiles/profile_chooser_view.cc",
         "views/profiles/profile_chooser_view.h",
         "views/screen_capture_notification_ui_views.cc",
-        "views/sync/one_click_signin_dialog_view.cc",
-        "views/sync/one_click_signin_dialog_view.h",
       ]
     }
 
@@ -2877,10 +2875,6 @@ split_static_library("ui") {
         "cocoa/omnibox/omnibox_view_mac.mm",
         "cocoa/omnibox_decoration_bubble_controller.h",
         "cocoa/omnibox_decoration_bubble_controller.mm",
-        "cocoa/one_click_signin_dialog_controller.h",
-        "cocoa/one_click_signin_dialog_controller.mm",
-        "cocoa/one_click_signin_view_controller.h",
-        "cocoa/one_click_signin_view_controller.mm",
         "cocoa/page_info/page_info_bubble_controller.h",
         "cocoa/page_info/page_info_bubble_controller.mm",
         "cocoa/page_info/page_info_utils_cocoa.h",
