# Modify other miscellaneous hardcoded settings

--- a/build/config/features.gni
+++ b/build/config/features.gni
@@ -73,7 +73,6 @@ declare_args() {
   # TODO(bshe): Enable for other architecture too. Currently we only support arm
   # and arm64.
   enable_webvr = is_android && (current_cpu == "arm" || current_cpu == "arm64")
-}
 
 # Additional dependent variables -----------------------------------------------
 
@@ -86,3 +85,4 @@ enable_chromevox_next = false
 # =============================================
 #
 # See comment at the top.
+}
