# Modify other miscellaneous hardcoded settings

--- a/build/config/features.gni
+++ b/build/config/features.gni
@@ -83,7 +83,6 @@ declare_args() {
   # TODO(bshe): Enable for other architecture too. Currently we only support arm
   # and arm64.
   enable_webvr = is_android && (current_cpu == "arm" || current_cpu == "arm64")
-}
 
 # Additional dependent variables -----------------------------------------------
 
@@ -123,3 +122,4 @@ enable_mac_keystone = is_mac && is_chrom
 # =============================================
 #
 # See comment at the top.
+}
