# Make popups go to tabs instead

--- a/content/renderer/render_view_impl.cc
+++ b/content/renderer/render_view_impl.cc
@@ -306,7 +306,7 @@
     case blink::kWebNavigationPolicyNewWindow:
       return WindowOpenDisposition::NEW_WINDOW;
     case blink::kWebNavigationPolicyNewPopup:
-      return WindowOpenDisposition::NEW_POPUP;
+      return WindowOpenDisposition::NEW_FOREGROUND_TAB;
   default:
     NOTREACHED() << "Unexpected WebNavigationPolicy";
     return WindowOpenDisposition::IGNORE_ACTION;
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -1399,7 +1399,7 @@
 
 WindowOpenDisposition BrowserView::GetDispositionForPopupBounds(
     const gfx::Rect& bounds) {
-  return WindowOpenDisposition::NEW_POPUP;
+  return WindowOpenDisposition::NEW_FOREGROUND_TAB;
 }
 
 FindBar* BrowserView::CreateFindBar() {
