# Build with gcc 4.8

--- a/build/toolchain/linux/BUILD.gn
+++ b/build/toolchain/linux/BUILD.gn
@@ -48,8 +48,8 @@ gcc_toolchain("arm64") {
 gcc_toolchain("arm") {
   toolprefix = "arm-linux-gnueabihf-"
 
-  cc = "${toolprefix}gcc"
-  cxx = "${toolprefix}g++"
+  cc = "${toolprefix}gcc-4.8"
+  cxx = "${toolprefix}g++-4.8"
 
   ar = "${toolprefix}ar"
   ld = cxx
@@ -91,8 +91,8 @@ clang_toolchain("clang_x86_v8_mipsel") {
 }
 
 gcc_toolchain("x86") {
-  cc = "gcc"
-  cxx = "g++"
+  cc = "gcc-4.8"
+  cxx = "g++-4.8"
 
   readelf = "readelf"
   nm = "nm"
@@ -134,8 +134,8 @@ clang_toolchain("clang_x64_v8_mips64el")
 }
 
 gcc_toolchain("x64") {
-  cc = "gcc"
-  cxx = "g++"
+  cc = "gcc-4.8"
+  cxx = "g++-4.8"
 
   readelf = "readelf"
   nm = "nm"
