# From Ubuntu's patch of the same name

--- a/build/config/allocator.gni
+++ b/build/config/allocator.gni
@@ -10,7 +10,7 @@ if (is_android || current_cpu == "mipsel
     (is_linux && target_cpu == "arm64")) {
   _default_allocator = "none"
 } else {
-  _default_allocator = "tcmalloc"
+  _default_allocator = "none"
 }
 
 # The debug CRT on Windows has some debug features that are incompatible with
@@ -18,7 +18,7 @@ if (is_android || current_cpu == "mipsel
 # against the debug CRT with "is_nacl=false".
 if ((is_linux || is_android || (is_win && !is_component_build && !is_debug)) &&
      !is_asan && !is_lsan && !is_tsan && !is_msan) {
-  _default_use_experimental_allocator_shim = true
+  _default_use_experimental_allocator_shim = false
 } else {
   _default_use_experimental_allocator_shim = false
 }
