# Fix clang 3.7 compiling flag warnings and errors

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -409,10 +409,6 @@ config("compiler") {
   # clang-cl (used if is_win) doesn't expose this flag.
   # Currently disabled for nacl since its toolchain lacks this flag (too old).
   # TODO(zforman): Once nacl's toolchain is updated, remove check.
-  if (is_clang && is_linux) {
-    absolute_path = rebase_path("//.")
-    cflags += [ "-fdebug-prefix-map=$absolute_path=." ]
-  }
 
   # C++11 compiler flags setup.
   # ---------------------------
@@ -461,11 +457,9 @@ config("compiler") {
       # For Full LTO, it provides a measurable runtime speedup of Chrome.
       cflags += [
         "-flto",
-        "-fwhole-program-vtables",
       ]
       ldflags += [
         "-flto",
-        "-fwhole-program-vtables",
       ]
 
       # Apply a lower LTO optimization level as the default is too slow.
@@ -1066,18 +1060,6 @@ config("default_warnings") {
       ]
     }
     if (!is_nacl && !use_xcode_clang) {
-      # Flags NaCl (Clang 3.7) and Xcode 7.3 (Clang clang-703.0.31) do not
-      # recognize.
-      cflags += [
-        # TODO(thakis): https://crbug.com/604888
-        "-Wno-undefined-var-template",
-
-        # TODO(thakis): https://crbug.com/617318
-        "-Wno-nonportable-include-path",
-
-        # TODO(hans): https://crbug.com/637306
-        "-Wno-address-of-packed-member",
-      ]
       if (llvm_force_head_revision) {
         cflags += [
           # TODO(hans): https://crbug.com/681136
