# Fix building with gcc 5

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -461,11 +461,9 @@ config("compiler") {
       # For Full LTO, it provides a measurable runtime speedup of Chrome.
       cflags += [
         "-flto",
-        "-fwhole-program-vtables",
       ]
       ldflags += [
         "-flto",
-        "-fwhole-program-vtables",
       ]
 
       # Apply a lower LTO optimization level as the default is too slow.
--- a/build/config/sanitizers/BUILD.gn
+++ b/build/config/sanitizers/BUILD.gn
@@ -225,7 +225,6 @@ config("default_sanitizer_ldflags") {
     }
 
     if (is_cfi && !is_nacl) {
-      ldflags += [ "-fsanitize=cfi-vcall" ]
       if (use_cfi_cast) {
         ldflags += [
           "-fsanitize=cfi-derived-cast",
@@ -327,7 +326,6 @@ config("asan_flags") {
         asan_blacklist_path =
             rebase_path("//tools/memory/asan/blacklist.txt", root_build_dir)
       }
-      cflags += [ "-fsanitize-blacklist=$asan_blacklist_path" ]
     }
   }
 }
@@ -361,15 +359,6 @@ config("link_shared_library") {
 config("cfi_flags") {
   cflags = []
   if (is_cfi && !is_nacl) {
-    if (!defined(cfi_blacklist_path)) {
-      cfi_blacklist_path =
-          rebase_path("//tools/cfi/blacklist.txt", root_build_dir)
-    }
-    cflags += [
-      "-fsanitize=cfi-vcall",
-      "-fsanitize-blacklist=$cfi_blacklist_path",
-    ]
-
     if (use_cfi_cast) {
       cflags += [
         "-fsanitize=cfi-derived-cast",
@@ -428,7 +417,6 @@ config("msan_flags") {
     cflags = [
       "-fsanitize=memory",
       "-fsanitize-memory-track-origins=$msan_track_origins",
-      "-fsanitize-blacklist=$msan_blacklist_path",
     ]
   }
 }
@@ -442,7 +430,6 @@ config("tsan_flags") {
     }
     cflags = [
       "-fsanitize=thread",
-      "-fsanitize-blacklist=$tsan_blacklist_path",
     ]
   }
 }
@@ -468,7 +455,6 @@ config("ubsan_flags") {
       "-fsanitize=signed-integer-overflow",
       "-fsanitize=unreachable",
       "-fsanitize=vla-bound",
-      "-fsanitize-blacklist=$ubsan_blacklist_path",
     ]
 
     # Chromecast ubsan builds fail to compile with these
@@ -504,7 +490,6 @@ config("ubsan_security_flags") {
     }
     cflags = [
       "-fsanitize=signed-integer-overflow,shift,vptr,function,vla-bound",
-      "-fsanitize-blacklist=$ubsan_security_blacklist_path",
     ]
   }
 }
@@ -523,7 +508,6 @@ config("ubsan_vptr_flags") {
     }
     cflags = [
       "-fsanitize=vptr",
-      "-fsanitize-blacklist=$ubsan_vptr_blacklist_path",
     ]
   }
 }
