description: disable build commands for embedded fontconfig
author: Michael Gilbert <mgilbert@debian.org>

--- a/build/config/posix/BUILD.gn
+++ b/build/config/posix/BUILD.gn
@@ -2,7 +2,6 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//buildtools/deps_revisions.gni")
 import("//build/config/c++/c++.gni")
 import("//build/config/clang/clang.gni")
 import("//build/config/sanitizers/sanitizers.gni")
--- a/base/test/BUILD.gn
+++ b/base/test/BUILD.gn
@@ -187,13 +187,6 @@ static_library("test_support") {
 
   if (is_linux) {
     public_deps += [ ":fontconfig_util_linux" ]
-    data_deps = [
-      "//third_party/test_fonts",
-    ]
-    if (current_toolchain == host_toolchain) {
-      data_deps += [ ":do_generate_fontconfig_caches" ]
-      data += [ "$root_out_dir/fontconfig_caches/" ]
-    }
   }
 
   if (is_ios) {
@@ -340,7 +333,7 @@ if (is_linux) {
     ]
   }
 
-  if (current_toolchain == host_toolchain) {
+  if (false) {
     executable("generate_fontconfig_caches") {
       testonly = true
       sources = [
--- a/content/shell/test_runner/BUILD.gn
+++ b/content/shell/test_runner/BUILD.gn
@@ -203,7 +203,7 @@ group("resources") {
     deps += [ ":copy_android_fonts_config" ]
     data_deps += [ ":copy_android_fonts_config" ]
   }
-  if (is_android || is_linux || is_fuchsia) {
+  if (false) {
     deps += [ "//third_party/test_fonts" ]
     data_deps += [ "//third_party/test_fonts" ]
   }
