description: add missing include needed to build with gtk2
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/ui/libgtkui/gtk_ui.cc
+++ b/chrome/browser/ui/libgtkui/gtk_ui.cc
@@ -67,6 +67,7 @@
 
 #if GTK_MAJOR_VERSION == 2
 #include "chrome/browser/ui/libgtkui/native_theme_gtk2.h"  // nogncheck
+#include "chrome/browser/ui/libgtkui/nav_button_layout_manager.h"
 #elif GTK_MAJOR_VERSION == 3
 #include "chrome/browser/ui/libgtkui/native_theme_gtk3.h"  // nogncheck
 #include "chrome/browser/ui/libgtkui/nav_button_layout_manager_gtk3.h"  // nogncheck
