description: use system lcms2
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/879153

--- a/third_party/pdfium/BUILD.gn
+++ b/third_party/pdfium/BUILD.gn
@@ -193,6 +193,7 @@ static_library("pdfium") {
     ":fxge",
     ":fxjs",
     ":pwl",
+    "third_party:lcms2",
     "third_party:bigint",
     "third_party:pdfium_base",
   ]
@@ -686,6 +687,7 @@ static_library("fpdftext") {
   configs += [ ":pdfium_core_config" ]
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 }
 
@@ -1082,6 +1084,7 @@ static_library("fxge") {
 
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 
   defines = [ "DEFINE_PS_TABLES" ]
