description: use system lcms2
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/879153

--- a/third_party/pdfium/BUILD.gn
+++ b/third_party/pdfium/BUILD.gn
@@ -191,6 +191,7 @@ static_library("pdfium") {
     ":fxge",
     ":javascript",
     ":pwl",
+    "third_party:lcms2",
     "third_party:bigint",
     "third_party:pdfium_base",
   ]
@@ -672,6 +673,7 @@ static_library("fpdfapi") {
   configs += [ ":pdfium_core_config" ]
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 }
 
@@ -1087,6 +1089,7 @@ static_library("fxge") {
 
   deps = [
     ":fxcrt",
+    "third_party:lcms2",
   ]
 
   defines = [ "DEFINE_PS_TABLES" ]
