description: use system nspr library
author: Michael Gilbert <mgilbert@debian.org>

--- a/base/time/pr_time_unittest.cc
+++ b/base/time/pr_time_unittest.cc
@@ -7,7 +7,7 @@
 
 #include "base/compiler_specific.h"
 #include "base/macros.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time/time.h"
 #include "build/build_config.h"
 #include "testing/gtest/include/gtest/gtest.h"
--- a/base/time/time.cc
+++ b/base/time/time.cc
@@ -14,7 +14,7 @@
 #include "base/macros.h"
 #include "base/no_destructor.h"
 #include "base/strings/stringprintf.h"
-#include "base/third_party/nspr/prtime.h"
+#include <nspr/prtime.h>
 #include "base/time/time_override.h"
 #include "build/build_config.h"
 
--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -605,7 +605,6 @@ def write_gn_ninja(path, root_gen_dir, o
       'base/third_party/dmg_fp/dtoa_wrapper.cc',
       'base/third_party/dmg_fp/g_fmt.cc',
       'base/third_party/icu/icu_utf.cc',
-      'base/third_party/nspr/prtime.cc',
       'base/threading/post_task_and_reply_impl.cc',
       'base/threading/scoped_blocking_call.cc',
       'base/threading/sequence_local_storage_map.cc',
@@ -747,7 +746,7 @@ def write_gn_ninja(path, root_gen_dir, o
         'base/allocator/allocator_shim.cc',
         'base/allocator/allocator_shim_default_dispatch_to_glibc.cc',
       ])
-      libs.extend(['-lrt', '-latomic'])
+      libs.extend(['-lrt', '-latomic', '-lnspr4'])
       static_libraries['libevent']['include_dirs'].extend([
           os.path.join(SRC_ROOT, 'base', 'third_party', 'libevent', 'linux')
       ])
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -75,6 +75,9 @@ config("base_flags") {
       "-Wexit-time-destructors",
     ]
   }
+  ldflags = [
+    "-lnspr4",
+  ]
 }
 
 config("base_implementation") {
@@ -899,8 +902,6 @@ jumbo_component("base") {
     "third_party/dmg_fp/g_fmt.cc",
     "third_party/icu/icu_utf.cc",
     "third_party/icu/icu_utf.h",
-    "third_party/nspr/prtime.cc",
-    "third_party/nspr/prtime.h",
     "third_party/superfasthash/superfasthash.c",
     "thread_annotations.h",
     "threading/platform_thread.h",
