description: search for master_preferences in /usr/share/chromium

--- a/chrome/browser/first_run/first_run_internal_linux.cc
+++ b/chrome/browser/first_run/first_run_internal_linux.cc
@@ -18,11 +18,7 @@ bool IsOrganicFirstRun() {
 }
 
 base::FilePath MasterPrefsPath() {
-  // The standard location of the master prefs is next to the chrome binary.
-  base::FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return base::FilePath();
-  return master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
+  return base::FilePath("/usr/share/chromium/master_preferences");
 }
 
 }  // namespace internal
