description: content_shell_fonts depends on prebuilt ttf files
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/shell/test_runner/BUILD.gn
+++ b/content/shell/test_runner/BUILD.gn
@@ -204,7 +204,7 @@ group("resources") {
     deps += [ ":copy_android_fonts_config" ]
     data_deps += [ ":copy_android_fonts_config" ]
   }
-  if (is_android || is_linux || is_fuchsia) {
+  if (false) {
     deps += [ "//third_party/content_shell_fonts" ]
     data_deps += [ "//third_party/content_shell_fonts" ]
   }
