description: enable chromecast feature switch
author: Olivier Tilloy <olivier.tilloy@canonical.com>
bug-debian: http://bugs.debian.org/878244

--- a/extensions/common/feature_switch.cc
+++ b/extensions/common/feature_switch.cc
@@ -52,7 +52,7 @@ class CommonSwitches {
                          FeatureSwitch::DEFAULT_ENABLED),
         load_media_router_component_extension(
             kLoadMediaRouterComponentExtensionFlag,
-#if defined(GOOGLE_CHROME_BUILD)
+#if 1
             FeatureSwitch::DEFAULT_ENABLED),
 #else
             FeatureSwitch::DEFAULT_DISABLED),
