description: suggest proxy misconfiguration when nework is unreachable
author: Michael Gilbert <mgilbert@debian.org>
bug-debian: http://bugs.debian.org/864539

--- a/components/error_page_strings.grdp
+++ b/components/error_page_strings.grdp
@@ -52,7 +52,7 @@
   </message>
   <message name="IDS_ERRORPAGES_SUGGESTION_CHECK_CONNECTION_BODY" desc="When a page fails to load, sometimes we suggest checking the network connections.  This contains details below the suggestion.">
     Check any cables and reboot any routers, modems, or other network
-    devices you may be using.
+    devices you may be using. Also check that any proxy server currently in use is reachable and configured correctly.
   </message>
   <if expr="not is_ios or not is_android">
     <message name="IDS_ERRORPAGES_SUGGESTION_DNS_CONFIG_HEADER" desc="When a page fails to load, sometimes we suggest checking the DNS configuration.  This is a header above above some details.">
