description: need complete type for use in unique_ptr
author: Michael Gilbert <mgilbert@debian.org>

--- a/services/preferences/tracked/pref_hash_filter.h
+++ b/services/preferences/tracked/pref_hash_filter.h
@@ -22,8 +22,12 @@
 #include "services/preferences/tracked/hash_store_contents.h"
 #include "services/preferences/tracked/interceptable_pref_filter.h"
 #include "services/preferences/tracked/tracked_preference.h"
+#if defined(__GNUC__) && !defined(__clang__)
+#include "services/preferences/tracked/pref_hash_store.h"
 
+#else
 class PrefHashStore;
+#endif // defined(__GNUC__) && !defined(__clang__)
 class PrefService;
 
 namespace base {
