description: fix mojo layout test build error
author: Michael Gilbert <mgilbert@debian.org>

--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -93,8 +93,6 @@ static_library("content_shell_lib") {
     "browser/layout_test/layout_test_resource_dispatcher_host_delegate.h",
     "browser/layout_test/layout_test_url_request_context_getter.cc",
     "browser/layout_test/layout_test_url_request_context_getter.h",
-    "browser/layout_test/mojo_layout_test_helper.cc",
-    "browser/layout_test/mojo_layout_test_helper.h",
     "browser/layout_test/secondary_test_window_observer.cc",
     "browser/layout_test/secondary_test_window_observer.h",
     "browser/layout_test/test_info_extractor.cc",
@@ -244,9 +242,7 @@ static_library("content_shell_lib") {
     "//content/public/common",
     "//content/public/common:service_names",
     "//content/shell/test_runner:test_runner",
-    "//content/test:content_test_mojo_bindings",
     "//content/test:layouttest_support",
-    "//content/test:mojo_layouttest_bindings",
     "//content/test:test_support",
     "//device/bluetooth",
     "//device/bluetooth:fake_bluetooth",
--- a/content/shell/browser/layout_test/layout_test_content_browser_client.cc
+++ b/content/shell/browser/layout_test/layout_test_content_browser_client.cc
@@ -19,7 +19,6 @@
 #include "content/shell/browser/layout_test/layout_test_message_filter.h"
 #include "content/shell/browser/layout_test/layout_test_notification_manager.h"
 #include "content/shell/browser/layout_test/layout_test_resource_dispatcher_host_delegate.h"
-#include "content/shell/browser/layout_test/mojo_layout_test_helper.h"
 #include "content/shell/browser/shell_browser_context.h"
 #include "content/shell/common/layout_test/layout_test_switches.h"
 #include "content/shell/common/shell_messages.h"
@@ -32,11 +31,6 @@ namespace {
 
 LayoutTestContentBrowserClient* g_layout_test_browser_client;
 
-void BindLayoutTestHelper(mojom::MojoLayoutTestHelperRequest request,
-                          RenderFrameHost* render_frame_host) {
-  MojoLayoutTestHelper::Create(std::move(request));
-}
-
 }  // namespace
 
 LayoutTestContentBrowserClient::LayoutTestContentBrowserClient() {
@@ -98,7 +92,6 @@ void LayoutTestContentBrowserClient::Exp
 
   registry->AddInterface(base::Bind(&bluetooth::FakeBluetooth::Create),
                          ui_task_runner);
-  registry->AddInterface(base::Bind(&MojoLayoutTestHelper::Create));
 }
 
 void LayoutTestContentBrowserClient::OverrideWebkitPrefs(
@@ -192,7 +185,6 @@ bool LayoutTestContentBrowserClient::Can
 void LayoutTestContentBrowserClient::ExposeInterfacesToFrame(
     service_manager::BinderRegistryWithArgs<content::RenderFrameHost*>*
         registry) {
-  registry->AddInterface(base::Bind(&BindLayoutTestHelper));
 }
 
 }  // namespace content
