description: set widevine version as undefined
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/widevine/cdm/stub/widevine_cdm_version.h
+++ b/third_party/widevine/cdm/stub/widevine_cdm_version.h
@@ -10,6 +10,7 @@
 
 #include "third_party/widevine/cdm/widevine_cdm_common.h"
 
+#define WIDEVINE_CDM_VERSION_STRING "undefined"
 #define WIDEVINE_CDM_AVAILABLE
 
 #endif  // WIDEVINE_CDM_VERSION_H_
