--- a/debian/scripts/mk-origtargz	2017-10-27 12:52:18.000000000 +0000
+++ b/debian/scripts/mk-origtargz	2017-11-26 23:39:05.818091051 +0000
@@ -551,6 +551,11 @@
 }
 
 if ($deletecount) {
+    for my $file (@to_delete) {
+        print "$file\n";
+    }
+    exit 0;
+
     $destfilebase = sprintf "%s_%s%s.%s.tar", $package, $version, $suffix, $orig;
     $destfiletar = sprintf "%s/%s", $destdir, $destfilebase;
     $destfile = sprintf "%s.%s", $destfiletar, $destext;
