--- b/debian/scripts/mk-origtargz
+++ a/debian/scripts/mk-origtargz
@@ -211,6 +212,11 @@
     }
 
     if ($deletecount or $self->config->force_repack) {
+        for my $filename (@to_delete) {
+            printf "$filename\n";
+        }
+        return $self->status(0);
+
         $destfilebase = sprintf "%s_%s%s.%s.tar", $self->config->package,
           $self->config->version, $self->config->repack_suffix,
           $self->config->orig;
@@ -558,4 +563,4 @@
       compression_get_property($self->config->compression, "file_ext");
 }
 
-1;
+exit Devscripts::MkOrigtargz->new->do;
