description: fuzzers aren't built, so don't depend on them
author: Michael Gilbert <mgilbert@debian.org>

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -747,8 +747,7 @@ group("gn_all") {
     }
   }
 
-  if ((is_linux && !is_chromecast) || (is_win && use_libfuzzer) ||
-      (use_libfuzzer && is_mac)) {
+  if (false) {
     deps += [
       "//chrome/services/ipp_parser/public/cpp:fuzzers",
       "//testing/libfuzzer/fuzzers",
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -2404,12 +2404,6 @@ test("content_perftests") {
   }
 }
 
-group("fuzzers") {
-  deps = [
-    "//content/test/fuzzer",
-  ]
-}
-
 # This group defines the isolate files needed to run
 # test_buildbucket_api_gpu_cases.py on bots. This also tells the build system
 # when the tests should be re-run - when one of the dependent files changes.
--- a/v8/tools/BUILD.gn
+++ b/v8/tools/BUILD.gn
@@ -12,7 +12,6 @@ group("gn_all") {
     ":v8_check_static_initializers",
     "debug_helper:v8_debug_helper",
     "gcmole:v8_run_gcmole",
-    "jsfunfuzz:v8_jsfunfuzz",
   ]
 }
 
