--- a/device/gamepad/gamepad_provider.cc
+++ b/device/gamepad/gamepad_provider.cc
@@ -72,9 +72,11 @@
   // Delete GamepadDataFetchers on |polling_thread_|. This is important because
   // some of them require their destructor to be called on the same sequence as
   // their other methods.
+#if !defined(OS_LINUX)
   polling_thread_->task_runner()->PostTask(
       FROM_HERE, base::Bind(&GamepadFetcherVector::clear,
                             base::Unretained(&data_fetchers_)));
+#endif
 
   // Use Stop() to join the polling thread, as there may be pending callbacks
   // which dereference |polling_thread_|.
