Description: Fixes builds with GCC version 6

--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -468,6 +468,7 @@
     # TODO(thakis): Eventually switch this to c++11 instead,
     # http://crbug.com/427584
     cflags_c += [ "-std=gnu11" ]
+    cflags_cc += [ "-fno-delete-null-pointer-checks" ]
     if (cxx11_override) {
       # Override Chromium's default for projects that wish to stay on C++11.
       cflags_cc += [ "-std=gnu++11" ]
--- a/chrome/browser/safe_browsing/BUILD.gn
+++ b/chrome/browser/safe_browsing/BUILD.gn
@@ -46,6 +46,8 @@
     "//chrome/app:generated_resources",
     "//components/chrome_cleaner/public/interfaces",
     "//components/safe_browsing/common:interfaces",
+    "//skia",
+    "//third_party/WebKit/public:blink_headers",
   ]
 
   if (enable_extensions) {
--- a/chrome/browser/devtools/devtools_file_system_indexer.cc
+++ b/chrome/browser/devtools/devtools_file_system_indexer.cc
@@ -39,8 +39,6 @@
 
 namespace {
 
-using std::set;
-
 base::SequencedTaskRunner* impl_task_runner() {
   constexpr base::TaskTraits kBlockingTraits = {base::MayBlock(),
                                                 base::TaskPriority::BACKGROUND};
@@ -192,7 +190,7 @@
     if (trigram != kUndefinedTrigram)
       trigrams.push_back(trigram);
   }
-  set<FileId> file_ids;
+  std::set<FileId> file_ids;
   bool first = true;
   vector<Trigram>::const_iterator it = trigrams.begin();
   for (; it != trigrams.end(); ++it) {
@@ -204,7 +202,7 @@
       first = false;
       continue;
     }
-    set<FileId> intersection = base::STLSetIntersection<set<FileId> >(
+    std::set<FileId> intersection = base::STLSetIntersection<std::set<FileId> >(
         file_ids, index_[trigram]);
     file_ids.swap(intersection);
   }
