description: fix int-in-bool-context gcc warning
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/webrtc/modules/video_processing/video_denoiser.cc
+++ b/third_party/webrtc/modules/video_processing/video_denoiser.cc
@@ -187,7 +187,7 @@ void VideoDenoiser::CopySrcOnMOB(const u
       // artifacts.
       if (mb_filter_decision_[mb_index] != FILTER_BLOCK ||
           IsTrailingBlock(moving_edge_, mb_row, mb_col) ||
-          (x_density_[mb_col] * y_density_[mb_row] &&
+          (x_density_[mb_col] * y_density_[mb_row] != 0 &&
            moving_object_[mb_row * mb_cols_ + mb_col])) {
         // Copy y source.
         filter_->CopyMem16x16(mb_src, stride_src, mb_dst, stride_dst);
