--- a/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni
+++ b/third_party/devtools-frontend/src/scripts/build/ninja/bundle.gni
@@ -11,7 +11,7 @@ template("bundle") {
          "You must define the 'entrypoint' for a bundle target")
 
   node_action(target_name) {
-    script = "scripts/build/esbuild.js"
+    script = "../../node/node_modules/@rollup/wasm-node/dist/bin/rollup"
     forward_variables_from(invoker,
                            [
                              "visibility",
@@ -21,27 +21,23 @@ template("bundle") {
 
     inputs = [
       invoker.entrypoint,
+      devtools_location_prepend + "scripts/build/rollup.config.mjs",
       devtools_location_prepend + "scripts/build/devtools_plugin.js",
       devtools_location_prepend + "scripts/devtools_paths.js",
     ]
 
-    _esbuild = devtools_location_prepend + "third_party/esbuild/esbuild"
-    if (host_os == "win") {
-      inputs += [ _esbuild + ".exe" ]
-    } else {
-      inputs += [ _esbuild ]
-    }
-
     args = [
+      "--config",
+      rebase_path(
+          devtools_location_prepend + "scripts/build/rollup.config.mjs",
+          root_build_dir),
+      "--input",
       rebase_path(invoker.entrypoint, root_build_dir),
+      "--file",
       rebase_path(invoker.output_file_location, root_build_dir),
       "--configSourcemaps",
     ]
 
-    if (!is_debug) {
-      args += [ "--minify" ]
-    }
-
     outputs = [ invoker.output_file_location ]
   }
 }
