# Disable some background communication with clients2.google.com

--- a/components/crash/core/app/crash_reporter_client.cc
+++ b/components/crash/core/app/crash_reporter_client.cc
@@ -146,7 +146,7 @@ void CrashReporterClient::GetSanitizatio
 #endif
 
 std::string CrashReporterClient::GetUploadUrl() {
-#if BUILDFLAG(GOOGLE_CHROME_BRANDING) && defined(OFFICIAL_BUILD)
+#if 0
   return kDefaultUploadURL;
 #else
   return std::string();
--- a/components/crash/core/app/crashpad.cc
+++ b/components/crash/core/app/crashpad.cc
@@ -78,6 +78,13 @@ bool InitializeCrashpadImpl(bool initial
                             const std::vector<std::string>& initial_arguments,
                             bool embedded_handler,
                             const std::vector<base::FilePath>& attachments) {
+// Crashpad is needed on Linux because it's used as a WASM signal handler.
+// This is not the case on other platforms, and so it can remain entirely
+// disabled there.
+#if !BUILDFLAG(IS_LINUX)
+  return false;
+#endif
+
   static bool initialized = false;
   DCHECK(!initialized);
   initialized = true;
--- a/third_party/crashpad/crashpad/handler/handler_main.cc
+++ b/third_party/crashpad/crashpad/handler/handler_main.cc
@@ -876,7 +876,7 @@ int HandlerMain(int argc,
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS) ||
         // BUILDFLAG(IS_ANDROID)
       case kOptionURL: {
-        options.url = optarg;
+        options.url = "";
         break;
       }
 #if BUILDFLAG(IS_CHROMEOS)
@@ -1017,7 +1017,7 @@ int HandlerMain(int argc,
   }
 
   ScopedStoppable upload_thread;
-  if (!options.url.empty()) {
+  if ((false)) {
     // TODO(scottmg): options.rate_limit should be removed when we have a
     // configurable database setting to control upload limiting.
     // See https://crashpad.chromium.org/bug/23.
