--- a/chrome/common/chrome_features.cc
+++ b/chrome/common/chrome_features.cc
@@ -329,12 +329,7 @@ const base::Feature kDesktopPWAsWebAppSe
 // Enable DNS over HTTPS (DoH).
 const base::Feature kDnsOverHttps {
   "DnsOverHttps",
-#if BUILDFLAG(IS_WIN) || BUILDFLAG(IS_CHROMEOS) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_LINUX)
-      base::FEATURE_ENABLED_BY_DEFAULT
-#else
       base::FEATURE_DISABLED_BY_DEFAULT
-#endif
 };
 
 // Provides a mechanism to remove providers from the dropdown list in the
--- a/net/dns/public/doh_provider_entry.cc
+++ b/net/dns/public/doh_provider_entry.cc
@@ -125,24 +125,6 @@ const DohProviderEntry::List& DohProvide
           /*privacy_policy=*/"https://dns.sb/privacy/",
           /*display_globally=*/false, /*display_countries=*/{"EE", "DE"},
           LoggingLevel::kNormal),
-      new DohProviderEntry("Google", DohProviderIdForHistogram::kGoogle,
-                           {"8.8.8.8", "8.8.4.4", "2001:4860:4860::8888",
-                            "2001:4860:4860::8844"},
-                           {"dns.google", "dns.google.com",
-                            "8888.google"} /* dns_over_tls_hostnames */,
-                           "https://dns.google/dns-query{?dns}",
-                           "Google (Public DNS)" /* ui_name */,
-                           "https://developers.google.com/speed/public-dns/"
-                           "privacy" /* privacy_policy */,
-                           true /* display_globally */,
-                           {} /* display_countries */, LoggingLevel::kExtra),
-      new DohProviderEntry(
-          "GoogleDns64", absl::nullopt /* provider_id_for_histogram */,
-          {"2001:4860:4860::64", "2001:4860:4860::6464"},
-          {"dns64.dns.google"} /* dns_over_tls_hostnames */,
-          "https://dns64.dns.google/dns-query{?dns}", "" /* ui_name */,
-          "" /* privacy_policy */, false /* display_globally */,
-          {} /* display_countries */, LoggingLevel::kNormal),
       new DohProviderEntry("Iij", DohProviderIdForHistogram::kIij,
                            {} /* ip_strs */, {} /* dns_over_tls_hostnames */,
                            "https://public.dns.iij.jp/dns-query",
--- a/services/network/public/cpp/features.cc
+++ b/services/network/public/cpp/features.cc
@@ -92,12 +92,7 @@ const base::Feature kSplitAuthCacheByNet
 // Enable usage of hardcoded DoH upgrade mapping for use in automatic mode.
 const base::Feature kDnsOverHttpsUpgrade {
   "DnsOverHttpsUpgrade",
-#if BUILDFLAG(IS_CHROMEOS_ASH) || BUILDFLAG(IS_MAC) || \
-    BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_WIN)
-      base::FEATURE_ENABLED_BY_DEFAULT
-#else
       base::FEATURE_DISABLED_BY_DEFAULT
-#endif
 };
 
 // If this feature is enabled, the mDNS responder service responds to queries
