--- a/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
+++ b/chrome/browser/extensions/api/developer_private/extension_info_generator.cc
@@ -871,16 +871,6 @@ void ExtensionInfoGenerator::FillExtensi
         toolbar_actions_model->IsActionPinned(extension.id());
   }
 
-  // MV2 deprecation.
-  ManifestV2Handler* mv2_handler = ManifestV2Handler::Get(profile);
-  CHECK(mv2_handler);
-  info.is_affected_by_mv2_deprecation =
-      mv2_handler->IsExtensionAffected(extension);
-  if (info.web_store_url.length() > 0) {
-    info.recommendations_url =
-        extension_urls::GetNewWebstoreItemRecommendationsUrl(extension.id())
-            .spec();
-  }
 
   info.can_upload_as_account_extension =
       AccountExtensionTracker::Get(profile)->CanUploadAsAccountExtension(
--- a/chrome/browser/ui/webui/extensions/extensions_ui.cc
+++ b/chrome/browser/ui/webui/extensions/extensions_ui.cc
@@ -486,7 +486,7 @@ content::WebUIDataSource* CreateAndAddEx
           extensions_features::kExtensionsMenuAccessControlWithPermittedSites));
   source->AddBoolean(
       "safetyHubThreeDotDetails",
-      base::FeatureList::IsEnabled(features::kSafetyHubThreeDotDetails));
+      false);
   source->AddBoolean("enableGlobalScopedShortcuts", IsGlobalShortcutEnabled());
 
   // MV2 deprecation.
--- a/extensions/browser/manifest_v2_handler.cc
+++ b/extensions/browser/manifest_v2_handler.cc
@@ -108,12 +108,7 @@ bool ManifestV2HandlerFactory::ServiceIs
 
 // Returns true if legacy extensions should be disabled.
 bool ShouldDisableLegacyExtensions() {
-  if (g_allow_mv2_for_testing) {
-    // We allow legacy MV2 extensions for testing purposes.
     return false;
-  }
-
-  return true;
 }
 
 }  // namespace
--- a/extensions/common/extension.cc
+++ b/extensions/common/extension.cc
@@ -100,12 +100,6 @@ bool IsManifestSupported(int manifest_ve
   // Supported versions are always safe.
   if (manifest_version >= kMinimumSupportedManifestVersion &&
       manifest_version <= kMaximumSupportedManifestVersion) {
-    // Emit a warning for unpacked extensions on Manifest V2 warning that
-    // MV2 is deprecated.
-    if (type == Manifest::Type::kExtension && manifest_version == 2 &&
-        Manifest::IsUnpackedLocation(location)) {
-      *warning = errors::kManifestV2IsDeprecatedWarning;
-    }
     return true;
   }
 
