# Fix building without one click signin

--- a/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
+++ b/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
@@ -16,9 +16,4 @@ OneClickSigninLinksDelegateImpl::OneClic
 OneClickSigninLinksDelegateImpl::~OneClickSigninLinksDelegateImpl() {}
 
 void OneClickSigninLinksDelegateImpl::OnLearnMoreLinkClicked(bool is_dialog) {
-  NavigateParams params(browser_, GURL(chrome::kChromeSyncLearnMoreURL),
-                        ui::PAGE_TRANSITION_LINK);
-  params.disposition = is_dialog ? WindowOpenDisposition::NEW_WINDOW
-                                 : WindowOpenDisposition::NEW_FOREGROUND_TAB;
-  Navigate(&params);
 }
--- a/chrome/browser/ui/webui/signin/inline_login_handler_impl.cc
+++ b/chrome/browser/ui/webui/signin/inline_login_handler_impl.cc
@@ -343,6 +343,7 @@ void InlineSigninHelper::OnClientOAuthSu
 void InlineSigninHelper::OnClientOAuthSuccessAndBrowserOpened(
     const ClientOAuthResult& result,
     Profile* /*profile*/) {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   HandlerSigninReason reason = GetHandlerSigninReason(current_url_);
   if (reason == HandlerSigninReason::kFetchLstOnly) {
     // Constants are only available on Windows for the Google Credential
@@ -419,6 +420,7 @@ void InlineSigninHelper::OnClientOAuthSu
   }
 
   base::ThreadTaskRunnerHandle::Get()->DeleteSoon(FROM_HERE, this);
+#endif
 }
 
 void InlineSigninHelper::UntrustedSigninConfirmed(
