# Fix building without one click signin

--- a/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
+++ b/chrome/browser/ui/sync/one_click_signin_sync_starter.cc
@@ -395,6 +395,7 @@ void OneClickSigninSyncStarter::CancelSi
 }
 
 void OneClickSigninSyncStarter::ConfirmAndSignin() {
+#if BUILDFLAG(ENABLE_ONE_CLICK_SIGNIN)
   SigninManager* signin = SigninManagerFactory::GetForProfile(profile_);
   if (confirmation_required_ == CONFIRM_UNTRUSTED_SIGNIN) {
     browser_ = EnsureBrowser(browser_, profile_);
@@ -409,6 +410,7 @@ void OneClickSigninSyncStarter::ConfirmA
     // No confirmation required - just sign in the user.
     signin->CompletePendingSignin();
   }
+#endif
 }
 
 void OneClickSigninSyncStarter::UntrustedSigninConfirmed(
--- a/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
+++ b/chrome/browser/ui/sync/one_click_signin_links_delegate_impl.cc
@@ -17,13 +17,7 @@ OneClickSigninLinksDelegateImpl::OneClic
 OneClickSigninLinksDelegateImpl::~OneClickSigninLinksDelegateImpl() {}
 
 void OneClickSigninLinksDelegateImpl::OnLearnMoreLinkClicked(bool is_dialog) {
-  NavigateParams params(browser_, GURL(chrome::kChromeSyncLearnMoreURL),
-                        ui::PAGE_TRANSITION_LINK);
-  params.disposition = is_dialog ? WindowOpenDisposition::NEW_WINDOW
-                                 : WindowOpenDisposition::NEW_FOREGROUND_TAB;
-  Navigate(&params);
 }
 
 void OneClickSigninLinksDelegateImpl::OnAdvancedLinkClicked() {
-  chrome::ShowSettingsSubPage(browser_, chrome::kSyncSetupSubPage);
 }
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -2993,8 +2993,6 @@ jumbo_split_static_library("ui") {
         "views/relaunch_notification/relaunch_required_dialog_view.cc",
         "views/relaunch_notification/relaunch_required_dialog_view.h",
         "views/screen_capture_notification_ui_views.cc",
-        "views/sync/one_click_signin_dialog_view.cc",
-        "views/sync/one_click_signin_dialog_view.h",
         "views/tabs/window_finder.cc",
       ]
     }
