--- a/chrome/browser/ui/accelerator_table.cc
+++ b/chrome/browser/ui/accelerator_table.cc
@@ -133,7 +133,6 @@ const AcceleratorMapping kAcceleratorMap
     {ui::VKEY_OEM_PLUS, ui::EF_PLATFORM_ACCELERATOR, IDC_ZOOM_PLUS},
     {ui::VKEY_ADD, ui::EF_PLATFORM_ACCELERATOR, IDC_ZOOM_PLUS},
 
-    {ui::VKEY_F1, ui::EF_NONE, IDC_HELP_PAGE_VIA_KEYBOARD},
     {ui::VKEY_F3, ui::EF_NONE, IDC_FIND_NEXT},
     {ui::VKEY_F3, ui::EF_SHIFT_DOWN, IDC_FIND_PREVIOUS},
     {ui::VKEY_F4, ui::EF_CONTROL_DOWN, IDC_CLOSE_TAB},
