--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -604,9 +604,6 @@ config("compiler") {
         # Experimentally determined to yield a reasonable trade-off between
         # build time, run-time performance, and binary size.
         "-mllvm:-import-instr-limit=10",
-
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm:-enable-dse-memoryssa=false",
       ]
     } else {
       ldflags += [ "-flto=thin" ]
@@ -631,9 +628,6 @@ config("compiler") {
         "-Wl,--thinlto-cache-policy,$cache_policy",
       ]
 
-      # TODO(https://crbug.com/1127713): Investigate, remove.
-      ldflags += [ "-Wl,-mllvm,-enable-dse-memoryssa=false" ]
-
       ldflags += [ "-Wl,--lto-O" + lto_opt_level ]
       if (thin_lto_enable_optimizations) {
         # TODO(gbiv): We ideally shouldn't need to specify this; ThinLTO
