--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -489,25 +489,6 @@ config("compiler") {
       "-mllvm",
       "-instcombine-lower-dbg-declare=0",
     ]
-
-    # Pinned clang has enable-dse-memoryssa enabled by default but it's broken,
-    # so we have to force it off.
-    # Trunk clang has it disabled by default but it might work, so we force it
-    # on so the ToT bots can check if it works now.
-    if (!llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=false",
-      ]
-    }
-    if (llvm_force_head_revision) {
-      cflags += [
-        # TODO(https://crbug.com/1127713): Investigate, remove.
-        "-mllvm",
-        "-enable-dse-memoryssa=true",
-      ]
-    }
   }
 
   # C11/C++11 compiler flags setup.
