--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -651,9 +651,6 @@ config("compiler") {
         "-Wl,--thinlto-cache-policy,$cache_policy",
       ]
 
-      # TODO(https://crbug.com/1127713): Investigate, remove.
-      ldflags += [ "-Wl,-mllvm,-enable-dse-memoryssa=false" ]
-
       ldflags += [ "-Wl,--lto-O" + lto_opt_level ]
       if (thin_lto_enable_optimizations) {
         # TODO(gbiv): We ideally shouldn't need to specify this; ThinLTO
