--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -2043,11 +2043,11 @@ const FeatureEntry kFeatureEntries[] = {
     {"enable-wifi-credential-sync", flag_descriptions::kWifiCredentialSyncName,
      flag_descriptions::kWifiCredentialSyncDescription, kOsCrOS,
      SINGLE_VALUE_TYPE(switches::kEnableWifiCredentialSync)},
+#endif  // OS_CHROMEOS
     {"enable-potentially-annoying-security-features",
      flag_descriptions::kExperimentalSecurityFeaturesName,
      flag_descriptions::kExperimentalSecurityFeaturesDescription, kOsAll,
      SINGLE_VALUE_TYPE(switches::kEnablePotentiallyAnnoyingSecurityFeatures)},
-#endif  // OS_CHROMEOS
     {"committed-interstitials", flag_descriptions::kCommittedInterstitialsName,
      flag_descriptions::kCommittedInterstitialsDescription, kOsAll,
      SINGLE_VALUE_TYPE(switches::kCommittedInterstitials)},
