--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -2247,11 +2247,11 @@ const FeatureEntry kFeatureEntries[] = {
     {"enable-wifi-credential-sync", flag_descriptions::kWifiCredentialSyncName,
      flag_descriptions::kWifiCredentialSyncDescription, kOsCrOS,
      SINGLE_VALUE_TYPE(switches::kEnableWifiCredentialSync)},
+#endif  // OS_CHROMEOS
     {"enable-potentially-annoying-security-features",
      flag_descriptions::kExperimentalSecurityFeaturesName,
      flag_descriptions::kExperimentalSecurityFeaturesDescription, kOsAll,
      SINGLE_VALUE_TYPE(switches::kEnablePotentiallyAnnoyingSecurityFeatures)},
-#endif  // OS_CHROMEOS
     {"ssl-committed-interstitials",
      flag_descriptions::kSSLCommittedInterstitialsName,
      flag_descriptions::kSSLCommittedInterstitialsDescription, kOsAll,
