description: Stop downloading of profile avatar (trk:271:...)

--- a/chrome/browser/profiles/profile_avatar_downloader.cc
+++ b/chrome/browser/profiles/profile_avatar_downloader.cc
@@ -23,8 +23,8 @@ ProfileAvatarDownloader::ProfileAvatarDownloader(
     : icon_index_(icon_index),
       profile_path_(profile_path),
       cache_(cache) {
-  GURL url(std::string(kHighResAvatarDownloadUrlPrefix) +
-           profiles::GetDefaultAvatarIconFileNameAtIndex(icon_index));
+  GURL url(std::string("about:blank"));
+  //         profiles::GetDefaultAvatarIconFileNameAtIndex(icon_index));
   fetcher_.reset(new chrome::BitmapFetcher(url, this));
 }
 
