# Disable domain reliability component

--- a/components/domain_reliability/baked_in_configs.gypi
+++ b/components/domain_reliability/baked_in_configs.gypi
@@ -5,24 +5,6 @@
 {
   'variables': {
     'baked_in_configs': [
-      'domain_reliability/baked_in_configs/c_android_clients_google_com.json',
-      'domain_reliability/baked_in_configs/c_bigcache_googleapis_com.json',
-      'domain_reliability/baked_in_configs/c_doc-0-0-sj_sj_googleusercontent_com.json',
-      'domain_reliability/baked_in_configs/c_docs_google_com.json',
-      'domain_reliability/baked_in_configs/c_drive_google_com.json',
-      'domain_reliability/baked_in_configs/c_googlesyndication_com.json',
-      'domain_reliability/baked_in_configs/c_pack_google_com.json',
-      'domain_reliability/baked_in_configs/c_play_google_com.json',
-      'domain_reliability/baked_in_configs/c_youtube_com.json',
-      'domain_reliability/baked_in_configs/clients2_google_com.json',
-      'domain_reliability/baked_in_configs/docs_google_com.json',
-      'domain_reliability/baked_in_configs/google-analytics_com.json',
-      'domain_reliability/baked_in_configs/googlevideo_com.json',
-      'domain_reliability/baked_in_configs/gvt1_com.json',
-      'domain_reliability/baked_in_configs/gvt2_com.json',
-      'domain_reliability/baked_in_configs/mail_google_com.json',
-      'domain_reliability/baked_in_configs/ssl_gstatic_com.json',
-      'domain_reliability/baked_in_configs/www_google_com.json',
     ],
   },
 }
--- a/components/domain_reliability/google_configs.cc
+++ b/components/domain_reliability/google_configs.cc
@@ -30,6 +30,7 @@ struct GoogleConfigParams {
 };
 
 const GoogleConfigParams kGoogleConfigs[] = {
+#if 0
   // Origins with subdomains and same-origin collectors.
   { "google.ac", true, true, true },
   { "google.ad", true, true, true },
@@ -485,9 +486,11 @@ const GoogleConfigParams kGoogleConfigs[] = {
   { "ad.60u613cl1c4.qjz9zk", false, false, false },
   { "drive.9oo91e.qjz9zk", false, false, false },
   { "redirector.9oo91evideo.qjz9zk", false, false, false },
+#endif
 };
 
 const char* kGoogleStandardCollectors[] = {
+#if 0
   "https://beacons.gvt2.com/domainreliability/upload",
   "https://beacons2.gvt2.com/domainreliability/upload",
   "https://beacons3.gvt2.com/domainreliability/upload",
@@ -495,6 +498,7 @@ const char* kGoogleStandardCollectors[] = {
   "https://beacons5.gvt2.com/domainreliability/upload",
   "https://beacons5.gvt3.com/domainreliability/upload",
   "https://clients2.9oo91e.qjz9zk/domainreliability/upload",
+#endif
 };
 
 const char* kGoogleOriginSpecificCollectorPathString =
--- a/components/domain_reliability/uploader.cc
+++ b/components/domain_reliability/uploader.cc
@@ -75,7 +75,7 @@ class DomainReliabilityUploaderImpl
     VLOG(1) << "Uploading report to " << upload_url;
     VLOG(2) << "Report JSON: " << report_json;
 
-    if (discard_uploads_) {
+    if (true) {
       VLOG(1) << "Discarding report instead of uploading.";
       UploadResult result;
       result.status = UploadResult::SUCCESS;
