description: Disable prompt about "unsafe" commandline switches

--- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
@@ -832,7 +832,6 @@ void StartupBrowserCreatorImpl::AddInfoBarsIfNecessary(
   // so we cannot add the info bars to the focused tabs here.
   if (is_process_startup == chrome::startup::IS_PROCESS_STARTUP &&
       !command_line_.HasSwitch(switches::kTestType)) {
-    chrome::ShowBadFlagsPrompt(browser);
     ObsoleteSystemInfoBarDelegate::Create(InfoBarService::FromWebContents(
         browser->tab_strip_model()->GetActiveWebContents()));
 

