# Disable the web resource service

--- a/./components/web_resource/web_resource_service.cc
+++ b/./components/web_resource/web_resource_service.cc
@@ -114,27 +114,8 @@ void WebResourceService::StartFetch() {
   prefs_->SetString(last_update_time_pref_name_,
                     base::DoubleToString(base::Time::Now().ToDoubleT()));
 
-  // If we are still fetching data, exit.
-  if (in_fetch_)
-    return;
-  in_fetch_ = true;
-
-  GURL web_resource_server =
-      application_locale_.empty()
-          ? web_resource_server_
-          : google_util::AppendGoogleLocaleParam(web_resource_server_,
-                                                 application_locale_);
-
-  DVLOG(1) << "WebResourceService StartFetch " << web_resource_server;
-  url_fetcher_ =
-      net::URLFetcher::Create(web_resource_server, net::URLFetcher::GET, this);
-  // Do not let url fetcher affect existing state in system context
-  // (by setting cookies, for example).
-  url_fetcher_->SetLoadFlags(net::LOAD_DISABLE_CACHE |
-                             net::LOAD_DO_NOT_SEND_COOKIES |
-                             net::LOAD_DO_NOT_SAVE_COOKIES);
-  url_fetcher_->SetRequestContext(request_context_.get());
-  url_fetcher_->Start();
+  in_fetch_ = false;
+  return;
 }
 
 void WebResourceService::EndFetch() {
