# Fix screen sharing in Google Meet
# http://crbug.com/829916#c16

--- a/chrome/common/chrome_content_client_constants.cc
+++ b/chrome/common/chrome_content_client_constants.cc
@@ -10,8 +10,8 @@ const char ChromeContentClient::kPDFExte
 const char ChromeContentClient::kPDFInternalPluginName[] = "Chrome PDF Plugin";
 #else
 const char ChromeContentClient::kPDFExtensionPluginName[] =
-    "Chromium PDF Viewer";
+    "Chrome PDF Viewer";
 const char ChromeContentClient::kPDFInternalPluginName[] =
-    "Chromium PDF Plugin";
+    "Chrome PDF Plugin";
 #endif
 const char ChromeContentClient::kPDFPluginPath[] = "internal-pdf-viewer";
