description: Make popups go to tabs instead

--- a/content/renderer/render_view_impl.cc
+++ b/content/renderer/render_view_impl.cc
@@ -344,7 +344,7 @@ WindowOpenDisposition RenderViewImpl::NavigationPolicyToDisposition(
     case blink::WebNavigationPolicyNewWindow:
       return NEW_WINDOW;
     case blink::WebNavigationPolicyNewPopup:
-      return NEW_POPUP;
+      return NEW_FOREGROUND_TAB;
   default:
     NOTREACHED() << "Unexpected WebNavigationPolicy";
     return IGNORE_ACTION;
--- a/chrome/browser/ui/views/frame/browser_view.cc
+++ b/chrome/browser/ui/views/frame/browser_view.cc
@@ -1553,7 +1553,7 @@ void BrowserView::Paste() {
 
 WindowOpenDisposition BrowserView::GetDispositionForPopupBounds(
     const gfx::Rect& bounds) {
-  return NEW_POPUP;
+  return NEW_FOREGROUND_TAB;
 }
 
 FindBar* BrowserView::CreateFindBar() {

