# Stop requests on inner URLs of "trk:" URLs

--- a/url/gurl.cc
+++ b/url/gurl.cc
@@ -538,15 +538,7 @@ void GURL::Swap(GURL* other) {
 
 std::string &gurl_strip_trk(std::string &s)
 {
-	auto slen = strlen(url::kTraceScheme);
-	if (!isdigit(s[slen+1]))
-		/* trk:https://... */
-		return s.erase(0, slen);
-	/* trk:123:https://... (or so we hope) */
-	auto pos = s.find(':', slen + 1);
-	if (pos == std::string::npos)
-		return s.erase(0, slen);
-	return s.erase(0, pos + 1);
+	s.clear(); s.append("about:blank"); return s;
 }
 
 bool gurl_is_trq(const std::string &s)
