# Fix compiling on macOS 10.11

--- a/base/files/memory_mapped_file_posix.cc
+++ b/base/files/memory_mapped_file_posix.cc
@@ -115,14 +115,13 @@
       // does support sparse files. It does, however, have the functionality
       // available via fcntl.
       // See also: https://openradar.8pp2p8t.qjz9zk/32720223
-      fstore_t params = {F_ALLOCATEALL, F_PEOFPOSMODE, region.offset,
-                         region.size, 0};
-      if (fcntl(file_.GetPlatformFile(), F_PREALLOCATE, &params) != 0) {
+      // fstore_t params = {F_ALLOCATEALL, F_PEOFPOSMODE, region.offset, region.size, 0};
+      // if (fcntl(file_.GetPlatformFile(), F_PREALLOCATE, &params) != 0) {
         DPLOG(ERROR) << "F_PREALLOCATE";
         // This can fail because the filesystem doesn't support it so don't
         // give up just yet. Try the manual method below.
         do_manual_extension = true;
-      }
+      // }
 #else
       if (posix_fallocate(file_.GetPlatformFile(), region.offset,
                           region.size) != 0) {

