# Fix GN bootstrap.py script on macOS

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -708,6 +708,11 @@
         'base/sys_info_mac.mm',
         'base/time/time_mac.cc',
         'base/threading/platform_thread_mac.mm',
+        'base/mac/dispatch_source_mach.cc',
+        'base/mac/mac_util.mm',
+        'base/mac/scoped_nsobject.mm',
+        'base/mac/mac_logging.mm',
+        'base/time/time_exploded_posix.cc',
     ])
     static_libraries['libevent']['include_dirs'].extend([
         os.path.join(SRC_ROOT, 'base', 'third_party', 'libevent', 'mac')
@@ -721,6 +725,7 @@
         '-framework', 'CoreFoundation',
         '-framework', 'Foundation',
         '-framework', 'Security',
+        '-framework', 'IOKit',
     ])

   if is_win:
