# Fix GN bootstrap.py script on macOS

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -707,10 +707,18 @@
         'base/process/process_iterator_mac.cc',
         'base/process/process_metrics_mac.cc',
         'base/strings/sys_string_conversions_mac.mm',
+        'base/threading/worker_pool_posix.cc',
+        'base/synchronization/read_write_lock_posix.cc',
         'base/synchronization/waitable_event_mac.cc',
         'base/sys_info_mac.mm',
         'base/time/time_mac.cc',
         'base/threading/platform_thread_mac.mm',
+        'base/mac/dispatch_source_mach.cc',
+        'base/tracking_info.cc',
+        'base/tracked_objects.cc',
+        'base/mac/mac_util.mm',
+        'base/mac/scoped_nsobject.mm',
+        'base/mac/mac_logging.mm',
     ])
     static_libraries['libevent']['include_dirs'].extend([
         os.path.join(SRC_ROOT, 'base', 'third_party', 'libevent', 'mac')
@@ -724,6 +732,7 @@
         '-framework', 'CoreFoundation',
         '-framework', 'Foundation',
         '-framework', 'Security',
+        '-framework', 'IOKit',
     ])

   if is_win:
