# Changes to gsutil that:
#   allow usage of python 3.14
--- a/gslib/commands/cp.py
+++ b/gslib/commands/cp.py
@@ -1043,8 +1043,8 @@ class CpCommand(Command):
       self.has_file_dst = self.has_file_dst or exp_dst_url.IsFileUrl()
       self.has_cloud_dst = self.has_cloud_dst or exp_dst_url.IsCloudUrl()
       self.provider_types.add(exp_dst_url.scheme)
-      self.combined_src_urls = itertools.chain(self.combined_src_urls,
-                                               src_url_str)
+      self.combined_src_urls = list(itertools.chain(self.combined_src_urls,
+                                               src_url_str))
 
       yield name_expansion_iterator_dst_tuple
 
--- a/gsutil.py
+++ b/gsutil.py
@@ -27,12 +27,12 @@ import warnings
 # TODO: gsutil-beta: Distribute a pylint rc file.
 
 ver = sys.version_info
-if ver.major != 3 or ver.minor < 9 or ver.minor > 13:
+if ver.major != 3 or ver.minor < 9 or ver.minor > 14:
   sys.exit(
-    "Error: gsutil requires Python version 3.9-3.13, but a different version is installed.\n"
+    "Error: gsutil requires Python version 3.9-3.14, but a different version is installed.\n"
     "You are currently running Python {}.{}\n"
     "Follow the steps below to resolve this issue:\n"
-    "\t1. Switch to Python 3.9-3.13 using your Python version manager or install an appropriate version.\n"
+    "\t1. Switch to Python 3.9-3.14 using your Python version manager or install an appropriate version.\n"
     "\t2. If you are unsure how to manage Python versions, visit [https://cloud.google.com/storage/docs/gsutil_install#specifications] for detailed instructions.".format(ver.major, ver.minor)
   )
 
