--- a/chrome/browser/enterprise/connectors/analysis/print_content_analysis_utils.cc
+++ b/chrome/browser/enterprise/connectors/analysis/print_content_analysis_utils.cc
@@ -59,6 +59,7 @@ void PrintIfAllowedByPolicy(scoped_refpt
                             base::OnceClosure hide_preview) {
   ContentAnalysisDelegate::Data scanning_data;
 
+#if BUILDFLAG(ENABLE_PRINT_CONTENT_ANALYSIS)
   if (ContentAnalysisDelegate::IsEnabled(
           Profile::FromBrowserContext(initiator->GetBrowserContext()),
           initiator->GetLastCommittedURL(), &scanning_data,
@@ -78,6 +79,7 @@ void PrintIfAllowedByPolicy(scoped_refpt
                  std::move(on_verdict));
     return;
   }
+#endif  // BUILDFLAG(ENABLE_PRINT_CONTENT_ANALYSIS)
   std::move(on_verdict).Run(/*allowed=*/true);
 }
 
@@ -92,11 +94,13 @@ absl::optional<ContentAnalysisDelegate::
     return absl::nullopt;
   }
 
+#if BUILDFLAG(ENABLE_PRINT_CONTENT_ANALYSIS)
   if ((base::FeatureList::IsEnabled(
            printing::features::kEnablePrintScanAfterPreview) &&
        scanning_data.settings.cloud_or_local_settings.is_local_analysis())) {
     return absl::nullopt;
   }
+#endif  // BUILDFLAG(ENABLE_PRINT_CONTENT_ANALYSIS)
 
   return scanning_data;
 }
