--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -29,6 +29,7 @@ namespace {
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* const engines_default[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
@@ -199,6 +200,7 @@ const PrepopulatedEngine* const engines_
 
 // China
 const PrepopulatedEngine* const engines_CN[] = {
+    &duckduckgo,
     &baidu,
     &sogou,
     &google,
@@ -280,6 +282,7 @@ const PrepopulatedEngine* const engines_
 
 // Estonia
 const PrepopulatedEngine* const engines_EE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yandex_ru,
@@ -325,6 +328,7 @@ const PrepopulatedEngine* const engines_
 
 // France
 const PrepopulatedEngine* const engines_FR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_fr,
@@ -442,6 +446,7 @@ const PrepopulatedEngine* const engines_
 
 // Iran
 const PrepopulatedEngine* const engines_IR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
@@ -505,6 +510,7 @@ const PrepopulatedEngine* const engines_
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
+    &duckduckgo,
     &google,
     &naver,
     &bing,
@@ -523,6 +529,7 @@ const PrepopulatedEngine* const engines_
 
 // Kazakhstan
 const PrepopulatedEngine* const engines_KZ[] = {
+    &duckduckgo,
     &google,
     &yandex_kz,
     &mail_ru,
@@ -604,6 +611,7 @@ const PrepopulatedEngine* const engines_
 
 // Moldova
 const PrepopulatedEngine* const engines_MD[] = {
+    &duckduckgo,
     &google,
     &yandex_ru,
     &mail_ru,
@@ -908,6 +916,7 @@ const PrepopulatedEngine* const engines_
 
 // Taiwan
 const PrepopulatedEngine* const engines_TW[] = {
+    &duckduckgo,
     &google,
     &yahoo_tw,
     &bing,
@@ -962,6 +971,7 @@ const PrepopulatedEngine* const engines_
 
 // Vietnam
 const PrepopulatedEngine* const engines_VN[] = {
+    &duckduckgo,
     &google,
     &coccoc,
     &bing,
