--- a/components/search_engines/template_url_prepopulate_data.cc
+++ b/components/search_engines/template_url_prepopulate_data.cc
@@ -29,6 +29,7 @@ namespace {
 
 // Default (for countries with no better engine set)
 const PrepopulatedEngine* const engines_default[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
@@ -38,167 +39,168 @@ const PrepopulatedEngine* const engines_
 // Engine selection by country ------------------------------------------------
 // United Arab Emirates
 const PrepopulatedEngine* const engines_AE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Albania
 const PrepopulatedEngine* const engines_AL[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Argentina
 const PrepopulatedEngine* const engines_AR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_ar,
-    &duckduckgo,
     &ecosia,
 };
 
 // Austria
 const PrepopulatedEngine* const engines_AT[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &yahoo_at,
     &ecosia,
 };
 
 // Australia
 const PrepopulatedEngine* const engines_AU[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_au,
-    &duckduckgo,
     &ecosia,
 };
 
 // Bosnia and Herzegovina
 const PrepopulatedEngine* const engines_BA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
 };
 
 // Belgium
 const PrepopulatedEngine* const engines_BE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &ecosia,
-    &duckduckgo,
 };
 
 // Bulgaria
 const PrepopulatedEngine* const engines_BG[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Bahrain
 const PrepopulatedEngine* const engines_BH[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Burundi
 const PrepopulatedEngine* const engines_BI[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Brunei
 const PrepopulatedEngine* const engines_BN[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Bolivia
 const PrepopulatedEngine* const engines_BO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Brazil
 const PrepopulatedEngine* const engines_BR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_br,
-    &duckduckgo,
     &ecosia,
 };
 
 // Belarus
 const PrepopulatedEngine* const engines_BY[] = {
+    &duckduckgo,
     &google,
     &yandex_by,
     &mail_ru,
     &bing,
-    &duckduckgo,
 };
 
 // Belize
 const PrepopulatedEngine* const engines_BZ[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ask,
 };
 
 // Canada
 const PrepopulatedEngine* const engines_CA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_ca,
-    &duckduckgo,
     &ecosia,
 };
 
 // Switzerland
 const PrepopulatedEngine* const engines_CH[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &ecosia,
     &yahoo_ch,
 };
 
 // Chile
 const PrepopulatedEngine* const engines_CL[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_es,
-    &duckduckgo,
     &ecosia,
 };
 
 // China
 const PrepopulatedEngine* const engines_CN[] = {
+    &duckduckgo,
     &baidu,
     &sogou,
     &google,
@@ -208,78 +210,79 @@ const PrepopulatedEngine* const engines_
 
 // Colombia
 const PrepopulatedEngine* const engines_CO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_es,
     &ecosia,
-    &duckduckgo,
 };
 
 // Costa Rica
 const PrepopulatedEngine* const engines_CR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Czech Republic
 const PrepopulatedEngine* const engines_CZ[] = {
+    &duckduckgo,
     &google,
     &seznam_cz,
     &bing,
     &yahoo,
-    &duckduckgo,
 };
 
 // Germany
 const PrepopulatedEngine* const engines_DE[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &ecosia,
     &yahoo_de,
 };
 
 // Denmark
 const PrepopulatedEngine* const engines_DK[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_dk,
-    &duckduckgo,
     &ecosia,
 };
 
 // Dominican Republic
 const PrepopulatedEngine* const engines_DO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Algeria
 const PrepopulatedEngine* const engines_DZ[] = {
+    &duckduckgo,
     &google,
     &yahoo_uk,
     &bing,
     &yandex_ru,
-    &duckduckgo,
 };
 
 // Ecuador
 const PrepopulatedEngine* const engines_EC[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Estonia
 const PrepopulatedEngine* const engines_EE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yandex_ru,
@@ -289,42 +292,43 @@ const PrepopulatedEngine* const engines_
 
 // Egypt
 const PrepopulatedEngine* const engines_EG[] = {
+    &duckduckgo,
     &google,
     &yahoo,
     &bing,
     &yandex_com,
-    &duckduckgo,
 };
 
 // Spain
 const PrepopulatedEngine* const engines_ES[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_es,
-    &duckduckgo,
     &ecosia,
 };
 
 // Finland
 const PrepopulatedEngine* const engines_FI[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_fi,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Faroe Islands
 const PrepopulatedEngine* const engines_FO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_dk,
-    &duckduckgo,
     &ecosia,
 };
 
 // France
 const PrepopulatedEngine* const engines_FR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_fr,
@@ -334,114 +338,115 @@ const PrepopulatedEngine* const engines_
 
 // United Kingdom
 const PrepopulatedEngine* const engines_GB[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_uk,
-    &duckduckgo,
     &ecosia,
 };
 
 // Greece
 const PrepopulatedEngine* const engines_GR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Guatemala
 const PrepopulatedEngine* const engines_GT[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Hong Kong
 const PrepopulatedEngine* const engines_HK[] = {
+    &duckduckgo,
     &google,
     &yahoo_hk,
     &bing,
     &baidu,
-    &duckduckgo,
 };
 
 // Honduras
 const PrepopulatedEngine* const engines_HN[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Croatia
 const PrepopulatedEngine* const engines_HR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Hungary
 const PrepopulatedEngine* const engines_HU[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Indonesia
 const PrepopulatedEngine* const engines_ID[] = {
+    &duckduckgo,
     &google,
     &yahoo_id,
     &bing,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Ireland
 const PrepopulatedEngine* const engines_IE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Israel
 const PrepopulatedEngine* const engines_IL[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yandex_ru,
     &yahoo,
-    &duckduckgo,
 };
 
 // India
 const PrepopulatedEngine* const engines_IN[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_in,
-    &duckduckgo,
     &ecosia,
 };
 
 // Iraq
 const PrepopulatedEngine* const engines_IQ[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_uk,
     &yandex_com,
-    &duckduckgo,
 };
 
 // Iran
 const PrepopulatedEngine* const engines_IR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
@@ -451,60 +456,61 @@ const PrepopulatedEngine* const engines_
 
 // Iceland
 const PrepopulatedEngine* const engines_IS[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Italy
 const PrepopulatedEngine* const engines_IT[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Jamaica
 const PrepopulatedEngine* const engines_JM[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Jordan
 const PrepopulatedEngine* const engines_JO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Japan
 const PrepopulatedEngine* const engines_JP[] = {
+    &duckduckgo,
     &google,
     &yahoo_jp,
     &bing,
     &baidu,
-    &duckduckgo,
 };
 
 // Kenya
 const PrepopulatedEngine* const engines_KE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // South Korea
 const PrepopulatedEngine* const engines_KR[] = {
+    &duckduckgo,
     &google,
     &naver,
     &bing,
@@ -514,15 +520,16 @@ const PrepopulatedEngine* const engines_
 
 // Kuwait
 const PrepopulatedEngine* const engines_KW[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Kazakhstan
 const PrepopulatedEngine* const engines_KZ[] = {
+    &duckduckgo,
     &google,
     &yandex_kz,
     &mail_ru,
@@ -532,78 +539,79 @@ const PrepopulatedEngine* const engines_
 
 // Lebanon
 const PrepopulatedEngine* const engines_LB[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Liechtenstein
 const PrepopulatedEngine* const engines_LI[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &yahoo,
     &ecosia,
 };
 
 // Lithuania
 const PrepopulatedEngine* const engines_LT[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_ru,
-    &duckduckgo,
 };
 
 // Luxembourg
 const PrepopulatedEngine* const engines_LU[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &yahoo,
     &ecosia,
 };
 
 // Latvia
 const PrepopulatedEngine* const engines_LV[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yandex_ru,
     &yahoo,
-    &duckduckgo,
 };
 
 // Libya
 const PrepopulatedEngine* const engines_LY[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_com,
-    &duckduckgo,
 };
 
 // Morocco
 const PrepopulatedEngine* const engines_MA[] = {
+    &duckduckgo,
     &google,
     &yahoo_fr,
     &bing,
     &yandex_com,
-    &duckduckgo,
 };
 
 // Monaco
 const PrepopulatedEngine* const engines_MC[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &qwant,
 };
 
 // Moldova
 const PrepopulatedEngine* const engines_MD[] = {
+    &duckduckgo,
     &google,
     &yandex_ru,
     &mail_ru,
@@ -613,301 +621,302 @@ const PrepopulatedEngine* const engines_
 
 // Montenegro
 const PrepopulatedEngine* const engines_ME[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_ru,
-    &duckduckgo,
 };
 
 // Macedonia
 const PrepopulatedEngine* const engines_MK[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &baidu,
 };
 
 // Mexico
 const PrepopulatedEngine* const engines_MX[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_mx,
-    &duckduckgo,
     &ecosia,
 };
 
 // Malaysia
 const PrepopulatedEngine* const engines_MY[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_my,
-    &duckduckgo,
     &baidu,
 };
 
 // Nicaragua
 const PrepopulatedEngine* const engines_NI[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Netherlands
 const PrepopulatedEngine* const engines_NL[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &yahoo_nl,
     &ecosia,
 };
 
 // Norway
 const PrepopulatedEngine* const engines_NO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // New Zealand
 const PrepopulatedEngine* const engines_NZ[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_nz,
-    &duckduckgo,
     &ecosia,
 };
 
 // Oman
 const PrepopulatedEngine* const engines_OM[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ask,
 };
 
 // Panama
 const PrepopulatedEngine* const engines_PA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Peru
 const PrepopulatedEngine* const engines_PE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_es,
     &ecosia,
-    &duckduckgo,
 };
 
 // Philippines
 const PrepopulatedEngine* const engines_PH[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &ecosia,
-    &duckduckgo,
 };
 
 // Pakistan
 const PrepopulatedEngine* const engines_PK[] = {
+    &duckduckgo,
     &google,
     &yahoo,
     &bing,
-    &duckduckgo,
     &yandex_com,
 };
 
 // Poland
 const PrepopulatedEngine* const engines_PL[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
 };
 
 // Puerto Rico
 const PrepopulatedEngine* const engines_PR[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Portugal
 const PrepopulatedEngine* const engines_PT[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Paraguay
 const PrepopulatedEngine* const engines_PY[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Qatar
 const PrepopulatedEngine* const engines_QA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Romania
 const PrepopulatedEngine* const engines_RO[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Serbia
 const PrepopulatedEngine* const engines_RS[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Russia
 const PrepopulatedEngine* const engines_RU[] = {
+    &duckduckgo,
     &google,
     &yandex_ru,
     &mail_ru,
     &bing,
-    &duckduckgo,
 };
 
 // Rwanda
 const PrepopulatedEngine* const engines_RW[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Saudi Arabia
 const PrepopulatedEngine* const engines_SA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_com,
-    &duckduckgo,
 };
 
 // Sweden
 const PrepopulatedEngine* const engines_SE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo_se,
-    &duckduckgo,
     &ecosia,
 };
 
 // Singapore
 const PrepopulatedEngine* const engines_SG[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &baidu,
-    &duckduckgo,
 };
 
 // Slovenia
 const PrepopulatedEngine* const engines_SI[] = {
+    &duckduckgo,
     &google,
     &bing,
-    &duckduckgo,
     &yahoo,
 };
 
 // Slovakia
 const PrepopulatedEngine* const engines_SK[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &seznam_cz,
 };
 
 // El Salvador
 const PrepopulatedEngine* const engines_SV[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Syria
 const PrepopulatedEngine* const engines_SY[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_ru,
-    &duckduckgo,
 };
 
 // Thailand
 const PrepopulatedEngine* const engines_TH[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &baidu,
 };
 
 // Tunisia
 const PrepopulatedEngine* const engines_TN[] = {
+    &duckduckgo,
     &google,
     &yahoo_fr,
     &bing,
-    &duckduckgo,
     &yandex_ru,
 };
 
 // Turkey
 const PrepopulatedEngine* const engines_TR[] = {
+    &duckduckgo,
     &google,
     &yandex_tr,
     &yahoo_tr,
     &bing,
-    &duckduckgo,
 };
 
 // Trinidad and Tobago
 const PrepopulatedEngine* const engines_TT[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Taiwan
 const PrepopulatedEngine* const engines_TW[] = {
+    &duckduckgo,
     &google,
     &yahoo_tw,
     &bing,
@@ -917,51 +926,52 @@ const PrepopulatedEngine* const engines_
 
 // Tanzania
 const PrepopulatedEngine* const engines_TZ[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Ukraine
 const PrepopulatedEngine* const engines_UA[] = {
+    &duckduckgo,
     &google,
     &yandex_ua,
     &bing,
-    &duckduckgo,
     &yahoo,
 };
 
 // United States
 const PrepopulatedEngine* const engines_US[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Uruguay
 const PrepopulatedEngine* const engines_UY[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Venezuela
 const PrepopulatedEngine* const engines_VE[] = {
+    &duckduckgo,
     &google,
     &yahoo_ve,
     &bing,
-    &duckduckgo,
     &ecosia,
 };
 
 // Vietnam
 const PrepopulatedEngine* const engines_VN[] = {
+    &duckduckgo,
     &google,
     &coccoc,
     &bing,
@@ -971,28 +981,28 @@ const PrepopulatedEngine* const engines_
 
 // Yemen
 const PrepopulatedEngine* const engines_YE[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
     &yandex_ru,
-    &duckduckgo,
 };
 
 // South Africa
 const PrepopulatedEngine* const engines_ZA[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ecosia,
 };
 
 // Zimbabwe
 const PrepopulatedEngine* const engines_ZW[] = {
+    &duckduckgo,
     &google,
     &bing,
     &yahoo,
-    &duckduckgo,
     &ask,
 };
 // ----------------------------------------------------------------------------
