--- a/chrome/browser/ui/startup/startup_tab_provider.cc
+++ b/chrome/browser/ui/startup/startup_tab_provider.cc
@@ -144,7 +144,7 @@ bool IsChromeControlledNtpUrl(const GURL
 
 StartupTabs StartupTabProviderImpl::GetOnboardingTabs(Profile* profile) const {
 // Chrome OS has its own welcome flow provided by OOBE.
-#if BUILDFLAG(IS_CHROMEOS_ASH)
+#if 1
   return StartupTabs();
 #else
   if (!profile)
