--- a/chrome/common/chrome_content_client.cc
+++ b/chrome/common/chrome_content_client.cc
@@ -95,11 +95,14 @@
 #if BUILDFLAG(ENABLE_WIDEVINE) && defined(OS_LINUX)
 #include "base/no_destructor.h"
 #include "chrome/common/media/cdm_manifest.h"
+#if !BUILDFLAG(ENABLE_WIDEVINE_CDM_COMPONENT)
+#define REGISTER_BUNDLED_WIDEVINE_CDM
 #include "third_party/widevine/cdm/widevine_cdm_common.h"  // nogncheck
 // TODO(crbug.com/663554): Needed for WIDEVINE_CDM_VERSION_STRING. Support
 // component updated CDM on all desktop platforms and remove this.
 // This file is In SHARED_INTERMEDIATE_DIR.
 #include "widevine_cdm_version.h"  // nogncheck
+#endif
 #if !defined(OS_CHROMEOS)
 #include "chrome/common/media/component_widevine_cdm_hint_file_linux.h"
 #endif  // !defined(OS_CHROMEOS)
--- a/third_party/widevine/cdm/widevine_cdm_version.h
+++ b/third_party/widevine/cdm/widevine_cdm_version.h
@@ -12,4 +12,6 @@
 //  - WIDEVINE_CDM_VERSION_STRING (with the version of the CDM that's available
 //    as a string, e.g., "1.0.123.456").
 
+#define WIDEVINE_CDM_VERSION_STRING "unknown"
+
 #endif  // WIDEVINE_CDM_VERSION_H_
