--- a/components/component_updater/configurator_impl.cc
+++ b/components/component_updater/configurator_impl.cc
@@ -65,14 +65,7 @@ int ConfiguratorImpl::UpdateDelay() cons
 }
 
 std::vector<GURL> ConfiguratorImpl::UpdateUrl() const {
-  if (url_source_override_.is_valid()) {
-    return {GURL(url_source_override_)};
-  }
-
-  std::vector<GURL> urls{GURL(kUpdaterDefaultUrl), GURL(kUpdaterFallbackUrl)};
-  if (require_encryption_)
-    update_client::RemoveUnsecureUrls(&urls);
-
+  std::vector<GURL> urls;
   return urls;
 }
 
