--- ./chrome/browser/browser_process_impl.cc
+++ ./chrome/browser/browser_process_impl.cc.new
@@ -1082,7 +1082,6 @@
         local_state(), chrome::GetChannel(), GetApplicationLocale(),
         system_request_context(), switches::kDisableBackgroundNetworking,
         base::Bind(safe_json::SafeJsonParser::Parse)));
-    promo_resource_service_->StartAfterDelay();
   }
 
 #if !defined(OS_ANDROID) && !defined(OS_IOS)



--- ./chrome/browser/signin/signin_promo.cc
+++ ./chrome/browser/signin/signin_promo.cc.new
@@ -105,6 +105,7 @@ bool ShouldShowPromo(Profile* profile) {
 
 bool ShouldShowPromoAtStartup(Profile* profile, bool is_new_profile) {
   DCHECK(profile);
+  return false;
 
   // Don't show if the profile is an incognito.
   if (profile->IsOffTheRecord())


--- ./components/web_resource/notification_promo.cc
+++ ./components/web_resource/notification_promo.cc.new
@@ -30,7 +30,7 @@ namespace {
 
 const int kDefaultGroupSize = 100;
 
-const char promo_server_url[] = "https://clients3.google.com/crsignal/client";
+const char promo_server_url[] = "about:blank";
 
 // The name of the preference that stores the promotion object.
 const char kPrefPromoObject[] = "promo";
