--- ./chrome/browser/profiles/avatar_menu.cc
+++ ./chrome/browser/profiles/avatar_menu.cc.new
@@ -93,12 +93,7 @@
   // TODO: Eliminate this ifdef. Add a delegate interface for the menu which
   // would also help remove the Browser dependency in AvatarMenuActions
   // implementations.
-#if defined(OS_CHROMEOS)
-  // On ChromeOS the menu will not be shown.
   return false;
-#else
-  return true;
-#endif
 }
 
 bool AvatarMenu::CompareItems(const Item* item1, const Item* item2) {


--- ./chrome/browser/ui/webui/options/browser_options_handler.cc
+++ ./chrome/browser/ui/webui/options/browser_options_handler.cc.new
@@ -1443,7 +1443,7 @@
   SigninManagerBase* signin = SigninManagerFactory::GetForProfile(profile);
   DCHECK(signin);
   sync_status->SetBoolean("signoutAllowed", !signout_prohibited);
-  sync_status->SetBoolean("signinAllowed", signin->IsSigninAllowed());
+  sync_status->SetBoolean("signinAllowed", false);
   sync_status->SetBoolean("syncSystemEnabled", (service != NULL));
   sync_status->SetBoolean("setupCompleted",
                           service && service->HasSyncSetupCompleted());

