--- ./chrome/browser/ui/startup/startup_browser_creator_impl.cc
+++ ./chrome/browser/ui/startup/startup_browser_creator_impl.cc.new
@@ -852,8 +852,6 @@
         // Replace magic names for the actual urls.
         if (it->host() == "new_tab_page") {
           startup_urls->push_back(GURL(chrome::kChromeUINewTabURL));
-        } else if (it->host() == "welcome_page") {
-          startup_urls->push_back(internals::GetWelcomePageURL());
         } else {
           startup_urls->push_back(*it);
         }
@@ -868,10 +866,6 @@
   if (startup_urls->empty()) {
     AddSpecialURLs(startup_urls);
     startup_urls->push_back(GURL(chrome::kChromeUINewTabURL));
-
-    // Special case the FIRST_RUN_LAST_TAB case of the welcome page.
-    if (welcome_run_type_ == WelcomeRunType::FIRST_RUN_LAST_TAB)
-      startup_urls->push_back(internals::GetWelcomePageURL());
   }
 
   if (signin::ShouldShowPromoAtStartup(profile_, is_first_run_)) {
@@ -905,10 +899,6 @@
 
 void StartupBrowserCreatorImpl::AddSpecialURLs(
     std::vector<GURL>* url_list) const {
-  // Optionally include the welcome page.
-  if (welcome_run_type_ == WelcomeRunType::FIRST_TAB)
-    url_list->insert(url_list->begin(), internals::GetWelcomePageURL());
-
   // If this Profile is marked for a reset prompt, ensure the reset
   // settings dialog appears.
   if (CheckAndClearProfileResetTrigger())


