--- a/components/autofill/core/browser/autofill_download_manager.cc
+++ b/components/autofill/core/browser/autofill_download_manager.cc
@@ -86,11 +86,6 @@ std::string RequestTypeToString(AutofillDownloadManager::RequestType type) {
   return std::string();
 }
 
-GURL GetRequestUrl(AutofillDownloadManager::RequestType request_type) {
-  return GURL("trk:158:https://clients1.9oo91e.qjz9zk/tbproxy/af/" +
-              RequestTypeToString(request_type) + "?client=" + kClientName);
-}
-
 }  // namespace
 
 struct AutofillDownloadManager::FormRequestData {
@@ -174,6 +169,7 @@ bool AutofillDownloadManager::StartUploadRequest(
 
 bool AutofillDownloadManager::StartRequest(
     const FormRequestData& request_data) {
+#if 0
   net::URLRequestContextGetter* request_context =
       driver_->GetURLRequestContext();
   DCHECK(request_context);
@@ -215,7 +211,7 @@ bool AutofillDownloadManager::StartRequest(
           << RequestTypeToString(request_data.request_type)
           << " request (compression " << compression_ratio
           << "): " << request_data.payload;
-
+#endif
   return true;
 }
 
