--- ./components/component_updater/configurator_impl.cc
+++ ./components/component_updater/configurator_impl.cc.new
@@ -153,14 +153,6 @@
 
 std::vector<GURL> ConfiguratorImpl::UpdateUrl() const {
   std::vector<GURL> urls;
-  if (url_source_override_.is_valid()) {
-    urls.push_back(GURL(url_source_override_));
-  } else {
-    urls.push_back(GURL(kUpdaterDefaultUrl));
-    if (fallback_to_alt_source_url_enabled_) {
-      urls.push_back(GURL(kUpdaterAltUrl));
-    }
-  }
   return urls;
 }
 
