--- ./components/autofill/core/browser/autofill_manager.cc
+++ ./components/autofill/core/browser/autofill_manager.cc.new
@@ -142,7 +142,7 @@ void AutofillManager::RegisterProfilePrefs(
     user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
       prefs::kAutofillEnabled,
-      true,
+      false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(prefs::kAutofillWalletSyncExperimentEnabled,
                                 false);


--- ./chrome/browser/io_thread.cc
+++ ./chrome/browser/io_thread.cc.new
@@ -1043,7 +1043,7 @@
                                std::string());
   registry->RegisterBooleanPref(prefs::kEnableReferrers, true);
   data_reduction_proxy::RegisterPrefs(registry);
-  registry->RegisterBooleanPref(prefs::kBuiltInDnsClientEnabled, true);
+  registry->RegisterBooleanPref(prefs::kBuiltInDnsClientEnabled, false);
   registry->RegisterBooleanPref(prefs::kQuickCheckEnabled, true);
 }
 

--- ./extensions/common/extension.cc
+++ ./extensions/common/extension.cc.new
@@ -358,7 +358,7 @@
   if (extensions::Manifest::IsComponentLocation(location()) &&
       !base::CommandLine::ForCurrentProcess()->HasSwitch(
           switches::kShowComponentExtensionOptions)) {
-    return true;
+    return false;
   }
 
   // Always show unpacked extensions and apps.
@@ -367,7 +367,7 @@
 
   // Don't show apps that aren't visible in either launcher or ntp.
   if (is_app() && !ShouldDisplayInAppLauncher() && !ShouldDisplayInNewTabPage())
-    return true;
+    return false;
 
 #endif
   return false;


--- ./components/bookmarks/browser/bookmark_utils.cc
+++ ./components/bookmarks/browser/bookmark_utils.cc.new
@@ -430,7 +430,7 @@
 void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(
       prefs::kShowBookmarkBar,
-      false,
+      true,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(prefs::kEditBookmarksEnabled, true);
   registry->RegisterBooleanPref(
@@ -435,7 +435,7 @@
   registry->RegisterBooleanPref(prefs::kEditBookmarksEnabled, true);
   registry->RegisterBooleanPref(
       prefs::kShowAppsShortcutInBookmarkBar,
-      true,
+      false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(
       prefs::kShowManagedBookmarksInBookmarkBar,


--- ./chrome/browser/profiles/profile.cc
+++ ./chrome/browser/profiles/profile.cc.new
@@ -89,7 +89,7 @@
   registry->RegisterStringPref(prefs::kSessionExitType, std::string());
   registry->RegisterBooleanPref(
       prefs::kSafeBrowsingEnabled,
-      true,
+      false,
       user_prefs::PrefRegistrySyncable::SYNCABLE_PREF);
   registry->RegisterBooleanPref(prefs::kSafeBrowsingExtendedReportingEnabled,
                                 false);
@@ -105,7 +105,7 @@
   // This pref is intentionally outside the above #if. That flag corresponds
   // to the Notifier extension and does not gate the launcher page.
   // TODO(skare): Remove or rename ENABLE_GOOGLE_NOW: http://crbug.com/459827.
-  registry->RegisterBooleanPref(prefs::kGoogleNowLauncherEnabled, true);
+  registry->RegisterBooleanPref(prefs::kGoogleNowLauncherEnabled, false);
   registry->RegisterBooleanPref(prefs::kDisableExtensions, false);
 #if defined(ENABLE_EXTENSIONS)
   registry->RegisterBooleanPref(extensions::pref_names::kAlertsInitialized,



--- ./chrome/service/cloud_print/connector_settings.cc
+++ ./chrome/service/cloud_print/connector_settings.cc.new
@@ -60,7 +60,7 @@
   DCHECK(server_url_.is_valid());
 
   connect_new_printers_ = prefs->GetBoolean(
-      prefs::kCloudPrintConnectNewPrinters, true);
+      prefs::kCloudPrintConnectNewPrinters, false);
 
   xmpp_ping_enabled_ = prefs->GetBoolean(
       prefs::kCloudPrintXmppPingEnabled, false);

