--- ./chrome/browser/ui/webui/ntp/new_tab_page_sync_handler.cc
+++ ./chrome/browser/ui/webui/ntp/new_tab_page_sync_handler.cc.new
@@ -91,14 +91,9 @@
   SigninManagerBase* signin = SigninManagerFactory::GetForProfile(
       Profile::FromWebUI(web_ui()));
 
-  // Hide the sync status section if sync is managed or disabled entirely.
-  if (!sync_service_ ||
-      sync_service_->IsManaged() ||
-      !signin ||
-      !signin->IsSigninAllowed()) {
-    HideSyncStatusSection();
-    return;
-  }
+  // Hide the sync status section
+  HideSyncStatusSection();
+  return;
 
   // Don't show sync status if setup is not complete.
   if (!sync_service_->HasSyncSetupCompleted()) {

