Description: manpage updates/fixes
Forwarded: http://code.google.com/p/chromium/issues/detail?id=73591
Author: Daniel Echeverry <epsilon77@gmail.com>

--- a/chrome/app/resources/manpage.1.in
+++ b/chrome/app/resources/manpage.1.in
@@ -1,5 +1,5 @@
-." This file is processed by chrome.gyp to generate manpages in the
-." build diretory.
+.\" This file is processed by chrome.gyp to generate manpages in the
+.\" build diretory.
 .TH @@FILENAME@@ 1 "" "" "USER COMMANDS"
 
 .SH NAME
@@ -20,6 +20,23 @@ This manpage only describes invocation,
 @@NAME@@ has hundreds of undocumented command-line flags that are added
 and removed at the whim of the developers.  Here, we document relatively
 stable flags.
+
+.TP
+\fB\-h \fRor\fB \-\-help\fR
+Show help output.
+
+.TP
+\fB\-g \fRor\fB \-\-debug\fR
+Start a debugging session within /usr/bin/gdb.
+
+.TP
+\fB\-\-temp-profile\fR
+Use a throw-away/temporary profile for this session.
+
+.TP
+\fB\-\-enable-remote-extensions\fR
+Allow installation and updates of remote extensions.
+
 .TP
 \fB\-\-user\-data\-dir\fR=\fIDIR\fR
 Specifies the directory that user data (your "profile") is kept in.
@@ -110,6 +127,7 @@ as
 See the GTK documentation for more:
 .IP
 <http://library.gnome.org/devel/gtk/stable/gtk-running.html>
+.IP
 <http://library.gnome.org/devel/gtk/stable/gtk-x11.html>
 
 .SH ENVIRONMENT
