# STATUS_SUCCESS conflicts with sandbox/win/src/nt_internals.h

--- a/extensions/browser/api/feedback_private/feedback_private_api.cc
+++ b/extensions/browser/api/feedback_private/feedback_private_api.cc
@@ -337,7 +337,7 @@ void FeedbackPrivateSendFeedbackFunction
     bool success) {
   Respond(TwoArguments(
       std::make_unique<base::Value>(feedback_private::ToString(
-          success ? feedback_private::STATUS_SUCCESS
+          success ? feedback_private::STATUS_SUCCESS1
                   : feedback_private::STATUS_DELAYED)),
       std::make_unique<base::Value>(feedback_private::ToString(type))));
   if (!success) {
--- a/extensions/common/api/feedback_private.idl
+++ b/extensions/common/api/feedback_private.idl
@@ -85,7 +85,7 @@ namespace feedbackPrivate {
   };
 
   // Status of the sending of a feedback report.
-  enum Status {success, delayed};
+  enum Status {success1, delayed};
 
   // The type of the landing page shown to the user when the feedback report is
   // successfully sent.
