# Fix building GN on Windows
# Author: shiromichi on GitHub

--- a/tools/gn/bootstrap/bootstrap.py
+++ b/tools/gn/bootstrap/bootstrap.py
@@ -64,7 +64,7 @@ def main(argv):
   else:
     build_rel = os.path.join('out', 'Release')
   out_dir = os.path.join(SRC_ROOT, build_rel)
-  gn_path = options.output or os.path.join(out_dir, 'gn')
+  gn_path = options.output or os.path.join(out_dir, 'gn.exe')
   gn_build_dir = os.path.join(out_dir, 'gn_build')
 
   cmd = [
@@ -81,11 +81,11 @@ def main(argv):
       os.path.join(BOOTSTRAP_DIR, 'last_commit_position.h'), gn_build_dir)
   if options.jobs:
     subprocess.check_call(
-        ['ninja', '-C', gn_build_dir, 'gn', '-w', 'dupbuild=err', '-j'+str(options.jobs)])
+        ['ninja', '-C', gn_build_dir, 'gn.exe', '-w', 'dupbuild=err', '-j'+str(options.jobs)])
   else:
     subprocess.check_call(
-        ['ninja', '-C', gn_build_dir, 'gn', '-w', 'dupbuild=err'])
-    shutil.copy2(os.path.join(gn_build_dir, 'gn'), gn_path)
+         ['ninja', '-C', gn_build_dir, 'gn.exe', '-w', 'dupbuild=err'])
+    shutil.copy2(os.path.join(gn_build_dir, 'gn.exe'), gn_path)
 
   if not options.skip_generate_buildfiles:
     gn_gen_args = options.gn_gen_args or ''
--- a/tools/gn/build/build_win.ninja.template
+++ b/tools/gn/build/build_win.ninja.template
@@ -1,12 +1,12 @@
 rule cc
-  command = ninja -t msvc -- $cc /nologo /showIncludes /FC @${out}.rsp /c ${in} /Fo${out}
+  command = ninja -t msvc -- $cc /utf-8 /nologo /showIncludes /FC @${out}.rsp /c ${in} /Fo${out}
   description = CC ${out}
   rspfile = ${out}.rsp
   rspfile_content = ${defines} ${includes} ${cflags} ${cflags_c}
   deps = msvc
 
 rule cxx
-  command = ninja -t msvc -- $cxx /nologo /showIncludes /FC @${out}.rsp /c ${in} /Fo${out}
+  command = ninja -t msvc -- $cxx /utf-8 /nologo /showIncludes /FC @${out}.rsp /c ${in} /Fo${out}
   description = CXX ${out}
   rspfile = ${out}.rsp
   rspfile_content = ${defines} ${includes} ${cflags} ${cflags_cc}
