Taken from upstream commit chromium@d9a89d6e5cfa15c2a58091f972ccdd6fde57ebb3.

--- a/services/preferences/tracked/tracked_persistent_pref_store_factory.cc
+++ b/services/preferences/tracked/tracked_persistent_pref_store_factory.cc
@@ -103,12 +103,18 @@ PersistentPrefStore* CreateTrackedPersis
   }
 #endif
 
-  mojo::Remote<prefs::mojom::TrackedPreferenceValidationDelegate>
-      validation_delegate;
-  validation_delegate.Bind(std::move(config->validation_delegate));
-  auto validation_delegate_ref = base::MakeRefCounted<base::RefCountedData<
-      mojo::Remote<prefs::mojom::TrackedPreferenceValidationDelegate>>>(
-      std::move(validation_delegate));
+  scoped_refptr<base::RefCountedData<
+      mojo::Remote<prefs::mojom::TrackedPreferenceValidationDelegate>>>
+      validation_delegate_ref;
+  if (config->validation_delegate) {
+    mojo::Remote<prefs::mojom::TrackedPreferenceValidationDelegate>
+        validation_delegate;
+    validation_delegate.Bind(std::move(config->validation_delegate));
+    validation_delegate_ref = base::MakeRefCounted<base::RefCountedData<
+        mojo::Remote<prefs::mojom::TrackedPreferenceValidationDelegate>>>(
+        std::move(validation_delegate));
+  }
+
   std::unique_ptr<PrefHashFilter> unprotected_pref_hash_filter(
       new PrefHashFilter(CreatePrefHashStore(*config, false),
                          GetExternalVerificationPrefHashStorePair(
