--- a/chrome/browser/extensions/BUILD.gn
+++ b/chrome/browser/extensions/BUILD.gn
@@ -957,6 +957,7 @@ source_set("extensions") {
       "//components/resources",
       "//components/safe_browsing:buildflags",
       "//components/safe_browsing/content/browser/web_ui:web_ui",
+      "//components/safe_browsing/content/common/proto:download_file_types_proto",
       "//components/safe_browsing/core/common/proto:csd_proto",
       "//components/search_engines",
       "//components/services/app_service",
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -544,6 +544,7 @@ static_library("url_constants") {
     "//components/optimization_guide/optimization_guide_internals/webui:url_constants",
     "//components/password_manager/content/common",
     "//components/safe_browsing/core/common",
+    "//components/supervised_user/core/common:buildflags",
     "//device/vr/buildflags",
   ]
 }
--- a/components/component_updater/installer_policies/BUILD.gn
+++ b/components/component_updater/installer_policies/BUILD.gn
@@ -49,6 +49,7 @@ static_library("installer_policies_no_co
     "//components/update_client",
     "//mojo/public/cpp/base:protobuf_support",
     "//services/network/public/cpp",
+    "//third_party/re2",
   ]
 
   # Disallow depending on content.
--- a/components/plus_addresses/BUILD.gn
+++ b/components/plus_addresses/BUILD.gn
@@ -127,6 +127,7 @@ source_set("plus_addresses") {
     "//net",
     "//services/data_decoder/public/cpp",
     "//services/network/public/cpp",
+    "//third_party/re2",
     "//ui/base",
   ]
   public_deps = [
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -111,6 +111,7 @@ source_set("browser") {
     "//components/filename_generation",
     "//components/fingerprinting_protection_filter/interventions/common:features",
     "//components/input",
+    "//components/lens:buildflags",
     "//components/language/core/common",
     "//components/language_detection/content/common",
     "//components/link_header_util",
--- a/services/passage_embeddings/BUILD.gn
+++ b/services/passage_embeddings/BUILD.gn
@@ -38,9 +38,9 @@ source_set("passage_embeddings") {
     deps += [ "//services/on_device_model:ml_internal_buildflags" ]
   }
 
+  deps += [ "//components/optimization_guide:machine_learning_tflite_buildflags" ]
   if (build_with_tflite_lib) {
     deps += [
-      "//components/optimization_guide:machine_learning_tflite_buildflags",
       "//components/optimization_guide/core:machine_learning",
     ]
   }
