description: add constexpr to methods where it is required
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc
+++ b/chrome/browser/resource_coordinator/tab_manager_stats_collector.cc
@@ -641,6 +641,6 @@ const char
         "TabManager.SessionOverlap.BackgroundTabOpening";
 
 // static
-const base::TimeDelta TabManagerStatsCollector::kLowFrequencySamplingInterval;
+constexpr base::TimeDelta TabManagerStatsCollector::kLowFrequencySamplingInterval;
 
 }  // namespace resource_coordinator
--- a/third_party/blink/renderer/core/animation/animation_time_delta.h
+++ b/third_party/blink/renderer/core/animation/animation_time_delta.h
@@ -53,7 +53,7 @@ class CORE_EXPORT AnimationTimeDelta {
     return AnimationTimeDelta(std::numeric_limits<double>::infinity());
   }
 
-  double InSecondsF() const { return delta_; }
+  constexpr double InSecondsF() const { return delta_; }
   double InMillisecondsF() const { return delta_ * 1000; }
 
   bool is_max() const {
