description: constructor of Algorithm needs to be public
author: Michael Gilbert <mgilbert@debian.org>

--- a/third_party/blink/renderer/core/streams/transform_stream.cc
+++ b/third_party/blink/renderer/core/streams/transform_stream.cc
@@ -37,7 +37,6 @@ class TransformStream::Algorithm : publi
     ScriptFunction::Trace(visitor);
   }
 
- protected:
   Algorithm(TransformStreamTransformer* transformer,
             ScriptState* script_state,
             ExceptionState& exception_state)
@@ -46,6 +45,7 @@ class TransformStream::Algorithm : publi
         interface_name_(exception_state.InterfaceName()),
         property_name_(exception_state.PropertyName()) {}
 
+ protected:
   // AlgorithmScope holds the stack-allocated objects used by the CallRaw()
   // methods for FlushAlgorithm and TransformAlgorithm.
   class AlgorithmScope {
