description: gcc6 doesn't support initializer list member assignment
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/ui/views/omnibox/omnibox_text_view.cc
+++ b/chrome/browser/ui/views/omnibox/omnibox_text_view.cc
@@ -175,7 +175,8 @@ void ApplyTextStyleForType(SuggestionAns
       style = {part_color, .baseline = gfx::SUPERIOR};
       break;
     case SuggestionAnswer::TextStyle::BOLD:
-      style = {part_color, .weight = gfx::Font::Weight::BOLD};
+      style.color = part_color;
+      style.weight = gfx::Font::Weight::BOLD;
       break;
     case SuggestionAnswer::TextStyle::NORMAL:
     case SuggestionAnswer::TextStyle::NORMAL_DIM:
