description: jumbo build has trouble with these namespaces
author: Michael Gilbert <mgilbert@debian.org>

--- a/chrome/browser/apps/platform_apps/api/sync_file_system/extension_sync_event_observer.h
+++ b/chrome/browser/apps/platform_apps/api/sync_file_system/extension_sync_event_observer.h
@@ -81,8 +81,10 @@ class ExtensionSyncEventObserver : publi
 }  // namespace api
 }  // namespace chrome_apps
 
+namespace extensions {
 template <>
-void extensions::BrowserContextKeyedAPIFactory<
+void BrowserContextKeyedAPIFactory<
     chrome_apps::api::ExtensionSyncEventObserver>::DeclareFactoryDependencies();
+}
 
 #endif  // CHROME_BROWSER_APPS_PLATFORM_APPS_API_SYNC_FILE_SYSTEM_EXTENSION_SYNC_EVENT_OBSERVER_H_
